{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pybedtools\n",
    "import subprocess\n",
    "import pandas as pd\n",
    "import statsmodels.api as sm\n",
    "from scipy import stats\n",
    "import plotnine as p9\n",
    "import pysam\n",
    "\n",
    "%matplotlib inline\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "pd.set_option('display.max_rows', None)\n",
    "pd.set_option('display.max_columns', None)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get ADPD SNPs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>r2_with_ld_tag</th>\n",
       "      <th>ld_tag_chr</th>\n",
       "      <th>ld_tag_pos</th>\n",
       "      <th>source_gwas</th>\n",
       "      <th>source</th>\n",
       "      <th>snp_id</th>\n",
       "      <th>locus_num</th>\n",
       "      <th>number_ld_buddies</th>\n",
       "      <th>ld_tag_locus</th>\n",
       "      <th>rsid</th>\n",
       "      <th>chrom_hg19</th>\n",
       "      <th>snp_pos_hg19</th>\n",
       "      <th>file</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>noneffect_allele</th>\n",
       "      <th>direction</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>has_coloc</th>\n",
       "      <th>direct_atac_overlap_narrow_tissue_regions</th>\n",
       "      <th>containing_atac_tissues_narrow_tissue_regions</th>\n",
       "      <th>nearest_atac_tissue_narrow_tissue_regions</th>\n",
       "      <th>start_narrow_tissue_regions</th>\n",
       "      <th>end_narrow_tissue_regions</th>\n",
       "      <th>dist_narrow_tissue_regions</th>\n",
       "      <th>direct_atac_overlap_broad_tissue_regions</th>\n",
       "      <th>containing_atac_tissues_broad_tissue_regions</th>\n",
       "      <th>nearest_atac_tissue_broad_tissue_regions</th>\n",
       "      <th>start_broad_tissue_regions</th>\n",
       "      <th>end_broad_tissue_regions</th>\n",
       "      <th>dist_broad_tissue_regions</th>\n",
       "      <th>direct_atac_overlap_single_cell</th>\n",
       "      <th>containing_atac_tissues_single_cell</th>\n",
       "      <th>nearest_atac_tissue_single_cell</th>\n",
       "      <th>start_single_cell</th>\n",
       "      <th>end_single_cell</th>\n",
       "      <th>dist_single_cell</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10</td>\n",
       "      <td>102207833</td>\n",
       "      <td>0.805869</td>\n",
       "      <td>10</td>\n",
       "      <td>102255522</td>\n",
       "      <td>Nalls_23andMe</td>\n",
       "      <td>LD</td>\n",
       "      <td>10_102207833</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>10_102255522</td>\n",
       "      <td>rs10883717</td>\n",
       "      <td>10.0</td>\n",
       "      <td>103967590.0</td>\n",
       "      <td>/users/mgloud/projects/ad-pd/data/gwas/prepare...</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>0.014434</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>CAUD</td>\n",
       "      <td>102225620</td>\n",
       "      <td>102226734</td>\n",
       "      <td>17780.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>ALLO</td>\n",
       "      <td>102225838</td>\n",
       "      <td>102226054</td>\n",
       "      <td>18005.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>microglia</td>\n",
       "      <td>102213693</td>\n",
       "      <td>102214072</td>\n",
       "      <td>5860.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10</td>\n",
       "      <td>102245653</td>\n",
       "      <td>0.987490</td>\n",
       "      <td>10</td>\n",
       "      <td>102255522</td>\n",
       "      <td>Nalls_23andMe</td>\n",
       "      <td>LD</td>\n",
       "      <td>10_102245653</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>10_102255522</td>\n",
       "      <td>rs2296887</td>\n",
       "      <td>10.0</td>\n",
       "      <td>104005410.0</td>\n",
       "      <td>/users/mgloud/projects/ad-pd/data/gwas/prepare...</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>0.006977</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>CAUD,HIPP,MDFG,PARL,PTMN,SMTG,SUNI</td>\n",
       "      <td>CAUD,HIPP,MDFG,PARL,PTMN,SMTG,SUNI</td>\n",
       "      <td>102244764</td>\n",
       "      <td>102246025</td>\n",
       "      <td>258.5</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>ALLO</td>\n",
       "      <td>102245836</td>\n",
       "      <td>102246040</td>\n",
       "      <td>183.0</td>\n",
       "      <td>True</td>\n",
       "      <td>astrocytes,doublets,excitatory_neurons,inhibit...</td>\n",
       "      <td>astrocytes,doublets,excitatory_neurons,inhibit...</td>\n",
       "      <td>102244973</td>\n",
       "      <td>102245978</td>\n",
       "      <td>177.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10</td>\n",
       "      <td>102250385</td>\n",
       "      <td>0.809348</td>\n",
       "      <td>10</td>\n",
       "      <td>102255522</td>\n",
       "      <td>Nalls_23andMe</td>\n",
       "      <td>LD</td>\n",
       "      <td>10_102250385</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>10_102255522</td>\n",
       "      <td>rs7913281</td>\n",
       "      <td>10.0</td>\n",
       "      <td>104010142.0</td>\n",
       "      <td>/users/mgloud/projects/ad-pd/data/gwas/prepare...</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>+</td>\n",
       "      <td>0.012859</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>CAUD</td>\n",
       "      <td>102244791</td>\n",
       "      <td>102246164</td>\n",
       "      <td>4221.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>ALLO</td>\n",
       "      <td>102245836</td>\n",
       "      <td>102246040</td>\n",
       "      <td>4345.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>opcs</td>\n",
       "      <td>102247774</td>\n",
       "      <td>102248243</td>\n",
       "      <td>2142.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10</td>\n",
       "      <td>102251214</td>\n",
       "      <td>0.809348</td>\n",
       "      <td>10</td>\n",
       "      <td>102255522</td>\n",
       "      <td>Nalls_23andMe</td>\n",
       "      <td>LD</td>\n",
       "      <td>10_102251214</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>10_102255522</td>\n",
       "      <td>rs10883720</td>\n",
       "      <td>10.0</td>\n",
       "      <td>104010971.0</td>\n",
       "      <td>/users/mgloud/projects/ad-pd/data/gwas/prepare...</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>0.012617</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>CAUD</td>\n",
       "      <td>102244791</td>\n",
       "      <td>102246164</td>\n",
       "      <td>5050.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>ALLO</td>\n",
       "      <td>102245836</td>\n",
       "      <td>102246040</td>\n",
       "      <td>5174.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>opcs</td>\n",
       "      <td>102247774</td>\n",
       "      <td>102248243</td>\n",
       "      <td>2971.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10</td>\n",
       "      <td>102255522</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>10</td>\n",
       "      <td>102255522</td>\n",
       "      <td>Nalls_23andMe</td>\n",
       "      <td>Nalls-Chang</td>\n",
       "      <td>10_102255522</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>10_102255522</td>\n",
       "      <td>rs10748818</td>\n",
       "      <td>10.0</td>\n",
       "      <td>104015279.0</td>\n",
       "      <td>/users/mgloud/projects/ad-pd/data/gwas/prepare...</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>+</td>\n",
       "      <td>0.006335</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>CAUD</td>\n",
       "      <td>102244791</td>\n",
       "      <td>102246164</td>\n",
       "      <td>9358.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>ALLO</td>\n",
       "      <td>102245836</td>\n",
       "      <td>102246040</td>\n",
       "      <td>9482.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>inhibitory_neurons</td>\n",
       "      <td>102255548</td>\n",
       "      <td>102256050</td>\n",
       "      <td>26.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chr        pos  r2_with_ld_tag  ld_tag_chr  ld_tag_pos    source_gwas  \\\n",
       "0   10  102207833        0.805869          10   102255522  Nalls_23andMe   \n",
       "1   10  102245653        0.987490          10   102255522  Nalls_23andMe   \n",
       "2   10  102250385        0.809348          10   102255522  Nalls_23andMe   \n",
       "3   10  102251214        0.809348          10   102255522  Nalls_23andMe   \n",
       "4   10  102255522        1.000000          10   102255522  Nalls_23andMe   \n",
       "\n",
       "        source        snp_id  locus_num  number_ld_buddies  ld_tag_locus  \\\n",
       "0           LD  10_102207833          1                  9  10_102255522   \n",
       "1           LD  10_102245653          1                  9  10_102255522   \n",
       "2           LD  10_102250385          1                  9  10_102255522   \n",
       "3           LD  10_102251214          1                  9  10_102255522   \n",
       "4  Nalls-Chang  10_102255522          1                  9  10_102255522   \n",
       "\n",
       "         rsid  chrom_hg19  snp_pos_hg19  \\\n",
       "0  rs10883717        10.0   103967590.0   \n",
       "1   rs2296887        10.0   104005410.0   \n",
       "2   rs7913281        10.0   104010142.0   \n",
       "3  rs10883720        10.0   104010971.0   \n",
       "4  rs10748818        10.0   104015279.0   \n",
       "\n",
       "                                                file effect_allele  \\\n",
       "0  /users/mgloud/projects/ad-pd/data/gwas/prepare...             T   \n",
       "1  /users/mgloud/projects/ad-pd/data/gwas/prepare...             T   \n",
       "2  /users/mgloud/projects/ad-pd/data/gwas/prepare...             G   \n",
       "3  /users/mgloud/projects/ad-pd/data/gwas/prepare...             G   \n",
       "4  /users/mgloud/projects/ad-pd/data/gwas/prepare...             G   \n",
       "\n",
       "  noneffect_allele direction    pvalue  has_coloc  \\\n",
       "0                C         -  0.014434      False   \n",
       "1                C         -  0.006977      False   \n",
       "2                A         +  0.012859      False   \n",
       "3                C         -  0.012617      False   \n",
       "4                A         +  0.006335      False   \n",
       "\n",
       "   direct_atac_overlap_narrow_tissue_regions  \\\n",
       "0                                      False   \n",
       "1                                       True   \n",
       "2                                      False   \n",
       "3                                      False   \n",
       "4                                      False   \n",
       "\n",
       "  containing_atac_tissues_narrow_tissue_regions  \\\n",
       "0                                          none   \n",
       "1            CAUD,HIPP,MDFG,PARL,PTMN,SMTG,SUNI   \n",
       "2                                          none   \n",
       "3                                          none   \n",
       "4                                          none   \n",
       "\n",
       "  nearest_atac_tissue_narrow_tissue_regions  start_narrow_tissue_regions  \\\n",
       "0                                      CAUD                    102225620   \n",
       "1        CAUD,HIPP,MDFG,PARL,PTMN,SMTG,SUNI                    102244764   \n",
       "2                                      CAUD                    102244791   \n",
       "3                                      CAUD                    102244791   \n",
       "4                                      CAUD                    102244791   \n",
       "\n",
       "   end_narrow_tissue_regions  dist_narrow_tissue_regions  \\\n",
       "0                  102226734                     17780.0   \n",
       "1                  102246025                       258.5   \n",
       "2                  102246164                      4221.0   \n",
       "3                  102246164                      5050.0   \n",
       "4                  102246164                      9358.0   \n",
       "\n",
       "   direct_atac_overlap_broad_tissue_regions  \\\n",
       "0                                     False   \n",
       "1                                     False   \n",
       "2                                     False   \n",
       "3                                     False   \n",
       "4                                     False   \n",
       "\n",
       "  containing_atac_tissues_broad_tissue_regions  \\\n",
       "0                                         none   \n",
       "1                                         none   \n",
       "2                                         none   \n",
       "3                                         none   \n",
       "4                                         none   \n",
       "\n",
       "  nearest_atac_tissue_broad_tissue_regions  start_broad_tissue_regions  \\\n",
       "0                                     ALLO                   102225838   \n",
       "1                                     ALLO                   102245836   \n",
       "2                                     ALLO                   102245836   \n",
       "3                                     ALLO                   102245836   \n",
       "4                                     ALLO                   102245836   \n",
       "\n",
       "   end_broad_tissue_regions  dist_broad_tissue_regions  \\\n",
       "0                 102226054                    18005.0   \n",
       "1                 102246040                      183.0   \n",
       "2                 102246040                     4345.0   \n",
       "3                 102246040                     5174.0   \n",
       "4                 102246040                     9482.0   \n",
       "\n",
       "   direct_atac_overlap_single_cell  \\\n",
       "0                            False   \n",
       "1                             True   \n",
       "2                            False   \n",
       "3                            False   \n",
       "4                            False   \n",
       "\n",
       "                 containing_atac_tissues_single_cell  \\\n",
       "0                                               none   \n",
       "1  astrocytes,doublets,excitatory_neurons,inhibit...   \n",
       "2                                               none   \n",
       "3                                               none   \n",
       "4                                               none   \n",
       "\n",
       "                     nearest_atac_tissue_single_cell  start_single_cell  \\\n",
       "0                                          microglia          102213693   \n",
       "1  astrocytes,doublets,excitatory_neurons,inhibit...          102244973   \n",
       "2                                               opcs          102247774   \n",
       "3                                               opcs          102247774   \n",
       "4                                 inhibitory_neurons          102255548   \n",
       "\n",
       "   end_single_cell  dist_single_cell  \n",
       "0        102214072            5860.0  \n",
       "1        102245978             177.5  \n",
       "2        102248243            2142.0  \n",
       "3        102248243            2971.0  \n",
       "4        102256050              26.0  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "(13773, 38)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "adpd_snps = pd.read_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/snps_final/191121_ld_buddies_table_stage3.tsv', sep='\\t')\n",
    "display(adpd_snps.head())\n",
    "display(adpd_snps.shape)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Make hg38 SNP BED files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2334</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39822150</td>\n",
       "      <td>39822151</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2335</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39822244</td>\n",
       "      <td>39822245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2336</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39822259</td>\n",
       "      <td>39822260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2337</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39822791</td>\n",
       "      <td>39822792</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2338</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39825224</td>\n",
       "      <td>39825225</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr     start       pos\n",
       "2334  chr1  39822150  39822151\n",
       "2335  chr1  39822244  39822245\n",
       "2336  chr1  39822259  39822260\n",
       "2337  chr1  39822791  39822792\n",
       "2338  chr1  39825224  39825225"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "(9707, 3)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "snps_bed = adpd_snps[['chr', 'pos']]\n",
    "snps_bed['start'] = snps_bed['pos'] - 1\n",
    "snps_bed = snps_bed[['chr', 'start', 'pos']]\n",
    "snps_bed['chr'] = 'chr' + snps_bed['chr'].astype(str)\n",
    "snps_bed.sort_values(by=['chr', 'start', 'pos'], inplace=True)\n",
    "snps_bed.drop_duplicates(inplace=True)\n",
    "snps_bed.to_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/snps/hg38_snps.bed', sep='\\t', index=False, header=False)\n",
    "display(snps_bed.head())\n",
    "display(snps_bed.shape)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Make No-chr hg19 SNP BED files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>pos</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>40287822</td>\n",
       "      <td>40287823</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>40287916</td>\n",
       "      <td>40287917</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>40287931</td>\n",
       "      <td>40287932</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>40288463</td>\n",
       "      <td>40288464</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>40290896</td>\n",
       "      <td>40290897</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chr     start       pos\n",
       "0   1  40287822  40287823\n",
       "1   1  40287916  40287917\n",
       "2   1  40287931  40287932\n",
       "3   1  40288463  40288464\n",
       "4   1  40290896  40290897"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "(9696, 3)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# ! module load ucsc_tools/latest <- Do on command line\n",
    "# ! liftOver /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/snps/hg38_snps.bed \\\n",
    "#   /mnt/data/annotations/liftOver/hg38/hg38ToHg19.over.chain.gz \\\n",
    "#   /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/snps/liftover_hg19_snps.bed \\\n",
    "#   unmapped_liftover_hg38_snps.bed\n",
    "\n",
    "liftover_hg19_snps = pd.read_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/snps/liftover_hg19_snps.bed', sep='\\t', header=None)\n",
    "liftover_hg19_snps.columns = ['chr', 'start', 'pos']\n",
    "liftover_hg19_snps['chr'] = liftover_hg19_snps['chr'].map(lambda x : x.replace('chr', ''))\n",
    "liftover_hg19_snps.to_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/snps/nochr_hg19_snps.bed', sep='\\t', header=False, index=False)\n",
    "display(liftover_hg19_snps.head())\n",
    "display(liftover_hg19_snps.shape)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Intersect 1KG VCF with hg19 ADPD SNPs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!  tabix -R /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/snps/nochr_hg19_snps.bed /mnt/lab_data3/soumyak/refs/1KG/ALL.wgs.phase3_shapeit2_mvncall_integrated_v5b.20130502.sites.vcf.gz > /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/snps/1KG_ADPD_hg19_snps.vcf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get hg38 Coordinates for 1KG intersected SNPs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsid</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>score</th>\n",
       "      <th>qc</th>\n",
       "      <th>info</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1239</th>\n",
       "      <td>.;esv3626682</td>\n",
       "      <td>T</td>\n",
       "      <td>&lt;CN0&gt;,&lt;CN2&gt;</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=0,1;AF=0,0.000199681;AN=5008;CS=DUP_gs;END=...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9850</th>\n",
       "      <td>DUP_uwash_chr5_60140563_60179087</td>\n",
       "      <td>A</td>\n",
       "      <td>&lt;CN0&gt;</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=0;AF=0;AN=5008;CS=DUP_uwash;END=60179087;NS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>esv3585792</td>\n",
       "      <td>G</td>\n",
       "      <td>&lt;CN0&gt;</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=1;AF=0.000199681;AN=5008;CIEND=0,500;CIPOS=...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>esv3585794</td>\n",
       "      <td>C</td>\n",
       "      <td>&lt;CN0&gt;</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=1;AF=0.000199681;AN=5008;CIEND=-23,24;CIPOS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>esv3587615</td>\n",
       "      <td>T</td>\n",
       "      <td>&lt;CN2&gt;</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=1;AF=0.000199681;AN=5008;CIEND=-150,150;CIP...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  rsid ref          alt  score    qc  \\\n",
       "1239                      .;esv3626682   T  <CN0>,<CN2>    100  PASS   \n",
       "9850  DUP_uwash_chr5_60140563_60179087   A        <CN0>    100  PASS   \n",
       "0                           esv3585792   G        <CN0>    100  PASS   \n",
       "45                          esv3585794   C        <CN0>    100  PASS   \n",
       "64                          esv3587615   T        <CN2>    100  PASS   \n",
       "\n",
       "                                                   info  \n",
       "1239  AC=0,1;AF=0,0.000199681;AN=5008;CS=DUP_gs;END=...  \n",
       "9850  AC=0;AF=0;AN=5008;CS=DUP_uwash;END=60179087;NS...  \n",
       "0     AC=1;AF=0.000199681;AN=5008;CIEND=0,500;CIPOS=...  \n",
       "45    AC=1;AF=0.000199681;AN=5008;CIEND=-23,24;CIPOS...  \n",
       "64    AC=1;AF=0.000199681;AN=5008;CIEND=-150,150;CIP...  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "(9928, 6)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "hg19_vcf = pd.read_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/snps/1KG_ADPD_hg19_snps.vcf', header=None, sep='\\t')\n",
    "hg19_vcf = hg19_vcf[[2, 3, 4, 5, 6, 7]]\n",
    "hg19_vcf.columns = ['rsid', 'ref', 'alt', 'score', 'qc', 'info']\n",
    "hg19_vcf.sort_values(by=['rsid', 'ref'], inplace=True)\n",
    "hg19_vcf.drop_duplicates(inplace=True)\n",
    "display(hg19_vcf.head())\n",
    "display(hg19_vcf.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>rsid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2334</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39822151</td>\n",
       "      <td>rs34640847</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2335</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39822245</td>\n",
       "      <td>rs36015266</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2336</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39822260</td>\n",
       "      <td>rs61779808</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2337</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39822792</td>\n",
       "      <td>rs61779809</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2338</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39825225</td>\n",
       "      <td>rs72666941</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr       pos        rsid\n",
       "2334  chr1  39822151  rs34640847\n",
       "2335  chr1  39822245  rs36015266\n",
       "2336  chr1  39822260  rs61779808\n",
       "2337  chr1  39822792  rs61779809\n",
       "2338  chr1  39825225  rs72666941"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "(9707, 3)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "adpd_snps_hg38 = adpd_snps[['chr', 'pos', 'rsid']]\n",
    "adpd_snps_hg38['chr'] = 'chr' + adpd_snps_hg38['chr'].astype('str')\n",
    "adpd_snps_hg38.sort_values(by=['chr', 'pos'], inplace=True)\n",
    "adpd_snps_hg38.drop_duplicates(subset=['rsid'], inplace=True)\n",
    "display(adpd_snps_hg38.head())\n",
    "display(adpd_snps_hg38.shape)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Make 1KG ADPD hg38 VCF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>rsid</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>score</th>\n",
       "      <th>qc</th>\n",
       "      <th>info</th>\n",
       "      <th>format</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39822151</td>\n",
       "      <td>rs34640847</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=595;AF=0.11881;AN=5008;NS=2504;DP=18110;EAS...</td>\n",
       "      <td>GT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39822245</td>\n",
       "      <td>rs36015266</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=595;AF=0.11881;AN=5008;NS=2504;DP=19296;EAS...</td>\n",
       "      <td>GT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39822260</td>\n",
       "      <td>rs61779808</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=493;AF=0.0984425;AN=5008;NS=2504;DP=19241;E...</td>\n",
       "      <td>GT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39822792</td>\n",
       "      <td>rs61779809</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=595;AF=0.11881;AN=5008;NS=2504;DP=19613;EAS...</td>\n",
       "      <td>GT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39825225</td>\n",
       "      <td>rs72666941</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=477;AF=0.0952476;AN=5008;NS=2504;DP=20561;E...</td>\n",
       "      <td>GT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr       pos        rsid ref alt  score    qc  \\\n",
       "0  chr1  39822151  rs34640847   G   A    100  PASS   \n",
       "1  chr1  39822245  rs36015266   C   T    100  PASS   \n",
       "2  chr1  39822260  rs61779808   T   A    100  PASS   \n",
       "3  chr1  39822792  rs61779809   C   T    100  PASS   \n",
       "4  chr1  39825225  rs72666941   C   A    100  PASS   \n",
       "\n",
       "                                                info format  \n",
       "0  AC=595;AF=0.11881;AN=5008;NS=2504;DP=18110;EAS...     GT  \n",
       "1  AC=595;AF=0.11881;AN=5008;NS=2504;DP=19296;EAS...     GT  \n",
       "2  AC=493;AF=0.0984425;AN=5008;NS=2504;DP=19241;E...     GT  \n",
       "3  AC=595;AF=0.11881;AN=5008;NS=2504;DP=19613;EAS...     GT  \n",
       "4  AC=477;AF=0.0952476;AN=5008;NS=2504;DP=20561;E...     GT  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "(9386, 9)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "merged = adpd_snps_hg38.merge(hg19_vcf, on='rsid')\n",
    "merged = merged.loc[merged['ref'].apply(lambda x : len(x) < 2)]\n",
    "merged = merged.loc[merged['alt'].apply(lambda x : len(x) < 2)]\n",
    "\n",
    "hg38_vcf = merged[['chr', 'pos', 'rsid', 'ref', 'alt', 'score', 'qc', 'info']]\n",
    "hg38_vcf['format'] = ['GT' for i in range(len(hg38_vcf))]\n",
    "hg38_vcf = hg38_vcf.sort_values(by=['chr', 'pos'])\n",
    "hg38_vcf.drop_duplicates(inplace=True)\n",
    "hg38_vcf.to_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/snps/hg38_snps_1KG_ADPD.vcf', header=False, index=False, sep='\\t')\n",
    "display(hg38_vcf.head())\n",
    "display(hg38_vcf.shape)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Make 1KG ADPD hg38 BED"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>pos</th>\n",
       "      <th>rsid</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>info</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39822150</td>\n",
       "      <td>39822151</td>\n",
       "      <td>rs34640847</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.11881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39822244</td>\n",
       "      <td>39822245</td>\n",
       "      <td>rs36015266</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.11881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39822259</td>\n",
       "      <td>39822260</td>\n",
       "      <td>rs61779808</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>0.0984425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39822791</td>\n",
       "      <td>39822792</td>\n",
       "      <td>rs61779809</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.11881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39825224</td>\n",
       "      <td>39825225</td>\n",
       "      <td>rs72666941</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.0952476</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr     start       pos        rsid ref alt       info\n",
       "0  chr1  39822150  39822151  rs34640847   G   A    0.11881\n",
       "1  chr1  39822244  39822245  rs36015266   C   T    0.11881\n",
       "2  chr1  39822259  39822260  rs61779808   T   A  0.0984425\n",
       "3  chr1  39822791  39822792  rs61779809   C   T    0.11881\n",
       "4  chr1  39825224  39825225  rs72666941   C   A  0.0952476"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "(9386, 7)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "hg38_bed = merged[['chr', 'pos', 'rsid', 'ref', 'alt', 'info']]\n",
    "hg38_bed['start'] = hg38_bed['pos'] - 1\n",
    "hg38_bed = hg38_bed[['chr', 'start', 'pos', 'rsid', 'ref', 'alt', 'info']]\n",
    "hg38_bed['info'] = hg38_bed['info'].apply(lambda x : x.split(';AF=')[1].split(';')[0])\n",
    "hg38_bed.sort_values(by=['chr', 'pos'], inplace=True)\n",
    "hg38_bed.drop_duplicates(inplace=True)\n",
    "hg38_bed.to_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/snps/hg38_snps_1KG_ADPD.bed', header=False, index=False, sep='\\t')\n",
    "display(hg38_bed.head())\n",
    "display(hg38_bed.shape)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get ADPD Metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OldName</th>\n",
       "      <th>NewName</th>\n",
       "      <th>HarmonizedName</th>\n",
       "      <th>Bam</th>\n",
       "      <th>Contrast</th>\n",
       "      <th>Type</th>\n",
       "      <th>Group</th>\n",
       "      <th>Region</th>\n",
       "      <th>RegionMod</th>\n",
       "      <th>PatientID</th>\n",
       "      <th>TypeMod</th>\n",
       "      <th>Cohort</th>\n",
       "      <th>TissueCenter</th>\n",
       "      <th>TypeGroup</th>\n",
       "      <th>xxx.TechRep</th>\n",
       "      <th>Batch</th>\n",
       "      <th>xxx.Sample</th>\n",
       "      <th>xxx.Library</th>\n",
       "      <th>xxx.BioRep</th>\n",
       "      <th>xxx.TechRep.1</th>\n",
       "      <th>Pool</th>\n",
       "      <th>MAPT</th>\n",
       "      <th>xxx.race</th>\n",
       "      <th>Gender</th>\n",
       "      <th>xxx.expired_age</th>\n",
       "      <th>PMI</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>MutationType</th>\n",
       "      <th>xxx.last_mmse_test_score</th>\n",
       "      <th>ApoE</th>\n",
       "      <th>xxx.last_mmse_test_date</th>\n",
       "      <th>xxx.motor_updrs_score_on</th>\n",
       "      <th>xxx.motor_updrs_score_off</th>\n",
       "      <th>xxx.motor_updrs_score_months_prior_to_death</th>\n",
       "      <th>xxx.Control</th>\n",
       "      <th>xxx.AD</th>\n",
       "      <th>xxx.PD</th>\n",
       "      <th>xxx.DLB</th>\n",
       "      <th>xxx.VAD</th>\n",
       "      <th>xxx.PSP</th>\n",
       "      <th>xxx.HS</th>\n",
       "      <th>xxx.DLDH</th>\n",
       "      <th>xxx.MND</th>\n",
       "      <th>xxx.CBD</th>\n",
       "      <th>xxx.PICKS</th>\n",
       "      <th>xxx.HD</th>\n",
       "      <th>xxx.MSA</th>\n",
       "      <th>xxx.ARG</th>\n",
       "      <th>xxx.CWMR</th>\n",
       "      <th>xxx.dementia_nos</th>\n",
       "      <th>xxx.MS</th>\n",
       "      <th>xxx.CAA</th>\n",
       "      <th>xxx.MCI</th>\n",
       "      <th>xxx.LBS</th>\n",
       "      <th>xxx.BRAIN_ABNORMAL</th>\n",
       "      <th>xxx.ACUTE_INFARCTS</th>\n",
       "      <th>xxx.PlaqueF</th>\n",
       "      <th>xxx.PlaqueT</th>\n",
       "      <th>xxx.PlaqueP</th>\n",
       "      <th>xxx.PlaqueH</th>\n",
       "      <th>xxx.PlaqueE</th>\n",
       "      <th>xxx.PlaqueTotal</th>\n",
       "      <th>xxx.Plaque density</th>\n",
       "      <th>xxx.TangleF</th>\n",
       "      <th>xxx.TangleT</th>\n",
       "      <th>xxx.TangleP</th>\n",
       "      <th>xxx.TangleE</th>\n",
       "      <th>xxx.TangleH</th>\n",
       "      <th>xxx.TangleTotal</th>\n",
       "      <th>xxx.Cerad NP</th>\n",
       "      <th>xxx.Braak score</th>\n",
       "      <th>xxx.NIA-R</th>\n",
       "      <th>xxx.Unified LB Stage</th>\n",
       "      <th>xxx.infarct_total_volume</th>\n",
       "      <th>xxx.infarct_cerebral_total_volume</th>\n",
       "      <th>xxx.obt</th>\n",
       "      <th>xxx.brain_stem_ix_x</th>\n",
       "      <th>xxx.brain_stem_lc</th>\n",
       "      <th>xxx.bf_amygdala</th>\n",
       "      <th>xxx.bf_nbm</th>\n",
       "      <th>xxx.brain_stem_sn</th>\n",
       "      <th>xxx.bf_trans</th>\n",
       "      <th>xxx.bf_cing</th>\n",
       "      <th>xxx.nctx_temporal</th>\n",
       "      <th>xxx.nctx_frontal</th>\n",
       "      <th>xxx.sum_lb_density</th>\n",
       "      <th>xxx.nctx_parietal</th>\n",
       "      <th>xxx.submondibular_gland</th>\n",
       "      <th>xxx.PathDXSummary</th>\n",
       "      <th>xxx.calc_NIA_AA</th>\n",
       "      <th>xxx.calc_thalPhase</th>\n",
       "      <th>xxx.micro_lewyBodyEvidenceTEXT</th>\n",
       "      <th>xxx.calc_HAAS_cerebralMicroinfarcts</th>\n",
       "      <th>xxx.calc_HAAS_deepGrayMicroinfarcts</th>\n",
       "      <th>xxx.AP_freshBrainWeight</th>\n",
       "      <th>xxx.CONS_CL_DX</th>\n",
       "      <th>xxx.CONS_CL_DX_UNK_OTHER</th>\n",
       "      <th>xxx.PRIMARY_NP_DX</th>\n",
       "      <th>xxx.CONTRIBUTING_NP_DX</th>\n",
       "      <th>xxx.Interval between death and last MMSE</th>\n",
       "      <th>xxx.ch_lastCasiScore</th>\n",
       "      <th>xxx.ch_lastCasiDate</th>\n",
       "      <th>xxx.micro_AmyloidAngiopathyOccipitalLobe_ID</th>\n",
       "      <th>xxx.GE_atherosclerosis_ID</th>\n",
       "      <th>xxx.calc_A</th>\n",
       "      <th>xxx.calc_B</th>\n",
       "      <th>xxx.C</th>\n",
       "      <th>xxx.CognitiveStatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PD_00_38_CTRL_CAUD_X014_S01_L001_B1_T1_P025</td>\n",
       "      <td>PD_00_38_CTRL_CAUD_X014_S01_L001_B1_T1_P025</td>\n",
       "      <td>CTRL_CAUD_PD_00x38xx_X014_S01_L001_B1_T1_P025</td>\n",
       "      <td>/oak/stanford/groups/howchang/users/mcorces/PD...</td>\n",
       "      <td>CTRL_CAUD</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>CAUD</td>\n",
       "      <td>CAUD</td>\n",
       "      <td>00_38</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>PD</td>\n",
       "      <td>UA</td>\n",
       "      <td>CTRL_CAUD</td>\n",
       "      <td>PD_00_38_CTRL_CAUD</td>\n",
       "      <td>PD_X014</td>\n",
       "      <td>S01</td>\n",
       "      <td>L001</td>\n",
       "      <td>B1</td>\n",
       "      <td>T1</td>\n",
       "      <td>PD_P025</td>\n",
       "      <td>H2</td>\n",
       "      <td>White</td>\n",
       "      <td>Female</td>\n",
       "      <td>91</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2_3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>zero</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.25</td>\n",
       "      <td>0.5</td>\n",
       "      <td>1.75</td>\n",
       "      <td>not AD</td>\n",
       "      <td>II</td>\n",
       "      <td>criteria not met</td>\n",
       "      <td>0. No Lewy bodies</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diseased Control; Left hippocampal sclerosis; ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PD_00_38_CTRL_CAUD_X014_S01_L002_B1_T2_P028</td>\n",
       "      <td>PD_00_38_CTRL_CAUD_X014_S01_L002_B1_T2_P028</td>\n",
       "      <td>CTRL_CAUD_PD_00x38xx_X014_S01_L002_B1_T2_P028</td>\n",
       "      <td>/oak/stanford/groups/howchang/users/mcorces/PD...</td>\n",
       "      <td>CTRL_CAUD</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>CAUD</td>\n",
       "      <td>CAUD</td>\n",
       "      <td>00_38</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>PD</td>\n",
       "      <td>UA</td>\n",
       "      <td>CTRL_CAUD</td>\n",
       "      <td>PD_00_38_CTRL_CAUD</td>\n",
       "      <td>PD_X014</td>\n",
       "      <td>S01</td>\n",
       "      <td>L002</td>\n",
       "      <td>B1</td>\n",
       "      <td>T2</td>\n",
       "      <td>PD_P028</td>\n",
       "      <td>H2</td>\n",
       "      <td>White</td>\n",
       "      <td>Female</td>\n",
       "      <td>91</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2_3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>zero</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.25</td>\n",
       "      <td>0.5</td>\n",
       "      <td>1.75</td>\n",
       "      <td>not AD</td>\n",
       "      <td>II</td>\n",
       "      <td>criteria not met</td>\n",
       "      <td>0. No Lewy bodies</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diseased Control; Left hippocampal sclerosis; ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PD_00_38_CTRL_HIPP_X002_S11_L045_B1_T1_P002</td>\n",
       "      <td>PD_00_38_CTRL_HIPP_X002_S11_L045_B1_T1_P002</td>\n",
       "      <td>CTRL_HIPP_PD_00x38xx_X002_S11_L045_B1_T1_P002</td>\n",
       "      <td>/oak/stanford/groups/howchang/users/mcorces/PD...</td>\n",
       "      <td>CTRL_HIPP</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>HIPP</td>\n",
       "      <td>HIPP</td>\n",
       "      <td>00_38</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>PD</td>\n",
       "      <td>UA</td>\n",
       "      <td>CTRL_HIPP</td>\n",
       "      <td>PD_00_38_CTRL_HIPP</td>\n",
       "      <td>PD_X002</td>\n",
       "      <td>S11</td>\n",
       "      <td>L045</td>\n",
       "      <td>B1</td>\n",
       "      <td>T1</td>\n",
       "      <td>PD_P002</td>\n",
       "      <td>H2</td>\n",
       "      <td>White</td>\n",
       "      <td>Female</td>\n",
       "      <td>91</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2_3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>zero</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.25</td>\n",
       "      <td>0.5</td>\n",
       "      <td>1.75</td>\n",
       "      <td>not AD</td>\n",
       "      <td>II</td>\n",
       "      <td>criteria not met</td>\n",
       "      <td>0. No Lewy bodies</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diseased Control; Left hippocampal sclerosis; ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PD_00_38_CTRL_HIPP_X002_S11_L046_B1_T2_P003</td>\n",
       "      <td>PD_00_38_CTRL_HIPP_X002_S11_L046_B1_T2_P003</td>\n",
       "      <td>CTRL_HIPP_PD_00x38xx_X002_S11_L046_B1_T2_P003</td>\n",
       "      <td>/oak/stanford/groups/howchang/users/mcorces/PD...</td>\n",
       "      <td>CTRL_HIPP</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>HIPP</td>\n",
       "      <td>HIPP</td>\n",
       "      <td>00_38</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>PD</td>\n",
       "      <td>UA</td>\n",
       "      <td>CTRL_HIPP</td>\n",
       "      <td>PD_00_38_CTRL_HIPP</td>\n",
       "      <td>PD_X002</td>\n",
       "      <td>S11</td>\n",
       "      <td>L046</td>\n",
       "      <td>B1</td>\n",
       "      <td>T2</td>\n",
       "      <td>PD_P003</td>\n",
       "      <td>H2</td>\n",
       "      <td>White</td>\n",
       "      <td>Female</td>\n",
       "      <td>91</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2_3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>zero</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.25</td>\n",
       "      <td>0.5</td>\n",
       "      <td>1.75</td>\n",
       "      <td>not AD</td>\n",
       "      <td>II</td>\n",
       "      <td>criteria not met</td>\n",
       "      <td>0. No Lewy bodies</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diseased Control; Left hippocampal sclerosis; ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PD_00_38_CTRL_MDFG_X007_S04_L055_B1_T1_P014</td>\n",
       "      <td>PD_00_38_CTRL_MDFG_X007_S04_L055_B1_T1_P014</td>\n",
       "      <td>CTRL_MDFG_PD_00x38xx_X007_S04_L055_B1_T1_P014</td>\n",
       "      <td>/oak/stanford/groups/howchang/users/mcorces/PD...</td>\n",
       "      <td>CTRL_MDFG</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>MDFG</td>\n",
       "      <td>MDFG</td>\n",
       "      <td>00_38</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>PD</td>\n",
       "      <td>UA</td>\n",
       "      <td>CTRL_MDFG</td>\n",
       "      <td>PD_00_38_CTRL_MDFG</td>\n",
       "      <td>PD_X007</td>\n",
       "      <td>S04</td>\n",
       "      <td>L055</td>\n",
       "      <td>B1</td>\n",
       "      <td>T1</td>\n",
       "      <td>PD_P014</td>\n",
       "      <td>H2</td>\n",
       "      <td>White</td>\n",
       "      <td>Female</td>\n",
       "      <td>91</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2_3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>zero</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.25</td>\n",
       "      <td>0.5</td>\n",
       "      <td>1.75</td>\n",
       "      <td>not AD</td>\n",
       "      <td>II</td>\n",
       "      <td>criteria not met</td>\n",
       "      <td>0. No Lewy bodies</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diseased Control; Left hippocampal sclerosis; ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       OldName  \\\n",
       "0  PD_00_38_CTRL_CAUD_X014_S01_L001_B1_T1_P025   \n",
       "1  PD_00_38_CTRL_CAUD_X014_S01_L002_B1_T2_P028   \n",
       "2  PD_00_38_CTRL_HIPP_X002_S11_L045_B1_T1_P002   \n",
       "3  PD_00_38_CTRL_HIPP_X002_S11_L046_B1_T2_P003   \n",
       "4  PD_00_38_CTRL_MDFG_X007_S04_L055_B1_T1_P014   \n",
       "\n",
       "                                       NewName  \\\n",
       "0  PD_00_38_CTRL_CAUD_X014_S01_L001_B1_T1_P025   \n",
       "1  PD_00_38_CTRL_CAUD_X014_S01_L002_B1_T2_P028   \n",
       "2  PD_00_38_CTRL_HIPP_X002_S11_L045_B1_T1_P002   \n",
       "3  PD_00_38_CTRL_HIPP_X002_S11_L046_B1_T2_P003   \n",
       "4  PD_00_38_CTRL_MDFG_X007_S04_L055_B1_T1_P014   \n",
       "\n",
       "                                  HarmonizedName  \\\n",
       "0  CTRL_CAUD_PD_00x38xx_X014_S01_L001_B1_T1_P025   \n",
       "1  CTRL_CAUD_PD_00x38xx_X014_S01_L002_B1_T2_P028   \n",
       "2  CTRL_HIPP_PD_00x38xx_X002_S11_L045_B1_T1_P002   \n",
       "3  CTRL_HIPP_PD_00x38xx_X002_S11_L046_B1_T2_P003   \n",
       "4  CTRL_MDFG_PD_00x38xx_X007_S04_L055_B1_T1_P014   \n",
       "\n",
       "                                                 Bam   Contrast  Type Group  \\\n",
       "0  /oak/stanford/groups/howchang/users/mcorces/PD...  CTRL_CAUD  CTRL  CTRL   \n",
       "1  /oak/stanford/groups/howchang/users/mcorces/PD...  CTRL_CAUD  CTRL  CTRL   \n",
       "2  /oak/stanford/groups/howchang/users/mcorces/PD...  CTRL_HIPP  CTRL  CTRL   \n",
       "3  /oak/stanford/groups/howchang/users/mcorces/PD...  CTRL_HIPP  CTRL  CTRL   \n",
       "4  /oak/stanford/groups/howchang/users/mcorces/PD...  CTRL_MDFG  CTRL  CTRL   \n",
       "\n",
       "  Region RegionMod PatientID TypeMod Cohort TissueCenter  TypeGroup  \\\n",
       "0   CAUD      CAUD     00_38    CTRL     PD           UA  CTRL_CAUD   \n",
       "1   CAUD      CAUD     00_38    CTRL     PD           UA  CTRL_CAUD   \n",
       "2   HIPP      HIPP     00_38    CTRL     PD           UA  CTRL_HIPP   \n",
       "3   HIPP      HIPP     00_38    CTRL     PD           UA  CTRL_HIPP   \n",
       "4   MDFG      MDFG     00_38    CTRL     PD           UA  CTRL_MDFG   \n",
       "\n",
       "          xxx.TechRep    Batch xxx.Sample xxx.Library xxx.BioRep  \\\n",
       "0  PD_00_38_CTRL_CAUD  PD_X014        S01        L001         B1   \n",
       "1  PD_00_38_CTRL_CAUD  PD_X014        S01        L002         B1   \n",
       "2  PD_00_38_CTRL_HIPP  PD_X002        S11        L045         B1   \n",
       "3  PD_00_38_CTRL_HIPP  PD_X002        S11        L046         B1   \n",
       "4  PD_00_38_CTRL_MDFG  PD_X007        S04        L055         B1   \n",
       "\n",
       "  xxx.TechRep.1     Pool MAPT xxx.race  Gender  xxx.expired_age  PMI  \\\n",
       "0            T1  PD_P025   H2    White  Female               91  2.0   \n",
       "1            T2  PD_P028   H2    White  Female               91  2.0   \n",
       "2            T1  PD_P002   H2    White  Female               91  2.0   \n",
       "3            T2  PD_P003   H2    White  Female               91  2.0   \n",
       "4            T1  PD_P014   H2    White  Female               91  2.0   \n",
       "\n",
       "   Mutation  MutationType  xxx.last_mmse_test_score ApoE  \\\n",
       "0       NaN           NaN                       NaN  2_3   \n",
       "1       NaN           NaN                       NaN  2_3   \n",
       "2       NaN           NaN                       NaN  2_3   \n",
       "3       NaN           NaN                       NaN  2_3   \n",
       "4       NaN           NaN                       NaN  2_3   \n",
       "\n",
       "  xxx.last_mmse_test_date  xxx.motor_updrs_score_on  \\\n",
       "0                     NaN                       0.0   \n",
       "1                     NaN                       0.0   \n",
       "2                     NaN                       0.0   \n",
       "3                     NaN                       0.0   \n",
       "4                     NaN                       0.0   \n",
       "\n",
       "   xxx.motor_updrs_score_off  xxx.motor_updrs_score_months_prior_to_death  \\\n",
       "0                       16.0                                         19.0   \n",
       "1                       16.0                                         19.0   \n",
       "2                       16.0                                         19.0   \n",
       "3                       16.0                                         19.0   \n",
       "4                       16.0                                         19.0   \n",
       "\n",
       "  xxx.Control xxx.AD xxx.PD xxx.DLB xxx.VAD xxx.PSP xxx.HS xxx.DLDH xxx.MND  \\\n",
       "0         yes     no     no      no      no      no    yes       no      no   \n",
       "1         yes     no     no      no      no      no    yes       no      no   \n",
       "2         yes     no     no      no      no      no    yes       no      no   \n",
       "3         yes     no     no      no      no      no    yes       no      no   \n",
       "4         yes     no     no      no      no      no    yes       no      no   \n",
       "\n",
       "  xxx.CBD xxx.PICKS xxx.HD xxx.MSA xxx.ARG xxx.CWMR xxx.dementia_nos xxx.MS  \\\n",
       "0      no        no     no      no      no       no               no     no   \n",
       "1      no        no     no      no      no       no               no     no   \n",
       "2      no        no     no      no      no       no               no     no   \n",
       "3      no        no     no      no      no       no               no     no   \n",
       "4      no        no     no      no      no       no               no     no   \n",
       "\n",
       "  xxx.CAA xxx.MCI xxx.LBS xxx.BRAIN_ABNORMAL xxx.ACUTE_INFARCTS  xxx.PlaqueF  \\\n",
       "0      no      no      no                yes                 no          0.0   \n",
       "1      no      no      no                yes                 no          0.0   \n",
       "2      no      no      no                yes                 no          0.0   \n",
       "3      no      no      no                yes                 no          0.0   \n",
       "4      no      no      no                yes                 no          0.0   \n",
       "\n",
       "   xxx.PlaqueT  xxx.PlaqueP  xxx.PlaqueH  xxx.PlaqueE  xxx.PlaqueTotal  \\\n",
       "0          0.0          0.0          0.0          0.0              0.0   \n",
       "1          0.0          0.0          0.0          0.0              0.0   \n",
       "2          0.0          0.0          0.0          0.0              0.0   \n",
       "3          0.0          0.0          0.0          0.0              0.0   \n",
       "4          0.0          0.0          0.0          0.0              0.0   \n",
       "\n",
       "  xxx.Plaque density  xxx.TangleF  xxx.TangleT  xxx.TangleP  xxx.TangleE  \\\n",
       "0               zero          0.0          0.0          0.0         1.25   \n",
       "1               zero          0.0          0.0          0.0         1.25   \n",
       "2               zero          0.0          0.0          0.0         1.25   \n",
       "3               zero          0.0          0.0          0.0         1.25   \n",
       "4               zero          0.0          0.0          0.0         1.25   \n",
       "\n",
       "   xxx.TangleH  xxx.TangleTotal xxx.Cerad NP xxx.Braak score  \\\n",
       "0          0.5             1.75       not AD              II   \n",
       "1          0.5             1.75       not AD              II   \n",
       "2          0.5             1.75       not AD              II   \n",
       "3          0.5             1.75       not AD              II   \n",
       "4          0.5             1.75       not AD              II   \n",
       "\n",
       "          xxx.NIA-R xxx.Unified LB Stage  xxx.infarct_total_volume  \\\n",
       "0  criteria not met    0. No Lewy bodies                       0.0   \n",
       "1  criteria not met    0. No Lewy bodies                       0.0   \n",
       "2  criteria not met    0. No Lewy bodies                       0.0   \n",
       "3  criteria not met    0. No Lewy bodies                       0.0   \n",
       "4  criteria not met    0. No Lewy bodies                       0.0   \n",
       "\n",
       "   xxx.infarct_cerebral_total_volume  xxx.obt  xxx.brain_stem_ix_x  \\\n",
       "0                                0.0      0.0                  0.0   \n",
       "1                                0.0      0.0                  0.0   \n",
       "2                                0.0      0.0                  0.0   \n",
       "3                                0.0      0.0                  0.0   \n",
       "4                                0.0      0.0                  0.0   \n",
       "\n",
       "   xxx.brain_stem_lc  xxx.bf_amygdala  xxx.bf_nbm  xxx.brain_stem_sn  \\\n",
       "0                0.0              0.0         0.0                0.0   \n",
       "1                0.0              0.0         0.0                0.0   \n",
       "2                0.0              0.0         0.0                0.0   \n",
       "3                0.0              0.0         0.0                0.0   \n",
       "4                0.0              0.0         0.0                0.0   \n",
       "\n",
       "   xxx.bf_trans  xxx.bf_cing  xxx.nctx_temporal  xxx.nctx_frontal  \\\n",
       "0           0.0          0.0                0.0               0.0   \n",
       "1           0.0          0.0                0.0               0.0   \n",
       "2           0.0          0.0                0.0               0.0   \n",
       "3           0.0          0.0                0.0               0.0   \n",
       "4           0.0          0.0                0.0               0.0   \n",
       "\n",
       "   xxx.sum_lb_density  xxx.nctx_parietal  xxx.submondibular_gland  \\\n",
       "0                 0.0                0.0                      NaN   \n",
       "1                 0.0                0.0                      NaN   \n",
       "2                 0.0                0.0                      NaN   \n",
       "3                 0.0                0.0                      NaN   \n",
       "4                 0.0                0.0                      NaN   \n",
       "\n",
       "                                   xxx.PathDXSummary xxx.calc_NIA_AA  \\\n",
       "0  Diseased Control; Left hippocampal sclerosis; ...             NaN   \n",
       "1  Diseased Control; Left hippocampal sclerosis; ...             NaN   \n",
       "2  Diseased Control; Left hippocampal sclerosis; ...             NaN   \n",
       "3  Diseased Control; Left hippocampal sclerosis; ...             NaN   \n",
       "4  Diseased Control; Left hippocampal sclerosis; ...             NaN   \n",
       "\n",
       "  xxx.calc_thalPhase xxx.micro_lewyBodyEvidenceTEXT  \\\n",
       "0                NaN                            NaN   \n",
       "1                NaN                            NaN   \n",
       "2                NaN                            NaN   \n",
       "3                NaN                            NaN   \n",
       "4                NaN                            NaN   \n",
       "\n",
       "   xxx.calc_HAAS_cerebralMicroinfarcts  xxx.calc_HAAS_deepGrayMicroinfarcts  \\\n",
       "0                                  NaN                                  NaN   \n",
       "1                                  NaN                                  NaN   \n",
       "2                                  NaN                                  NaN   \n",
       "3                                  NaN                                  NaN   \n",
       "4                                  NaN                                  NaN   \n",
       "\n",
       "   xxx.AP_freshBrainWeight xxx.CONS_CL_DX xxx.CONS_CL_DX_UNK_OTHER  \\\n",
       "0                      NaN            NaN                      NaN   \n",
       "1                      NaN            NaN                      NaN   \n",
       "2                      NaN            NaN                      NaN   \n",
       "3                      NaN            NaN                      NaN   \n",
       "4                      NaN            NaN                      NaN   \n",
       "\n",
       "  xxx.PRIMARY_NP_DX xxx.CONTRIBUTING_NP_DX  \\\n",
       "0               NaN                    NaN   \n",
       "1               NaN                    NaN   \n",
       "2               NaN                    NaN   \n",
       "3               NaN                    NaN   \n",
       "4               NaN                    NaN   \n",
       "\n",
       "   xxx.Interval between death and last MMSE  xxx.ch_lastCasiScore  \\\n",
       "0                                       NaN                   NaN   \n",
       "1                                       NaN                   NaN   \n",
       "2                                       NaN                   NaN   \n",
       "3                                       NaN                   NaN   \n",
       "4                                       NaN                   NaN   \n",
       "\n",
       "   xxx.ch_lastCasiDate  xxx.micro_AmyloidAngiopathyOccipitalLobe_ID  \\\n",
       "0                  NaN                                          NaN   \n",
       "1                  NaN                                          NaN   \n",
       "2                  NaN                                          NaN   \n",
       "3                  NaN                                          NaN   \n",
       "4                  NaN                                          NaN   \n",
       "\n",
       "   xxx.GE_atherosclerosis_ID  xxx.calc_A  xxx.calc_B  xxx.C  \\\n",
       "0                        NaN         NaN         NaN    NaN   \n",
       "1                        NaN         NaN         NaN    NaN   \n",
       "2                        NaN         NaN         NaN    NaN   \n",
       "3                        NaN         NaN         NaN    NaN   \n",
       "4                        NaN         NaN         NaN    NaN   \n",
       "\n",
       "  xxx.CognitiveStatus  \n",
       "0                 NaN  \n",
       "1                 NaN  \n",
       "2                 NaN  \n",
       "3                 NaN  \n",
       "4                 NaN  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "(268, 108)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "metadata = pd.read_excel('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/190215_Brain-ControlsOnly_Metadata_Merged.xlsx')\n",
    "display(metadata.head())\n",
    "display(metadata.shape)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get Bams for each brain region"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['CAUD', 'HIPP', 'MDFG', 'MDTG', 'PTMN', 'SUNI', 'SMTG', 'PARL'],\n",
       "      dtype=object)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "display(metadata['Region'].unique())\n",
    "regions = list(metadata['Region'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CAUD\n",
      "HIPP\n",
      "MDFG\n",
      "MDTG\n",
      "PTMN\n",
      "SUNI\n",
      "SMTG\n",
      "PARL\n"
     ]
    }
   ],
   "source": [
    "with open('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/ctrl_bams.txt') as infile:\n",
    "    ctrl_bams = infile.readlines()\n",
    "multi_bam_regs = []\n",
    "for reg in regions:\n",
    "    print(reg)\n",
    "    region_bams = []\n",
    "    for bam in ctrl_bams:\n",
    "        if reg in bam:\n",
    "            region_bams.append(bam)\n",
    "    if len(region_bams) > 0:\n",
    "        multi_bam_regs.append(reg)\n",
    "        region_bams.sort()\n",
    "        if not os.path.isdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' + reg):\n",
    "            os.mkdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' + reg)\n",
    "        if not os.path.isdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/' + reg):\n",
    "            os.mkdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/' + reg)\n",
    "        if not os.path.isdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/scripts/' + reg):\n",
    "            os.mkdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/scripts/' + reg)\n",
    "        if not os.path.isdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/logs/' + reg):\n",
    "            os.mkdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/logs/' + reg)\n",
    "        with open('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/bam_lists/' + reg + '_ctrl_bams.txt', 'w') as outfile:\n",
    "            for bam in region_bams:\n",
    "                outfile.write(bam)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Make Input VCF files for each region"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# rm_cmd = 'rm /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/scripts/run_asvcf.sh'\n",
    "# print(rm_cmd)\n",
    "# ! {rm_cmd}\n",
    "\n",
    "# for reg in multi_bam_regs:\n",
    "#     print(reg)\n",
    "#     with open('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/bam_lists/' + reg + '_ctrl_bams.txt') as infile:\n",
    "#         reg_bams = [i.strip() for i in infile.readlines()]\n",
    "#     reg_vcf = hg38_vcf.copy(deep=True)\n",
    "#     for bam in reg_bams:\n",
    "#         bam_name = bam.split('/')[10]\n",
    "#         print(bam_name)\n",
    "#         reg_vcf[bam_name] = ['./.' for i in range(len(reg_vcf))]\n",
    "#     reg_vcf.to_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' + reg + '/initial.vcf', header=False, index=False, sep='\\t')\n",
    "#     bgzip_cmd = 'bgzip /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' + reg + '/initial.vcf'\n",
    "#     ! {bgzip_cmd}\n",
    "#     tabix_cmd = 'tabix -p vcf /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' + reg + '/initial.vcf.gz'\n",
    "#     ! {tabix_cmd}\n",
    "#     asvcf_cmd_1 = 'echo sbatch --export=ALL -n 1 -t 1-0 -p akundaje --mail-type=ALL -J ' \\\n",
    "#                 + reg + ' -o /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/logs/' \\\n",
    "#                 + reg + '/asvcf.o -e /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/logs/' \\\n",
    "#                 + reg + '/asvcf.e /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/scripts/' \\\n",
    "#                 + reg + '/asvcf.sh >> /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/scripts/run_asvcf.sh'\n",
    "#     asvcf_cmd_2 = 'echo \\'#!/bin/bash\\' > /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/scripts/' \\\n",
    "#                 + reg + '/asvcf.sh'\n",
    "#     asvcf_cmd_3 = 'echo /home/users/soumyak/rasqual/src/ASVCF/createASVCF.sh paired_end ' \\\n",
    "#                 + '/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/bam_lists/' \\\n",
    "#                 + reg + '_ctrl_bams.txt /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' \\\n",
    "#                 + reg + '/initial.vcf.gz /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' \\\n",
    "#                 + reg + '/asvcf.gz atac >> /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/scripts/' \\\n",
    "#                 + reg + '/asvcf.sh'\n",
    "#     print(asvcf_cmd_1)\n",
    "#     print(asvcf_cmd_2)\n",
    "#     print(asvcf_cmd_3)\n",
    "#     ! {asvcf_cmd_1}\n",
    "#     ! {asvcf_cmd_2}\n",
    "#     ! {asvcf_cmd_3}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load IDR Counts Matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>ADAD_CAUD_00_0281</th>\n",
       "      <th>ADAD_CAUD_00_0387</th>\n",
       "      <th>ADAD_CAUD_01_0164</th>\n",
       "      <th>ADAD_CAUD_01_1400</th>\n",
       "      <th>ADAD_CAUD_06_0194</th>\n",
       "      <th>ADAD_CAUD_06_1486</th>\n",
       "      <th>ADAD_CAUD_07_0787</th>\n",
       "      <th>ADAD_CAUD_12829xx</th>\n",
       "      <th>ADAD_CAUD_23156xx</th>\n",
       "      <th>ADAD_CAUD_24281xx</th>\n",
       "      <th>ADAD_CAUD_26436xx</th>\n",
       "      <th>ADAD_CAUD_AWQ6JZx</th>\n",
       "      <th>ADAD_CAUD_C0T7AAx</th>\n",
       "      <th>ADAD_CAUD_DCQGVNx</th>\n",
       "      <th>ADAD_CAUD_SXIONRx</th>\n",
       "      <th>ADAD_CAUD_WYYIWGx</th>\n",
       "      <th>ADAD_CAUD_YUJW97x</th>\n",
       "      <th>ADAD_HIPP_12829xx</th>\n",
       "      <th>ADAD_PARL_00_0700</th>\n",
       "      <th>ADAD_PARL_02_0621</th>\n",
       "      <th>ADAD_PARL_06_1486</th>\n",
       "      <th>ADAD_PARL_12540xx</th>\n",
       "      <th>ADAD_PARL_24281xx</th>\n",
       "      <th>ADAD_PARL_25044xx</th>\n",
       "      <th>ADAD_PARL_26436xx</th>\n",
       "      <th>ADAD_PARL_DCQGVNx</th>\n",
       "      <th>ADAD_PARL_HQFTL0x</th>\n",
       "      <th>ADAD_PARL_SXIONRx</th>\n",
       "      <th>ADAD_PARL_WYYIWGx</th>\n",
       "      <th>ADAD_PARL_YUJW97x</th>\n",
       "      <th>ADAD_SMTG_00_0387</th>\n",
       "      <th>ADAD_SMTG_00_0700</th>\n",
       "      <th>ADAD_SMTG_02_0621</th>\n",
       "      <th>ADAD_SMTG_06_1486</th>\n",
       "      <th>ADAD_SMTG_07_0787</th>\n",
       "      <th>ADAD_SMTG_12540xx</th>\n",
       "      <th>ADAD_SMTG_12829xx</th>\n",
       "      <th>ADAD_SMTG_23156xx</th>\n",
       "      <th>ADAD_SMTG_23710xx</th>\n",
       "      <th>ADAD_SMTG_24281xx</th>\n",
       "      <th>ADAD_SMTG_26436xx</th>\n",
       "      <th>ADAD_SMTG_AWQ6JZx</th>\n",
       "      <th>ADAD_SMTG_C0T7AAx</th>\n",
       "      <th>ADAD_SMTG_DCQGVNx</th>\n",
       "      <th>ADAD_SMTG_SXIONRx</th>\n",
       "      <th>ADAD_SMTG_WYYIWGx</th>\n",
       "      <th>ADAD_SMTG_YUJW97x</th>\n",
       "      <th>CTRH_CAUD_07_1144</th>\n",
       "      <th>CTRH_CAUD_07_1287</th>\n",
       "      <th>CTRH_CAUD_08_0712</th>\n",
       "      <th>CTRH_CAUD_10_0867</th>\n",
       "      <th>CTRH_CAUD_13_0962</th>\n",
       "      <th>CTRH_CAUD_14_0380</th>\n",
       "      <th>CTRH_CAUD_14_0513</th>\n",
       "      <th>CTRH_CAUD_14_1383</th>\n",
       "      <th>CTRH_CAUD_15_1023</th>\n",
       "      <th>CTRH_CAUD_15_1025</th>\n",
       "      <th>CTRH_HIPP_07_1058</th>\n",
       "      <th>CTRH_HIPP_07_1144</th>\n",
       "      <th>CTRH_HIPP_07_1287</th>\n",
       "      <th>CTRH_HIPP_08_0298</th>\n",
       "      <th>CTRH_HIPP_10_0867</th>\n",
       "      <th>CTRH_HIPP_13_0962</th>\n",
       "      <th>CTRH_HIPP_14_0380</th>\n",
       "      <th>CTRH_HIPP_14_0513</th>\n",
       "      <th>CTRH_HIPP_14_1383</th>\n",
       "      <th>CTRH_HIPP_15_1023</th>\n",
       "      <th>CTRH_HIPP_15_1025</th>\n",
       "      <th>CTRH_PARL_07_1058</th>\n",
       "      <th>CTRH_PARL_07_1144</th>\n",
       "      <th>CTRH_PARL_07_1287</th>\n",
       "      <th>CTRH_PARL_08_0298</th>\n",
       "      <th>CTRH_PARL_10_0867</th>\n",
       "      <th>CTRH_PARL_13_0962</th>\n",
       "      <th>CTRH_PARL_14_0380</th>\n",
       "      <th>CTRH_PARL_14_0513</th>\n",
       "      <th>CTRH_PARL_14_1383</th>\n",
       "      <th>CTRH_PARL_15_1023</th>\n",
       "      <th>CTRH_PARL_15_1025</th>\n",
       "      <th>CTRH_SMTG_07_1058</th>\n",
       "      <th>CTRH_SMTG_07_1144</th>\n",
       "      <th>CTRH_SMTG_07_1287</th>\n",
       "      <th>CTRH_SMTG_08_0712</th>\n",
       "      <th>CTRH_SMTG_10_0867</th>\n",
       "      <th>CTRH_SMTG_13_0962</th>\n",
       "      <th>CTRH_SMTG_14_0380</th>\n",
       "      <th>CTRH_SMTG_14_0513</th>\n",
       "      <th>CTRH_SMTG_14_1383</th>\n",
       "      <th>CTRH_SMTG_15_1023</th>\n",
       "      <th>CTRH_SMTG_15_1025</th>\n",
       "      <th>CTRL_CAUD_00_38</th>\n",
       "      <th>CTRL_CAUD_01_31</th>\n",
       "      <th>CTRL_CAUD_03_15</th>\n",
       "      <th>CTRL_CAUD_03_39</th>\n",
       "      <th>CTRL_CAUD_03_41</th>\n",
       "      <th>CTRL_CAUD_03_66</th>\n",
       "      <th>CTRL_CAUD_04_38</th>\n",
       "      <th>CTRL_CAUD_05_16</th>\n",
       "      <th>CTRL_CAUD_06_0615</th>\n",
       "      <th>CTRL_CAUD_06_1516</th>\n",
       "      <th>CTRL_CAUD_08_90</th>\n",
       "      <th>CTRL_CAUD_09_1589</th>\n",
       "      <th>CTRL_CAUD_09_35</th>\n",
       "      <th>CTRL_CAUD_11_0393</th>\n",
       "      <th>CTRL_CAUD_13_0038</th>\n",
       "      <th>CTRL_CAUD_13_1226</th>\n",
       "      <th>CTRL_CAUD_14_0586</th>\n",
       "      <th>CTRL_CAUD_14_0941</th>\n",
       "      <th>CTRL_CAUD_14_1018</th>\n",
       "      <th>CTRL_CAUD_15_78</th>\n",
       "      <th>CTRL_CAUD_16_10</th>\n",
       "      <th>CTRL_CAUD_16_32</th>\n",
       "      <th>CTRL_HIPP_00_38</th>\n",
       "      <th>CTRL_HIPP_04_38</th>\n",
       "      <th>CTRL_HIPP_05_16</th>\n",
       "      <th>CTRL_HIPP_06_1516</th>\n",
       "      <th>CTRL_HIPP_09_35</th>\n",
       "      <th>CTRL_HIPP_11_0393</th>\n",
       "      <th>CTRL_HIPP_13_0038</th>\n",
       "      <th>CTRL_HIPP_13_1226</th>\n",
       "      <th>CTRL_HIPP_14_0586</th>\n",
       "      <th>CTRL_HIPP_14_0941</th>\n",
       "      <th>CTRL_HIPP_14_1018</th>\n",
       "      <th>CTRL_MDFG_00_38</th>\n",
       "      <th>CTRL_MDFG_01_37</th>\n",
       "      <th>CTRL_MDFG_03_39</th>\n",
       "      <th>CTRL_MDFG_03_41</th>\n",
       "      <th>CTRL_MDFG_03_66</th>\n",
       "      <th>CTRL_MDFG_04_38</th>\n",
       "      <th>CTRL_MDFG_05_16</th>\n",
       "      <th>CTRL_MDFG_09_35</th>\n",
       "      <th>CTRL_MDFG_09_50</th>\n",
       "      <th>CTRL_MDFG_15_78</th>\n",
       "      <th>CTRL_MDFG_16_10</th>\n",
       "      <th>CTRL_MDFG_16_32</th>\n",
       "      <th>CTRL_MDTG_00_38</th>\n",
       "      <th>CTRL_MDTG_03_15</th>\n",
       "      <th>CTRL_MDTG_03_39</th>\n",
       "      <th>CTRL_MDTG_03_41</th>\n",
       "      <th>CTRL_MDTG_03_66</th>\n",
       "      <th>CTRL_MDTG_04_38</th>\n",
       "      <th>CTRL_MDTG_05_16</th>\n",
       "      <th>CTRL_MDTG_09_35</th>\n",
       "      <th>CTRL_MDTG_09_50</th>\n",
       "      <th>CTRL_MDTG_15_78</th>\n",
       "      <th>CTRL_MDTG_16_10</th>\n",
       "      <th>CTRL_MDTG_16_32</th>\n",
       "      <th>CTRL_PARL_09_1589</th>\n",
       "      <th>CTRL_PARL_11_0311</th>\n",
       "      <th>CTRL_PARL_11_0393</th>\n",
       "      <th>CTRL_PARL_13_0038</th>\n",
       "      <th>CTRL_PARL_13_1226</th>\n",
       "      <th>CTRL_PARL_14_0586</th>\n",
       "      <th>CTRL_PARL_14_1018</th>\n",
       "      <th>CTRL_PTMN_00_38</th>\n",
       "      <th>CTRL_PTMN_03_39</th>\n",
       "      <th>CTRL_PTMN_03_41</th>\n",
       "      <th>CTRL_PTMN_03_66</th>\n",
       "      <th>CTRL_PTMN_04_38</th>\n",
       "      <th>CTRL_PTMN_05_16</th>\n",
       "      <th>CTRL_PTMN_07_28</th>\n",
       "      <th>CTRL_PTMN_09_35</th>\n",
       "      <th>CTRL_PTMN_09_50</th>\n",
       "      <th>CTRL_PTMN_15_78</th>\n",
       "      <th>CTRL_PTMN_16_32</th>\n",
       "      <th>CTRL_SMTG_06_1516</th>\n",
       "      <th>CTRL_SMTG_09_1589</th>\n",
       "      <th>CTRL_SMTG_11_0311</th>\n",
       "      <th>CTRL_SMTG_11_0393</th>\n",
       "      <th>CTRL_SMTG_13_0038</th>\n",
       "      <th>CTRL_SMTG_13_1226</th>\n",
       "      <th>CTRL_SMTG_14_0586</th>\n",
       "      <th>CTRL_SMTG_14_1018</th>\n",
       "      <th>CTRL_SUNI_00_38</th>\n",
       "      <th>CTRL_SUNI_01_31</th>\n",
       "      <th>CTRL_SUNI_03_15</th>\n",
       "      <th>CTRL_SUNI_03_39</th>\n",
       "      <th>CTRL_SUNI_03_41</th>\n",
       "      <th>CTRL_SUNI_03_66</th>\n",
       "      <th>CTRL_SUNI_04_38</th>\n",
       "      <th>CTRL_SUNI_05_16</th>\n",
       "      <th>CTRL_SUNI_06_57</th>\n",
       "      <th>CTRL_SUNI_07_28</th>\n",
       "      <th>CTRL_SUNI_09_35</th>\n",
       "      <th>CTRL_SUNI_09_50</th>\n",
       "      <th>CTRL_SUNI_15_78</th>\n",
       "      <th>CTRL_SUNI_16_32</th>\n",
       "      <th>GBA1_CAUD_00_26</th>\n",
       "      <th>GBA1_CAUD_00_27</th>\n",
       "      <th>GBA1_CAUD_00_45</th>\n",
       "      <th>GBA1_CAUD_03_43</th>\n",
       "      <th>GBA1_CAUD_03_51</th>\n",
       "      <th>GBA1_CAUD_04_37</th>\n",
       "      <th>GBA1_CAUD_07_41</th>\n",
       "      <th>GBA1_CAUD_11_90</th>\n",
       "      <th>GBA1_CAUD_12_65</th>\n",
       "      <th>GBA1_CAUD_12_69</th>\n",
       "      <th>GBA1_CAUD_93_19</th>\n",
       "      <th>GBA1_CAUD_94_31</th>\n",
       "      <th>GBA1_CAUD_97_52</th>\n",
       "      <th>GBA1_HIPP_03_43</th>\n",
       "      <th>GBA1_HIPP_11_90</th>\n",
       "      <th>GBA1_HIPP_12_65</th>\n",
       "      <th>GBA1_HIPP_93_19</th>\n",
       "      <th>GBA1_HIPP_94_31</th>\n",
       "      <th>GBA1_HIPP_97_52</th>\n",
       "      <th>GBA1_MDFG_00_26</th>\n",
       "      <th>GBA1_MDFG_00_27</th>\n",
       "      <th>GBA1_MDFG_00_45</th>\n",
       "      <th>GBA1_MDFG_03_43</th>\n",
       "      <th>GBA1_MDFG_03_51</th>\n",
       "      <th>GBA1_MDFG_04_37</th>\n",
       "      <th>GBA1_MDFG_07_41</th>\n",
       "      <th>GBA1_MDFG_11_90</th>\n",
       "      <th>GBA1_MDFG_12_65</th>\n",
       "      <th>GBA1_MDFG_12_69</th>\n",
       "      <th>GBA1_MDFG_13_22</th>\n",
       "      <th>GBA1_MDFG_93_19</th>\n",
       "      <th>GBA1_MDFG_94_31</th>\n",
       "      <th>GBA1_MDFG_97_52</th>\n",
       "      <th>GBA1_MDTG_00_09</th>\n",
       "      <th>GBA1_MDTG_00_26</th>\n",
       "      <th>GBA1_MDTG_00_27</th>\n",
       "      <th>GBA1_MDTG_00_45</th>\n",
       "      <th>GBA1_MDTG_03_43</th>\n",
       "      <th>GBA1_MDTG_03_51</th>\n",
       "      <th>GBA1_MDTG_04_37</th>\n",
       "      <th>GBA1_MDTG_11_90</th>\n",
       "      <th>GBA1_MDTG_12_50</th>\n",
       "      <th>GBA1_MDTG_12_65</th>\n",
       "      <th>GBA1_MDTG_12_69</th>\n",
       "      <th>GBA1_MDTG_13_22</th>\n",
       "      <th>GBA1_MDTG_93_19</th>\n",
       "      <th>GBA1_MDTG_94_31</th>\n",
       "      <th>GBA1_MDTG_97_52</th>\n",
       "      <th>GBA1_PTMN_00_09</th>\n",
       "      <th>GBA1_PTMN_00_26</th>\n",
       "      <th>GBA1_PTMN_00_45</th>\n",
       "      <th>GBA1_PTMN_03_43</th>\n",
       "      <th>GBA1_PTMN_03_51</th>\n",
       "      <th>GBA1_PTMN_04_37</th>\n",
       "      <th>GBA1_PTMN_07_41</th>\n",
       "      <th>GBA1_PTMN_11_90</th>\n",
       "      <th>GBA1_PTMN_12_65</th>\n",
       "      <th>GBA1_PTMN_12_69</th>\n",
       "      <th>GBA1_PTMN_13_22</th>\n",
       "      <th>GBA1_PTMN_93_19</th>\n",
       "      <th>GBA1_PTMN_94_31</th>\n",
       "      <th>GBA1_PTMN_97_52</th>\n",
       "      <th>GBA1_SUNI_00_26</th>\n",
       "      <th>GBA1_SUNI_00_27</th>\n",
       "      <th>GBA1_SUNI_00_45</th>\n",
       "      <th>GBA1_SUNI_03_43</th>\n",
       "      <th>GBA1_SUNI_03_51</th>\n",
       "      <th>GBA1_SUNI_04_37</th>\n",
       "      <th>GBA1_SUNI_07_41</th>\n",
       "      <th>GBA1_SUNI_11_90</th>\n",
       "      <th>GBA1_SUNI_12_50</th>\n",
       "      <th>GBA1_SUNI_12_65</th>\n",
       "      <th>GBA1_SUNI_13_22</th>\n",
       "      <th>GBA1_SUNI_94_31</th>\n",
       "      <th>GBA1_SUNI_97_52</th>\n",
       "      <th>LOAD_CAUD_07_0226</th>\n",
       "      <th>LOAD_CAUD_07_0664</th>\n",
       "      <th>LOAD_CAUD_07_1112</th>\n",
       "      <th>LOAD_CAUD_09_1166</th>\n",
       "      <th>LOAD_CAUD_10_0188</th>\n",
       "      <th>LOAD_CAUD_10_1363</th>\n",
       "      <th>LOAD_CAUD_10_1770</th>\n",
       "      <th>LOAD_CAUD_10_1930</th>\n",
       "      <th>LOAD_CAUD_10_2007</th>\n",
       "      <th>LOAD_CAUD_11_0352</th>\n",
       "      <th>LOAD_CAUD_11_0773</th>\n",
       "      <th>LOAD_CAUD_11_1475</th>\n",
       "      <th>LOAD_CAUD_11_1686</th>\n",
       "      <th>LOAD_CAUD_13_0855</th>\n",
       "      <th>LOAD_CAUD_13_1300</th>\n",
       "      <th>LOAD_HIPP_06_0308</th>\n",
       "      <th>LOAD_HIPP_07_0226</th>\n",
       "      <th>LOAD_HIPP_07_0664</th>\n",
       "      <th>LOAD_HIPP_07_0997</th>\n",
       "      <th>LOAD_HIPP_07_1112</th>\n",
       "      <th>LOAD_HIPP_09_1166</th>\n",
       "      <th>LOAD_HIPP_10_0188</th>\n",
       "      <th>LOAD_HIPP_10_1363</th>\n",
       "      <th>LOAD_HIPP_10_1770</th>\n",
       "      <th>LOAD_HIPP_11_0352</th>\n",
       "      <th>LOAD_HIPP_11_0773</th>\n",
       "      <th>LOAD_HIPP_11_1475</th>\n",
       "      <th>LOAD_HIPP_12_1181</th>\n",
       "      <th>LOAD_HIPP_13_0855</th>\n",
       "      <th>LOAD_HIPP_13_1300</th>\n",
       "      <th>LOAD_PARL_06_0308</th>\n",
       "      <th>LOAD_PARL_07_0226</th>\n",
       "      <th>LOAD_PARL_07_0664</th>\n",
       "      <th>LOAD_PARL_07_0997</th>\n",
       "      <th>LOAD_PARL_07_1112</th>\n",
       "      <th>LOAD_PARL_08_0789</th>\n",
       "      <th>LOAD_PARL_09_1166</th>\n",
       "      <th>LOAD_PARL_10_0188</th>\n",
       "      <th>LOAD_PARL_10_0213</th>\n",
       "      <th>LOAD_PARL_10_1363</th>\n",
       "      <th>LOAD_PARL_10_1770</th>\n",
       "      <th>LOAD_PARL_10_2007</th>\n",
       "      <th>LOAD_PARL_11_0352</th>\n",
       "      <th>LOAD_PARL_11_1686</th>\n",
       "      <th>LOAD_PARL_12_1181</th>\n",
       "      <th>LOAD_PARL_13_0855</th>\n",
       "      <th>LOAD_PARL_13_1300</th>\n",
       "      <th>LOAD_SMTG_06_0308</th>\n",
       "      <th>LOAD_SMTG_07_0226</th>\n",
       "      <th>LOAD_SMTG_07_1112</th>\n",
       "      <th>LOAD_SMTG_09_1166</th>\n",
       "      <th>LOAD_SMTG_10_0188</th>\n",
       "      <th>LOAD_SMTG_10_0213</th>\n",
       "      <th>LOAD_SMTG_10_1770</th>\n",
       "      <th>LOAD_SMTG_10_1930</th>\n",
       "      <th>LOAD_SMTG_10_2007</th>\n",
       "      <th>LOAD_SMTG_11_0352</th>\n",
       "      <th>LOAD_SMTG_11_1475</th>\n",
       "      <th>LOAD_SMTG_12_1181</th>\n",
       "      <th>LOAD_SMTG_13_0855</th>\n",
       "      <th>LOAD_SMTG_13_1300</th>\n",
       "      <th>LOPD_CAUD_10_28</th>\n",
       "      <th>LOPD_CAUD_10_30</th>\n",
       "      <th>LOPD_CAUD_10_76</th>\n",
       "      <th>LOPD_CAUD_10_83</th>\n",
       "      <th>LOPD_CAUD_11_70</th>\n",
       "      <th>LOPD_CAUD_12_04</th>\n",
       "      <th>LOPD_CAUD_12_22</th>\n",
       "      <th>LOPD_CAUD_12_29</th>\n",
       "      <th>LOPD_CAUD_12_42</th>\n",
       "      <th>LOPD_CAUD_13_05</th>\n",
       "      <th>LOPD_CAUD_13_11</th>\n",
       "      <th>LOPD_CAUD_13_17</th>\n",
       "      <th>LOPD_CAUD_13_44</th>\n",
       "      <th>LOPD_CAUD_14_04</th>\n",
       "      <th>LOPD_HIPP_10_28</th>\n",
       "      <th>LOPD_HIPP_10_83</th>\n",
       "      <th>LOPD_HIPP_11_70</th>\n",
       "      <th>LOPD_HIPP_12_42</th>\n",
       "      <th>LOPD_HIPP_13_05</th>\n",
       "      <th>LOPD_HIPP_13_11</th>\n",
       "      <th>LOPD_HIPP_14_04</th>\n",
       "      <th>LOPD_MDFG_10_28</th>\n",
       "      <th>LOPD_MDFG_10_30</th>\n",
       "      <th>LOPD_MDFG_10_76</th>\n",
       "      <th>LOPD_MDFG_10_83</th>\n",
       "      <th>LOPD_MDFG_11_70</th>\n",
       "      <th>LOPD_MDFG_12_04</th>\n",
       "      <th>LOPD_MDFG_12_22</th>\n",
       "      <th>LOPD_MDFG_12_29</th>\n",
       "      <th>LOPD_MDFG_12_42</th>\n",
       "      <th>LOPD_MDFG_13_05</th>\n",
       "      <th>LOPD_MDFG_13_11</th>\n",
       "      <th>LOPD_MDFG_13_17</th>\n",
       "      <th>LOPD_MDFG_13_44</th>\n",
       "      <th>LOPD_MDFG_14_04</th>\n",
       "      <th>LOPD_MDTG_01_38</th>\n",
       "      <th>LOPD_MDTG_10_28</th>\n",
       "      <th>LOPD_MDTG_10_30</th>\n",
       "      <th>LOPD_MDTG_10_76</th>\n",
       "      <th>LOPD_MDTG_10_83</th>\n",
       "      <th>LOPD_MDTG_11_70</th>\n",
       "      <th>LOPD_MDTG_12_22</th>\n",
       "      <th>LOPD_MDTG_12_42</th>\n",
       "      <th>LOPD_MDTG_13_05</th>\n",
       "      <th>LOPD_MDTG_13_11</th>\n",
       "      <th>LOPD_MDTG_13_44</th>\n",
       "      <th>LOPD_MDTG_14_04</th>\n",
       "      <th>LOPD_PTMN_10_28</th>\n",
       "      <th>LOPD_PTMN_10_30</th>\n",
       "      <th>LOPD_PTMN_10_76</th>\n",
       "      <th>LOPD_PTMN_10_83</th>\n",
       "      <th>LOPD_PTMN_11_70</th>\n",
       "      <th>LOPD_PTMN_12_04</th>\n",
       "      <th>LOPD_PTMN_12_22</th>\n",
       "      <th>LOPD_PTMN_12_29</th>\n",
       "      <th>LOPD_PTMN_12_42</th>\n",
       "      <th>LOPD_PTMN_13_05</th>\n",
       "      <th>LOPD_PTMN_13_11</th>\n",
       "      <th>LOPD_PTMN_13_17</th>\n",
       "      <th>LOPD_PTMN_13_44</th>\n",
       "      <th>LOPD_PTMN_14_04</th>\n",
       "      <th>LOPD_SUNI_10_28</th>\n",
       "      <th>LOPD_SUNI_10_76</th>\n",
       "      <th>LOPD_SUNI_10_83</th>\n",
       "      <th>LOPD_SUNI_11_70</th>\n",
       "      <th>LOPD_SUNI_12_04</th>\n",
       "      <th>LOPD_SUNI_12_22</th>\n",
       "      <th>LOPD_SUNI_12_42</th>\n",
       "      <th>LOPD_SUNI_13_05</th>\n",
       "      <th>LOPD_SUNI_13_11</th>\n",
       "      <th>LOPD_SUNI_13_17</th>\n",
       "      <th>LOPD_SUNI_13_44</th>\n",
       "      <th>LOPD_SUNI_14_04</th>\n",
       "      <th>LRRK_CAUD_04_10</th>\n",
       "      <th>LRRK_CAUD_10_37</th>\n",
       "      <th>LRRK_CAUD_13_60</th>\n",
       "      <th>LRRK_HIPP_04_10</th>\n",
       "      <th>LRRK_MDFG_04_10</th>\n",
       "      <th>LRRK_MDFG_13_60</th>\n",
       "      <th>LRRK_MDTG_01_39</th>\n",
       "      <th>LRRK_MDTG_04_10</th>\n",
       "      <th>LRRK_MDTG_10_37</th>\n",
       "      <th>LRRK_MDTG_13_60</th>\n",
       "      <th>LRRK_PTMN_01_39</th>\n",
       "      <th>LRRK_PTMN_04_10</th>\n",
       "      <th>LRRK_PTMN_10_37</th>\n",
       "      <th>LRRK_PTMN_13_60</th>\n",
       "      <th>LRRK_SUNI_04_10</th>\n",
       "      <th>LRRK_SUNI_10_37</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10015</td>\n",
       "      <td>10231</td>\n",
       "      <td>12</td>\n",
       "      <td>16</td>\n",
       "      <td>22</td>\n",
       "      <td>12</td>\n",
       "      <td>14</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>22</td>\n",
       "      <td>16</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>10</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "      <td>14</td>\n",
       "      <td>12</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>20</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>16</td>\n",
       "      <td>6</td>\n",
       "      <td>26</td>\n",
       "      <td>8</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>10</td>\n",
       "      <td>28</td>\n",
       "      <td>38</td>\n",
       "      <td>34</td>\n",
       "      <td>24</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>16</td>\n",
       "      <td>24</td>\n",
       "      <td>8</td>\n",
       "      <td>6</td>\n",
       "      <td>70</td>\n",
       "      <td>28</td>\n",
       "      <td>64</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>28</td>\n",
       "      <td>34</td>\n",
       "      <td>37</td>\n",
       "      <td>18</td>\n",
       "      <td>6</td>\n",
       "      <td>12</td>\n",
       "      <td>22</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>28</td>\n",
       "      <td>26</td>\n",
       "      <td>32</td>\n",
       "      <td>22</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>12</td>\n",
       "      <td>13</td>\n",
       "      <td>41</td>\n",
       "      <td>16</td>\n",
       "      <td>34</td>\n",
       "      <td>38</td>\n",
       "      <td>6</td>\n",
       "      <td>40</td>\n",
       "      <td>13</td>\n",
       "      <td>56</td>\n",
       "      <td>26</td>\n",
       "      <td>12</td>\n",
       "      <td>26</td>\n",
       "      <td>16</td>\n",
       "      <td>30</td>\n",
       "      <td>14</td>\n",
       "      <td>18</td>\n",
       "      <td>30</td>\n",
       "      <td>14</td>\n",
       "      <td>28</td>\n",
       "      <td>11</td>\n",
       "      <td>22</td>\n",
       "      <td>10</td>\n",
       "      <td>28</td>\n",
       "      <td>50</td>\n",
       "      <td>12</td>\n",
       "      <td>20</td>\n",
       "      <td>23</td>\n",
       "      <td>14</td>\n",
       "      <td>22</td>\n",
       "      <td>24</td>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "      <td>19</td>\n",
       "      <td>22</td>\n",
       "      <td>24</td>\n",
       "      <td>30</td>\n",
       "      <td>8</td>\n",
       "      <td>18</td>\n",
       "      <td>6</td>\n",
       "      <td>37</td>\n",
       "      <td>32</td>\n",
       "      <td>17</td>\n",
       "      <td>43</td>\n",
       "      <td>12</td>\n",
       "      <td>24</td>\n",
       "      <td>36</td>\n",
       "      <td>18</td>\n",
       "      <td>14</td>\n",
       "      <td>6</td>\n",
       "      <td>11</td>\n",
       "      <td>20</td>\n",
       "      <td>28</td>\n",
       "      <td>26</td>\n",
       "      <td>6</td>\n",
       "      <td>24</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>16</td>\n",
       "      <td>35</td>\n",
       "      <td>8</td>\n",
       "      <td>18</td>\n",
       "      <td>8</td>\n",
       "      <td>39</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>24</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>31</td>\n",
       "      <td>24</td>\n",
       "      <td>28</td>\n",
       "      <td>6</td>\n",
       "      <td>16</td>\n",
       "      <td>26</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>26</td>\n",
       "      <td>10</td>\n",
       "      <td>12</td>\n",
       "      <td>18</td>\n",
       "      <td>26</td>\n",
       "      <td>4</td>\n",
       "      <td>18</td>\n",
       "      <td>20</td>\n",
       "      <td>8</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>12</td>\n",
       "      <td>36</td>\n",
       "      <td>2</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>16</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>16</td>\n",
       "      <td>12</td>\n",
       "      <td>20</td>\n",
       "      <td>34</td>\n",
       "      <td>22</td>\n",
       "      <td>8</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>26</td>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>20</td>\n",
       "      <td>8</td>\n",
       "      <td>6</td>\n",
       "      <td>14</td>\n",
       "      <td>10</td>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>26</td>\n",
       "      <td>12</td>\n",
       "      <td>16</td>\n",
       "      <td>30</td>\n",
       "      <td>18</td>\n",
       "      <td>16</td>\n",
       "      <td>21</td>\n",
       "      <td>16</td>\n",
       "      <td>38</td>\n",
       "      <td>20</td>\n",
       "      <td>8</td>\n",
       "      <td>6</td>\n",
       "      <td>22</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>50</td>\n",
       "      <td>14</td>\n",
       "      <td>6</td>\n",
       "      <td>18</td>\n",
       "      <td>6</td>\n",
       "      <td>16</td>\n",
       "      <td>26</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>16</td>\n",
       "      <td>13</td>\n",
       "      <td>14</td>\n",
       "      <td>11</td>\n",
       "      <td>16</td>\n",
       "      <td>4</td>\n",
       "      <td>21</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>26</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>6</td>\n",
       "      <td>16</td>\n",
       "      <td>20</td>\n",
       "      <td>2</td>\n",
       "      <td>14</td>\n",
       "      <td>18</td>\n",
       "      <td>28</td>\n",
       "      <td>8</td>\n",
       "      <td>21</td>\n",
       "      <td>12</td>\n",
       "      <td>24</td>\n",
       "      <td>26</td>\n",
       "      <td>16</td>\n",
       "      <td>18</td>\n",
       "      <td>42</td>\n",
       "      <td>36</td>\n",
       "      <td>40</td>\n",
       "      <td>24</td>\n",
       "      <td>8</td>\n",
       "      <td>18</td>\n",
       "      <td>35</td>\n",
       "      <td>14</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>10</td>\n",
       "      <td>12</td>\n",
       "      <td>21</td>\n",
       "      <td>8</td>\n",
       "      <td>18</td>\n",
       "      <td>16</td>\n",
       "      <td>17</td>\n",
       "      <td>14</td>\n",
       "      <td>20</td>\n",
       "      <td>32</td>\n",
       "      <td>44</td>\n",
       "      <td>33</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>14</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>22</td>\n",
       "      <td>20</td>\n",
       "      <td>36</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "      <td>16</td>\n",
       "      <td>20</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>22</td>\n",
       "      <td>10</td>\n",
       "      <td>20</td>\n",
       "      <td>10</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>32</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>26</td>\n",
       "      <td>6</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>29</td>\n",
       "      <td>18</td>\n",
       "      <td>20</td>\n",
       "      <td>32</td>\n",
       "      <td>28</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>24</td>\n",
       "      <td>23</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>20</td>\n",
       "      <td>22</td>\n",
       "      <td>16</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>66</td>\n",
       "      <td>6</td>\n",
       "      <td>10</td>\n",
       "      <td>26</td>\n",
       "      <td>42</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>19</td>\n",
       "      <td>6</td>\n",
       "      <td>8</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>40</td>\n",
       "      <td>28</td>\n",
       "      <td>32</td>\n",
       "      <td>24</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>24</td>\n",
       "      <td>26</td>\n",
       "      <td>16</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>17</td>\n",
       "      <td>18</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>18</td>\n",
       "      <td>6</td>\n",
       "      <td>44</td>\n",
       "      <td>6</td>\n",
       "      <td>42</td>\n",
       "      <td>12</td>\n",
       "      <td>22</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>24</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>8</td>\n",
       "      <td>10</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>14</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>9</td>\n",
       "      <td>22</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>181363</td>\n",
       "      <td>181563</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>22</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>14</td>\n",
       "      <td>31</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>18</td>\n",
       "      <td>8</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>11</td>\n",
       "      <td>18</td>\n",
       "      <td>9</td>\n",
       "      <td>39</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>25</td>\n",
       "      <td>9</td>\n",
       "      <td>16</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>9</td>\n",
       "      <td>27</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>7</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>11</td>\n",
       "      <td>13</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>22</td>\n",
       "      <td>7</td>\n",
       "      <td>17</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>1</td>\n",
       "      <td>25</td>\n",
       "      <td>6</td>\n",
       "      <td>22</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>21</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>11</td>\n",
       "      <td>8</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>19</td>\n",
       "      <td>2</td>\n",
       "      <td>29</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>26</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>6</td>\n",
       "      <td>14</td>\n",
       "      <td>21</td>\n",
       "      <td>9</td>\n",
       "      <td>27</td>\n",
       "      <td>2</td>\n",
       "      <td>19</td>\n",
       "      <td>38</td>\n",
       "      <td>10</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>18</td>\n",
       "      <td>19</td>\n",
       "      <td>16</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>20</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>8</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>11</td>\n",
       "      <td>59</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>8</td>\n",
       "      <td>23</td>\n",
       "      <td>13</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>16</td>\n",
       "      <td>11</td>\n",
       "      <td>15</td>\n",
       "      <td>11</td>\n",
       "      <td>12</td>\n",
       "      <td>10</td>\n",
       "      <td>14</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>9</td>\n",
       "      <td>11</td>\n",
       "      <td>15</td>\n",
       "      <td>10</td>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>12</td>\n",
       "      <td>26</td>\n",
       "      <td>15</td>\n",
       "      <td>17</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>18</td>\n",
       "      <td>14</td>\n",
       "      <td>6</td>\n",
       "      <td>19</td>\n",
       "      <td>28</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>10</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>7</td>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>12</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>11</td>\n",
       "      <td>38</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>3</td>\n",
       "      <td>11</td>\n",
       "      <td>7</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>7</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>28</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>27</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>19</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>29</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>6</td>\n",
       "      <td>13</td>\n",
       "      <td>12</td>\n",
       "      <td>9</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>6</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>12</td>\n",
       "      <td>10</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td>14</td>\n",
       "      <td>6</td>\n",
       "      <td>23</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>11</td>\n",
       "      <td>9</td>\n",
       "      <td>15</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>20</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>3</td>\n",
       "      <td>22</td>\n",
       "      <td>12</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>27</td>\n",
       "      <td>7</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>12</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>21</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>19</td>\n",
       "      <td>9</td>\n",
       "      <td>71</td>\n",
       "      <td>7</td>\n",
       "      <td>48</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>8</td>\n",
       "      <td>12</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>183716</td>\n",
       "      <td>183916</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "      <td>19</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>16</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>10</td>\n",
       "      <td>3</td>\n",
       "      <td>21</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>7</td>\n",
       "      <td>15</td>\n",
       "      <td>12</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>17</td>\n",
       "      <td>31</td>\n",
       "      <td>11</td>\n",
       "      <td>19</td>\n",
       "      <td>4</td>\n",
       "      <td>7</td>\n",
       "      <td>12</td>\n",
       "      <td>20</td>\n",
       "      <td>18</td>\n",
       "      <td>9</td>\n",
       "      <td>3</td>\n",
       "      <td>25</td>\n",
       "      <td>15</td>\n",
       "      <td>23</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>14</td>\n",
       "      <td>9</td>\n",
       "      <td>16</td>\n",
       "      <td>22</td>\n",
       "      <td>11</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>20</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "      <td>8</td>\n",
       "      <td>21</td>\n",
       "      <td>11</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>16</td>\n",
       "      <td>13</td>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "      <td>18</td>\n",
       "      <td>5</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>17</td>\n",
       "      <td>25</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>23</td>\n",
       "      <td>12</td>\n",
       "      <td>18</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>10</td>\n",
       "      <td>12</td>\n",
       "      <td>7</td>\n",
       "      <td>21</td>\n",
       "      <td>10</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>24</td>\n",
       "      <td>15</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>17</td>\n",
       "      <td>9</td>\n",
       "      <td>30</td>\n",
       "      <td>12</td>\n",
       "      <td>24</td>\n",
       "      <td>5</td>\n",
       "      <td>36</td>\n",
       "      <td>21</td>\n",
       "      <td>12</td>\n",
       "      <td>35</td>\n",
       "      <td>6</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>21</td>\n",
       "      <td>47</td>\n",
       "      <td>15</td>\n",
       "      <td>26</td>\n",
       "      <td>4</td>\n",
       "      <td>24</td>\n",
       "      <td>15</td>\n",
       "      <td>16</td>\n",
       "      <td>13</td>\n",
       "      <td>10</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>18</td>\n",
       "      <td>15</td>\n",
       "      <td>16</td>\n",
       "      <td>15</td>\n",
       "      <td>9</td>\n",
       "      <td>8</td>\n",
       "      <td>24</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>22</td>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>5</td>\n",
       "      <td>14</td>\n",
       "      <td>12</td>\n",
       "      <td>7</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>8</td>\n",
       "      <td>14</td>\n",
       "      <td>18</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "      <td>13</td>\n",
       "      <td>16</td>\n",
       "      <td>5</td>\n",
       "      <td>27</td>\n",
       "      <td>3</td>\n",
       "      <td>11</td>\n",
       "      <td>21</td>\n",
       "      <td>8</td>\n",
       "      <td>6</td>\n",
       "      <td>21</td>\n",
       "      <td>7</td>\n",
       "      <td>13</td>\n",
       "      <td>26</td>\n",
       "      <td>24</td>\n",
       "      <td>7</td>\n",
       "      <td>10</td>\n",
       "      <td>16</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>21</td>\n",
       "      <td>2</td>\n",
       "      <td>21</td>\n",
       "      <td>5</td>\n",
       "      <td>8</td>\n",
       "      <td>15</td>\n",
       "      <td>6</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>13</td>\n",
       "      <td>5</td>\n",
       "      <td>7</td>\n",
       "      <td>14</td>\n",
       "      <td>11</td>\n",
       "      <td>25</td>\n",
       "      <td>9</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>27</td>\n",
       "      <td>14</td>\n",
       "      <td>32</td>\n",
       "      <td>5</td>\n",
       "      <td>16</td>\n",
       "      <td>22</td>\n",
       "      <td>8</td>\n",
       "      <td>3</td>\n",
       "      <td>19</td>\n",
       "      <td>17</td>\n",
       "      <td>28</td>\n",
       "      <td>35</td>\n",
       "      <td>8</td>\n",
       "      <td>35</td>\n",
       "      <td>29</td>\n",
       "      <td>19</td>\n",
       "      <td>18</td>\n",
       "      <td>29</td>\n",
       "      <td>9</td>\n",
       "      <td>5</td>\n",
       "      <td>16</td>\n",
       "      <td>31</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>16</td>\n",
       "      <td>9</td>\n",
       "      <td>10</td>\n",
       "      <td>27</td>\n",
       "      <td>17</td>\n",
       "      <td>4</td>\n",
       "      <td>7</td>\n",
       "      <td>17</td>\n",
       "      <td>10</td>\n",
       "      <td>6</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>33</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>13</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>20</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>9</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>9</td>\n",
       "      <td>18</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>29</td>\n",
       "      <td>14</td>\n",
       "      <td>43</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>9</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>20</td>\n",
       "      <td>5</td>\n",
       "      <td>21</td>\n",
       "      <td>13</td>\n",
       "      <td>16</td>\n",
       "      <td>13</td>\n",
       "      <td>9</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>13</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>9</td>\n",
       "      <td>3</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>6</td>\n",
       "      <td>24</td>\n",
       "      <td>6</td>\n",
       "      <td>12</td>\n",
       "      <td>10</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>26</td>\n",
       "      <td>18</td>\n",
       "      <td>5</td>\n",
       "      <td>9</td>\n",
       "      <td>17</td>\n",
       "      <td>12</td>\n",
       "      <td>10</td>\n",
       "      <td>21</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>9</td>\n",
       "      <td>7</td>\n",
       "      <td>14</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>18</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td>15</td>\n",
       "      <td>9</td>\n",
       "      <td>28</td>\n",
       "      <td>6</td>\n",
       "      <td>35</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>19</td>\n",
       "      <td>9</td>\n",
       "      <td>14</td>\n",
       "      <td>25</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>13</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>28</td>\n",
       "      <td>16</td>\n",
       "      <td>5</td>\n",
       "      <td>9</td>\n",
       "      <td>25</td>\n",
       "      <td>54</td>\n",
       "      <td>16</td>\n",
       "      <td>19</td>\n",
       "      <td>19</td>\n",
       "      <td>4</td>\n",
       "      <td>21</td>\n",
       "      <td>10</td>\n",
       "      <td>12</td>\n",
       "      <td>22</td>\n",
       "      <td>29</td>\n",
       "      <td>4</td>\n",
       "      <td>23</td>\n",
       "      <td>24</td>\n",
       "      <td>17</td>\n",
       "      <td>10</td>\n",
       "      <td>16</td>\n",
       "      <td>12</td>\n",
       "      <td>5</td>\n",
       "      <td>19</td>\n",
       "      <td>21</td>\n",
       "      <td>9</td>\n",
       "      <td>26</td>\n",
       "      <td>23</td>\n",
       "      <td>10</td>\n",
       "      <td>20</td>\n",
       "      <td>8</td>\n",
       "      <td>20</td>\n",
       "      <td>19</td>\n",
       "      <td>8</td>\n",
       "      <td>10</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>8</td>\n",
       "      <td>28</td>\n",
       "      <td>10</td>\n",
       "      <td>44</td>\n",
       "      <td>15</td>\n",
       "      <td>62</td>\n",
       "      <td>11</td>\n",
       "      <td>26</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>25</td>\n",
       "      <td>9</td>\n",
       "      <td>17</td>\n",
       "      <td>11</td>\n",
       "      <td>16</td>\n",
       "      <td>5</td>\n",
       "      <td>13</td>\n",
       "      <td>15</td>\n",
       "      <td>16</td>\n",
       "      <td>24</td>\n",
       "      <td>1</td>\n",
       "      <td>19</td>\n",
       "      <td>10</td>\n",
       "      <td>21</td>\n",
       "      <td>3</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>6</td>\n",
       "      <td>20</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>184083</td>\n",
       "      <td>184283</td>\n",
       "      <td>11</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>19</td>\n",
       "      <td>20</td>\n",
       "      <td>27</td>\n",
       "      <td>6</td>\n",
       "      <td>18</td>\n",
       "      <td>24</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>16</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>22</td>\n",
       "      <td>13</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>14</td>\n",
       "      <td>10</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>13</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>28</td>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>17</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>7</td>\n",
       "      <td>19</td>\n",
       "      <td>17</td>\n",
       "      <td>15</td>\n",
       "      <td>16</td>\n",
       "      <td>23</td>\n",
       "      <td>15</td>\n",
       "      <td>11</td>\n",
       "      <td>26</td>\n",
       "      <td>6</td>\n",
       "      <td>17</td>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "      <td>27</td>\n",
       "      <td>9</td>\n",
       "      <td>24</td>\n",
       "      <td>53</td>\n",
       "      <td>14</td>\n",
       "      <td>23</td>\n",
       "      <td>15</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>13</td>\n",
       "      <td>22</td>\n",
       "      <td>14</td>\n",
       "      <td>25</td>\n",
       "      <td>11</td>\n",
       "      <td>44</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td>25</td>\n",
       "      <td>9</td>\n",
       "      <td>17</td>\n",
       "      <td>6</td>\n",
       "      <td>33</td>\n",
       "      <td>22</td>\n",
       "      <td>15</td>\n",
       "      <td>20</td>\n",
       "      <td>24</td>\n",
       "      <td>25</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>23</td>\n",
       "      <td>11</td>\n",
       "      <td>26</td>\n",
       "      <td>17</td>\n",
       "      <td>20</td>\n",
       "      <td>37</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>25</td>\n",
       "      <td>16</td>\n",
       "      <td>25</td>\n",
       "      <td>9</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>20</td>\n",
       "      <td>10</td>\n",
       "      <td>11</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>13</td>\n",
       "      <td>32</td>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>11</td>\n",
       "      <td>22</td>\n",
       "      <td>24</td>\n",
       "      <td>14</td>\n",
       "      <td>7</td>\n",
       "      <td>30</td>\n",
       "      <td>11</td>\n",
       "      <td>47</td>\n",
       "      <td>13</td>\n",
       "      <td>18</td>\n",
       "      <td>20</td>\n",
       "      <td>54</td>\n",
       "      <td>19</td>\n",
       "      <td>24</td>\n",
       "      <td>30</td>\n",
       "      <td>12</td>\n",
       "      <td>18</td>\n",
       "      <td>22</td>\n",
       "      <td>25</td>\n",
       "      <td>52</td>\n",
       "      <td>25</td>\n",
       "      <td>53</td>\n",
       "      <td>4</td>\n",
       "      <td>50</td>\n",
       "      <td>31</td>\n",
       "      <td>21</td>\n",
       "      <td>15</td>\n",
       "      <td>7</td>\n",
       "      <td>11</td>\n",
       "      <td>17</td>\n",
       "      <td>38</td>\n",
       "      <td>39</td>\n",
       "      <td>15</td>\n",
       "      <td>26</td>\n",
       "      <td>16</td>\n",
       "      <td>12</td>\n",
       "      <td>14</td>\n",
       "      <td>7</td>\n",
       "      <td>26</td>\n",
       "      <td>19</td>\n",
       "      <td>14</td>\n",
       "      <td>22</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>13</td>\n",
       "      <td>14</td>\n",
       "      <td>17</td>\n",
       "      <td>19</td>\n",
       "      <td>51</td>\n",
       "      <td>7</td>\n",
       "      <td>15</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>8</td>\n",
       "      <td>12</td>\n",
       "      <td>13</td>\n",
       "      <td>21</td>\n",
       "      <td>16</td>\n",
       "      <td>30</td>\n",
       "      <td>17</td>\n",
       "      <td>31</td>\n",
       "      <td>23</td>\n",
       "      <td>15</td>\n",
       "      <td>10</td>\n",
       "      <td>24</td>\n",
       "      <td>5</td>\n",
       "      <td>25</td>\n",
       "      <td>28</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>16</td>\n",
       "      <td>19</td>\n",
       "      <td>26</td>\n",
       "      <td>18</td>\n",
       "      <td>36</td>\n",
       "      <td>8</td>\n",
       "      <td>43</td>\n",
       "      <td>13</td>\n",
       "      <td>17</td>\n",
       "      <td>9</td>\n",
       "      <td>21</td>\n",
       "      <td>13</td>\n",
       "      <td>9</td>\n",
       "      <td>24</td>\n",
       "      <td>13</td>\n",
       "      <td>12</td>\n",
       "      <td>13</td>\n",
       "      <td>19</td>\n",
       "      <td>31</td>\n",
       "      <td>24</td>\n",
       "      <td>12</td>\n",
       "      <td>6</td>\n",
       "      <td>33</td>\n",
       "      <td>31</td>\n",
       "      <td>37</td>\n",
       "      <td>15</td>\n",
       "      <td>34</td>\n",
       "      <td>5</td>\n",
       "      <td>14</td>\n",
       "      <td>22</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>33</td>\n",
       "      <td>33</td>\n",
       "      <td>40</td>\n",
       "      <td>46</td>\n",
       "      <td>14</td>\n",
       "      <td>40</td>\n",
       "      <td>34</td>\n",
       "      <td>22</td>\n",
       "      <td>11</td>\n",
       "      <td>29</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>31</td>\n",
       "      <td>61</td>\n",
       "      <td>10</td>\n",
       "      <td>19</td>\n",
       "      <td>45</td>\n",
       "      <td>13</td>\n",
       "      <td>9</td>\n",
       "      <td>61</td>\n",
       "      <td>14</td>\n",
       "      <td>10</td>\n",
       "      <td>20</td>\n",
       "      <td>21</td>\n",
       "      <td>20</td>\n",
       "      <td>10</td>\n",
       "      <td>11</td>\n",
       "      <td>18</td>\n",
       "      <td>19</td>\n",
       "      <td>38</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>22</td>\n",
       "      <td>32</td>\n",
       "      <td>5</td>\n",
       "      <td>19</td>\n",
       "      <td>12</td>\n",
       "      <td>20</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>16</td>\n",
       "      <td>4</td>\n",
       "      <td>17</td>\n",
       "      <td>5</td>\n",
       "      <td>14</td>\n",
       "      <td>9</td>\n",
       "      <td>17</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>16</td>\n",
       "      <td>40</td>\n",
       "      <td>15</td>\n",
       "      <td>58</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>15</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>33</td>\n",
       "      <td>7</td>\n",
       "      <td>40</td>\n",
       "      <td>26</td>\n",
       "      <td>11</td>\n",
       "      <td>25</td>\n",
       "      <td>6</td>\n",
       "      <td>13</td>\n",
       "      <td>18</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>33</td>\n",
       "      <td>18</td>\n",
       "      <td>11</td>\n",
       "      <td>9</td>\n",
       "      <td>13</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>12</td>\n",
       "      <td>16</td>\n",
       "      <td>7</td>\n",
       "      <td>16</td>\n",
       "      <td>7</td>\n",
       "      <td>17</td>\n",
       "      <td>24</td>\n",
       "      <td>16</td>\n",
       "      <td>6</td>\n",
       "      <td>10</td>\n",
       "      <td>30</td>\n",
       "      <td>20</td>\n",
       "      <td>7</td>\n",
       "      <td>22</td>\n",
       "      <td>17</td>\n",
       "      <td>24</td>\n",
       "      <td>12</td>\n",
       "      <td>14</td>\n",
       "      <td>11</td>\n",
       "      <td>15</td>\n",
       "      <td>27</td>\n",
       "      <td>26</td>\n",
       "      <td>8</td>\n",
       "      <td>23</td>\n",
       "      <td>6</td>\n",
       "      <td>19</td>\n",
       "      <td>29</td>\n",
       "      <td>19</td>\n",
       "      <td>11</td>\n",
       "      <td>14</td>\n",
       "      <td>29</td>\n",
       "      <td>6</td>\n",
       "      <td>40</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>17</td>\n",
       "      <td>31</td>\n",
       "      <td>6</td>\n",
       "      <td>16</td>\n",
       "      <td>29</td>\n",
       "      <td>21</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>10</td>\n",
       "      <td>15</td>\n",
       "      <td>39</td>\n",
       "      <td>37</td>\n",
       "      <td>11</td>\n",
       "      <td>23</td>\n",
       "      <td>19</td>\n",
       "      <td>50</td>\n",
       "      <td>35</td>\n",
       "      <td>32</td>\n",
       "      <td>27</td>\n",
       "      <td>8</td>\n",
       "      <td>37</td>\n",
       "      <td>11</td>\n",
       "      <td>19</td>\n",
       "      <td>18</td>\n",
       "      <td>31</td>\n",
       "      <td>4</td>\n",
       "      <td>26</td>\n",
       "      <td>15</td>\n",
       "      <td>19</td>\n",
       "      <td>17</td>\n",
       "      <td>18</td>\n",
       "      <td>24</td>\n",
       "      <td>12</td>\n",
       "      <td>16</td>\n",
       "      <td>22</td>\n",
       "      <td>19</td>\n",
       "      <td>33</td>\n",
       "      <td>20</td>\n",
       "      <td>27</td>\n",
       "      <td>24</td>\n",
       "      <td>14</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>15</td>\n",
       "      <td>13</td>\n",
       "      <td>6</td>\n",
       "      <td>8</td>\n",
       "      <td>23</td>\n",
       "      <td>20</td>\n",
       "      <td>23</td>\n",
       "      <td>28</td>\n",
       "      <td>15</td>\n",
       "      <td>33</td>\n",
       "      <td>30</td>\n",
       "      <td>52</td>\n",
       "      <td>15</td>\n",
       "      <td>42</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>31</td>\n",
       "      <td>12</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>17</td>\n",
       "      <td>2</td>\n",
       "      <td>27</td>\n",
       "      <td>18</td>\n",
       "      <td>27</td>\n",
       "      <td>26</td>\n",
       "      <td>1</td>\n",
       "      <td>20</td>\n",
       "      <td>26</td>\n",
       "      <td>25</td>\n",
       "      <td>3</td>\n",
       "      <td>22</td>\n",
       "      <td>18</td>\n",
       "      <td>11</td>\n",
       "      <td>26</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>184370</td>\n",
       "      <td>184570</td>\n",
       "      <td>7</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>9</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>15</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>18</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>8</td>\n",
       "      <td>12</td>\n",
       "      <td>11</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>7</td>\n",
       "      <td>16</td>\n",
       "      <td>15</td>\n",
       "      <td>17</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>7</td>\n",
       "      <td>11</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>28</td>\n",
       "      <td>22</td>\n",
       "      <td>20</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>12</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>6</td>\n",
       "      <td>8</td>\n",
       "      <td>16</td>\n",
       "      <td>10</td>\n",
       "      <td>25</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>11</td>\n",
       "      <td>14</td>\n",
       "      <td>3</td>\n",
       "      <td>13</td>\n",
       "      <td>5</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>9</td>\n",
       "      <td>23</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>23</td>\n",
       "      <td>9</td>\n",
       "      <td>25</td>\n",
       "      <td>12</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>7</td>\n",
       "      <td>6</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>10</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>9</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td>28</td>\n",
       "      <td>6</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>39</td>\n",
       "      <td>6</td>\n",
       "      <td>24</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>24</td>\n",
       "      <td>9</td>\n",
       "      <td>27</td>\n",
       "      <td>3</td>\n",
       "      <td>37</td>\n",
       "      <td>15</td>\n",
       "      <td>8</td>\n",
       "      <td>5</td>\n",
       "      <td>7</td>\n",
       "      <td>15</td>\n",
       "      <td>19</td>\n",
       "      <td>5</td>\n",
       "      <td>15</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>6</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>22</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td>21</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>24</td>\n",
       "      <td>8</td>\n",
       "      <td>6</td>\n",
       "      <td>10</td>\n",
       "      <td>5</td>\n",
       "      <td>8</td>\n",
       "      <td>10</td>\n",
       "      <td>24</td>\n",
       "      <td>6</td>\n",
       "      <td>29</td>\n",
       "      <td>20</td>\n",
       "      <td>7</td>\n",
       "      <td>10</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>11</td>\n",
       "      <td>10</td>\n",
       "      <td>18</td>\n",
       "      <td>6</td>\n",
       "      <td>8</td>\n",
       "      <td>15</td>\n",
       "      <td>11</td>\n",
       "      <td>13</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>8</td>\n",
       "      <td>9</td>\n",
       "      <td>19</td>\n",
       "      <td>16</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>14</td>\n",
       "      <td>15</td>\n",
       "      <td>6</td>\n",
       "      <td>24</td>\n",
       "      <td>14</td>\n",
       "      <td>12</td>\n",
       "      <td>7</td>\n",
       "      <td>13</td>\n",
       "      <td>10</td>\n",
       "      <td>16</td>\n",
       "      <td>5</td>\n",
       "      <td>26</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>12</td>\n",
       "      <td>11</td>\n",
       "      <td>21</td>\n",
       "      <td>20</td>\n",
       "      <td>9</td>\n",
       "      <td>16</td>\n",
       "      <td>10</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>17</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>31</td>\n",
       "      <td>4</td>\n",
       "      <td>7</td>\n",
       "      <td>18</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>41</td>\n",
       "      <td>9</td>\n",
       "      <td>7</td>\n",
       "      <td>10</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>9</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>18</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>9</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>13</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>5</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>9</td>\n",
       "      <td>10</td>\n",
       "      <td>20</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>9</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "      <td>13</td>\n",
       "      <td>4</td>\n",
       "      <td>18</td>\n",
       "      <td>9</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>13</td>\n",
       "      <td>5</td>\n",
       "      <td>14</td>\n",
       "      <td>7</td>\n",
       "      <td>12</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "      <td>7</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>18</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>12</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>12</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>4</td>\n",
       "      <td>14</td>\n",
       "      <td>23</td>\n",
       "      <td>10</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>20</td>\n",
       "      <td>2</td>\n",
       "      <td>18</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>19</td>\n",
       "      <td>4</td>\n",
       "      <td>15</td>\n",
       "      <td>17</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>25</td>\n",
       "      <td>22</td>\n",
       "      <td>7</td>\n",
       "      <td>13</td>\n",
       "      <td>11</td>\n",
       "      <td>25</td>\n",
       "      <td>13</td>\n",
       "      <td>9</td>\n",
       "      <td>16</td>\n",
       "      <td>9</td>\n",
       "      <td>18</td>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>18</td>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>15</td>\n",
       "      <td>8</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>10</td>\n",
       "      <td>11</td>\n",
       "      <td>9</td>\n",
       "      <td>17</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>18</td>\n",
       "      <td>11</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>8</td>\n",
       "      <td>9</td>\n",
       "      <td>30</td>\n",
       "      <td>5</td>\n",
       "      <td>22</td>\n",
       "      <td>15</td>\n",
       "      <td>32</td>\n",
       "      <td>6</td>\n",
       "      <td>13</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>20</td>\n",
       "      <td>6</td>\n",
       "      <td>9</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>14</td>\n",
       "      <td>9</td>\n",
       "      <td>19</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>16</td>\n",
       "      <td>11</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td>23</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chrom   start     end  ADAD_CAUD_00_0281  ADAD_CAUD_00_0387  \\\n",
       "0  chr1   10015   10231                 12                 16   \n",
       "1  chr1  181363  181563                  1                  6   \n",
       "2  chr1  183716  183916                  7                  4   \n",
       "3  chr1  184083  184283                 11                  8   \n",
       "4  chr1  184370  184570                  7                  6   \n",
       "\n",
       "   ADAD_CAUD_01_0164  ADAD_CAUD_01_1400  ADAD_CAUD_06_0194  ADAD_CAUD_06_1486  \\\n",
       "0                 22                 12                 14                 20   \n",
       "1                  4                  2                 10                  6   \n",
       "2                  4                 12                  2                 10   \n",
       "3                  4                 11                  2                 19   \n",
       "4                  6                  0                  6                  9   \n",
       "\n",
       "   ADAD_CAUD_07_0787  ADAD_CAUD_12829xx  ADAD_CAUD_23156xx  ADAD_CAUD_24281xx  \\\n",
       "0                 12                 22                 16                 10   \n",
       "1                  1                  9                  1                  3   \n",
       "2                 19                 14                  8                 10   \n",
       "3                 20                 27                  6                 18   \n",
       "4                  6                 11                  5                  9   \n",
       "\n",
       "   ADAD_CAUD_26436xx  ADAD_CAUD_AWQ6JZx  ADAD_CAUD_C0T7AAx  ADAD_CAUD_DCQGVNx  \\\n",
       "0                 10                  4                  0                  6   \n",
       "1                  6                  1                  6                  7   \n",
       "2                 10                  5                  5                  5   \n",
       "3                 24                  1                 11                 16   \n",
       "4                  7                  0                  0                  9   \n",
       "\n",
       "   ADAD_CAUD_SXIONRx  ADAD_CAUD_WYYIWGx  ADAD_CAUD_YUJW97x  ADAD_HIPP_12829xx  \\\n",
       "0                  2                  6                 10                 22   \n",
       "1                  4                  2                  6                 22   \n",
       "2                 11                  1                 10                 16   \n",
       "3                  9                  0                 12                 22   \n",
       "4                  3                  4                  9                 15   \n",
       "\n",
       "   ADAD_PARL_00_0700  ADAD_PARL_02_0621  ADAD_PARL_06_1486  ADAD_PARL_12540xx  \\\n",
       "0                  0                  8                  4                  4   \n",
       "1                  4                  5                  6                  2   \n",
       "2                  7                  4                  8                 11   \n",
       "3                 13                  6                  3                 10   \n",
       "4                  9                  0                  7                  9   \n",
       "\n",
       "   ADAD_PARL_24281xx  ADAD_PARL_25044xx  ADAD_PARL_26436xx  ADAD_PARL_DCQGVNx  \\\n",
       "0                  4                 10                 12                  4   \n",
       "1                  0                  3                  9                  7   \n",
       "2                  3                  6                  7                  2   \n",
       "3                  3                  3                 14                 10   \n",
       "4                  2                  1                  2                  3   \n",
       "\n",
       "   ADAD_PARL_HQFTL0x  ADAD_PARL_SXIONRx  ADAD_PARL_WYYIWGx  ADAD_PARL_YUJW97x  \\\n",
       "0                 12                 10                  4                  6   \n",
       "1                  1                  5                  3                 10   \n",
       "2                 15                  3                  0                  2   \n",
       "3                  6                  4                  3                 13   \n",
       "4                  1                  8                  0                  3   \n",
       "\n",
       "   ADAD_SMTG_00_0387  ADAD_SMTG_00_0700  ADAD_SMTG_02_0621  ADAD_SMTG_06_1486  \\\n",
       "0                 10                  4                  2                 10   \n",
       "1                  3                  6                  0                  6   \n",
       "2                  9                 10                  3                 21   \n",
       "3                 11                 11                  4                 12   \n",
       "4                  7                 10                  2                  5   \n",
       "\n",
       "   ADAD_SMTG_07_0787  ADAD_SMTG_12540xx  ADAD_SMTG_12829xx  ADAD_SMTG_23156xx  \\\n",
       "0                 14                 12                 10                  4   \n",
       "1                  5                  2                 10                  3   \n",
       "2                 11                 11                 13                 13   \n",
       "3                  4                  9                 28                 15   \n",
       "4                  4                  4                 18                 11   \n",
       "\n",
       "   ADAD_SMTG_23710xx  ADAD_SMTG_24281xx  ADAD_SMTG_26436xx  ADAD_SMTG_AWQ6JZx  \\\n",
       "0                  8                  4                  8                  8   \n",
       "1                  2                  2                  5                  2   \n",
       "2                  1                 10                 10                  4   \n",
       "3                  1                  7                 15                  0   \n",
       "4                  3                  2                 10                  0   \n",
       "\n",
       "   ADAD_SMTG_C0T7AAx  ADAD_SMTG_DCQGVNx  ADAD_SMTG_SXIONRx  ADAD_SMTG_WYYIWGx  \\\n",
       "0                  4                  2                 20                  4   \n",
       "1                  2                  9                 14                  2   \n",
       "2                  4                  4                 16                  0   \n",
       "3                  2                  9                 17                  0   \n",
       "4                  6                  3                 13                  0   \n",
       "\n",
       "   ADAD_SMTG_YUJW97x  CTRH_CAUD_07_1144  CTRH_CAUD_07_1287  CTRH_CAUD_08_0712  \\\n",
       "0                  2                 16                  6                 26   \n",
       "1                  0                  0                  3                  3   \n",
       "2                 17                  4                  5                  7   \n",
       "3                 16                  4                  4                  7   \n",
       "4                  7                  0                  5                  8   \n",
       "\n",
       "   CTRH_CAUD_10_0867  CTRH_CAUD_13_0962  CTRH_CAUD_14_0380  CTRH_CAUD_14_0513  \\\n",
       "0                  8                 24                 24                 10   \n",
       "1                  7                  7                  5                  3   \n",
       "2                 15                 12                 18                  4   \n",
       "3                 19                 17                 15                 16   \n",
       "4                 12                 11                  7                  9   \n",
       "\n",
       "   CTRH_CAUD_14_1383  CTRH_CAUD_15_1023  CTRH_CAUD_15_1025  CTRH_HIPP_07_1058  \\\n",
       "0                 28                 38                 34                 24   \n",
       "1                  4                 14                 31                  5   \n",
       "2                 17                 31                 11                 19   \n",
       "3                 23                 15                 11                 26   \n",
       "4                  7                 16                 15                 17   \n",
       "\n",
       "   CTRH_HIPP_07_1144  CTRH_HIPP_07_1287  CTRH_HIPP_08_0298  CTRH_HIPP_10_0867  \\\n",
       "0                 20                 20                 20                 16   \n",
       "1                  2                  9                 18                  8   \n",
       "2                  4                  7                 12                 20   \n",
       "3                  6                 17                 23                 23   \n",
       "4                  0                 10                  7                 11   \n",
       "\n",
       "   CTRH_HIPP_13_0962  CTRH_HIPP_14_0380  CTRH_HIPP_14_0513  CTRH_HIPP_14_1383  \\\n",
       "0                 24                  8                  6                 70   \n",
       "1                  5                  2                 11                 18   \n",
       "2                 18                  9                  3                 25   \n",
       "3                 27                  9                 24                 53   \n",
       "4                 12                  1                  8                 28   \n",
       "\n",
       "   CTRH_HIPP_15_1023  CTRH_HIPP_15_1025  CTRH_PARL_07_1058  CTRH_PARL_07_1144  \\\n",
       "0                 28                 64                 12                 12   \n",
       "1                  9                 39                  4                  0   \n",
       "2                 15                 23                 10                  0   \n",
       "3                 14                 23                 15                  3   \n",
       "4                 22                 20                 11                  1   \n",
       "\n",
       "   CTRH_PARL_07_1287  CTRH_PARL_08_0298  CTRH_PARL_10_0867  CTRH_PARL_13_0962  \\\n",
       "0                 12                 16                 16                  8   \n",
       "1                 15                 25                  9                 16   \n",
       "2                  9                 14                  9                 16   \n",
       "3                 10                 21                 21                 13   \n",
       "4                  8                 12                 17                  8   \n",
       "\n",
       "   CTRH_PARL_14_0380  CTRH_PARL_14_0513  CTRH_PARL_14_1383  CTRH_PARL_15_1023  \\\n",
       "0                  4                 10                 28                 34   \n",
       "1                  5                  5                 18                  9   \n",
       "2                 22                 11                 12                 12   \n",
       "3                 22                 14                 25                 11   \n",
       "4                  6                  8                 16                 10   \n",
       "\n",
       "   CTRH_PARL_15_1025  CTRH_SMTG_07_1058  CTRH_SMTG_07_1144  CTRH_SMTG_07_1287  \\\n",
       "0                 37                 18                  6                 12   \n",
       "1                 27                  5                  0                  5   \n",
       "2                 20                  9                  0                 21   \n",
       "3                 44                  7                  4                 25   \n",
       "4                 25                  7                  1                  5   \n",
       "\n",
       "   CTRH_SMTG_08_0712  CTRH_SMTG_10_0867  CTRH_SMTG_13_0962  CTRH_SMTG_14_0380  \\\n",
       "0                 22                  8                  8                 18   \n",
       "1                  7                 11                  5                  8   \n",
       "2                  5                 10                  8                 21   \n",
       "3                  9                 17                  6                 33   \n",
       "4                 11                 14                  3                 13   \n",
       "\n",
       "   CTRH_SMTG_14_0513  CTRH_SMTG_14_1383  CTRH_SMTG_15_1023  CTRH_SMTG_15_1025  \\\n",
       "0                 10                 10                 28                 26   \n",
       "1                  1                 12                 11                 13   \n",
       "2                 11                 15                 15                 16   \n",
       "3                 22                 15                 20                 24   \n",
       "4                  5                  9                  9                 14   \n",
       "\n",
       "   CTRL_CAUD_00_38  CTRL_CAUD_01_31  CTRL_CAUD_03_15  CTRL_CAUD_03_39  \\\n",
       "0               32               22                4                8   \n",
       "1                9                4                3                5   \n",
       "2               13               15                1               18   \n",
       "3               25               15                2               23   \n",
       "4               14                8                1               13   \n",
       "\n",
       "   CTRL_CAUD_03_41  CTRL_CAUD_03_66  CTRL_CAUD_04_38  CTRL_CAUD_05_16  \\\n",
       "0               12               13               41               16   \n",
       "1                4               22                7               17   \n",
       "2                5               16               16               17   \n",
       "3               11               26               17               20   \n",
       "4                2               14               15                9   \n",
       "\n",
       "   CTRL_CAUD_06_0615  CTRL_CAUD_06_1516  CTRL_CAUD_08_90  CTRL_CAUD_09_1589  \\\n",
       "0                 34                 38                6                 40   \n",
       "1                 24                 24                1                 25   \n",
       "2                 25                  4                2                 23   \n",
       "3                 37                 16                8                 25   \n",
       "4                 23                 11                2                 23   \n",
       "\n",
       "   CTRL_CAUD_09_35  CTRL_CAUD_11_0393  CTRL_CAUD_13_0038  CTRL_CAUD_13_1226  \\\n",
       "0               13                 56                 26                 12   \n",
       "1                6                 22                  2                  3   \n",
       "2               12                 18                  5                  6   \n",
       "3               16                 25                  9                 14   \n",
       "4                9                 25                 12                  5   \n",
       "\n",
       "   CTRL_CAUD_14_0586  CTRL_CAUD_14_0941  CTRL_CAUD_14_1018  CTRL_CAUD_15_78  \\\n",
       "0                 26                 16                 30               14   \n",
       "1                  2                  3                  2                3   \n",
       "2                  2                 11                  2                8   \n",
       "3                  8                 20                 10               11   \n",
       "4                  2                  6                  5                7   \n",
       "\n",
       "   CTRL_CAUD_16_10  CTRL_CAUD_16_32  CTRL_HIPP_00_38  CTRL_HIPP_04_38  \\\n",
       "0               18               30               14               28   \n",
       "1               21                8                7               11   \n",
       "2               10               12                7               21   \n",
       "3               26               19               13               32   \n",
       "4                6                9                6               10   \n",
       "\n",
       "   CTRL_HIPP_05_16  CTRL_HIPP_06_1516  CTRL_HIPP_09_35  CTRL_HIPP_11_0393  \\\n",
       "0               11                 22               10                 28   \n",
       "1                8                 10                1                  8   \n",
       "2               10                  3               10                 24   \n",
       "3                6                  7               11                 22   \n",
       "4                6                  1                5                  9   \n",
       "\n",
       "   CTRL_HIPP_13_0038  CTRL_HIPP_13_1226  CTRL_HIPP_14_0586  CTRL_HIPP_14_0941  \\\n",
       "0                 50                 12                 20                 23   \n",
       "1                  9                  6                  3                 19   \n",
       "2                 15                 13                  2                 17   \n",
       "3                 24                 14                  7                 30   \n",
       "4                  8                  8                  4                 12   \n",
       "\n",
       "   CTRL_HIPP_14_1018  CTRL_MDFG_00_38  CTRL_MDFG_01_37  CTRL_MDFG_03_39  \\\n",
       "0                 14               22               24                2   \n",
       "1                  2               29                3                6   \n",
       "2                  9               30               12               24   \n",
       "3                 11               47               13               18   \n",
       "4                  3               28                6               17   \n",
       "\n",
       "   CTRL_MDFG_03_41  CTRL_MDFG_03_66  CTRL_MDFG_04_38  CTRL_MDFG_05_16  \\\n",
       "0               10               19               22               24   \n",
       "1                2               26               10               10   \n",
       "2                5               36               21               12   \n",
       "3               20               54               19               24   \n",
       "4                8               39                6               24   \n",
       "\n",
       "   CTRL_MDFG_09_35  CTRL_MDFG_09_50  CTRL_MDFG_15_78  CTRL_MDFG_16_10  \\\n",
       "0               30                8               18                6   \n",
       "1               12                3                7                6   \n",
       "2               35                6               14               14   \n",
       "3               30               12               18               22   \n",
       "4               14                8                9                9   \n",
       "\n",
       "   CTRL_MDFG_16_32  CTRL_MDTG_00_38  CTRL_MDTG_03_15  CTRL_MDTG_03_39  \\\n",
       "0               37               32               17               43   \n",
       "1               14               21                9               27   \n",
       "2               21               47               15               26   \n",
       "3               25               52               25               53   \n",
       "4                9               24                9               27   \n",
       "\n",
       "   CTRL_MDTG_03_41  CTRL_MDTG_03_66  CTRL_MDTG_04_38  CTRL_MDTG_05_16  \\\n",
       "0               12               24               36               18   \n",
       "1                2               19               38               10   \n",
       "2                4               24               15               16   \n",
       "3                4               50               31               21   \n",
       "4                3               37               15                8   \n",
       "\n",
       "   CTRL_MDTG_09_35  CTRL_MDTG_09_50  CTRL_MDTG_15_78  CTRL_MDTG_16_10  \\\n",
       "0               14                6               11               20   \n",
       "1                5                5                6               18   \n",
       "2               13               10               15               15   \n",
       "3               15                7               11               17   \n",
       "4                5                7               15               19   \n",
       "\n",
       "   CTRL_MDTG_16_32  CTRL_PARL_09_1589  CTRL_PARL_11_0311  CTRL_PARL_11_0393  \\\n",
       "0               28                 26                  6                 24   \n",
       "1               19                 16                  2                  0   \n",
       "2               21                 21                 18                 15   \n",
       "3               38                 39                 15                 26   \n",
       "4                5                 15                  3                  7   \n",
       "\n",
       "   CTRL_PARL_13_0038  CTRL_PARL_13_1226  CTRL_PARL_14_0586  CTRL_PARL_14_1018  \\\n",
       "0                 18                 10                 10                 16   \n",
       "1                 12                  8                  4                  3   \n",
       "2                 16                 15                  9                  8   \n",
       "3                 16                 12                 14                  7   \n",
       "4                  3                  3                  6                  5   \n",
       "\n",
       "   CTRL_PTMN_00_38  CTRL_PTMN_03_39  CTRL_PTMN_03_41  CTRL_PTMN_03_66  \\\n",
       "0               35                8               18                8   \n",
       "1               20                9                6                8   \n",
       "2               24               11                5               16   \n",
       "3               26               19               14               22   \n",
       "4               18                6               12               12   \n",
       "\n",
       "   CTRL_PTMN_04_38  CTRL_PTMN_05_16  CTRL_PTMN_07_28  CTRL_PTMN_09_35  \\\n",
       "0               39               10                2               24   \n",
       "1               12                2                5                4   \n",
       "2                8               22                1               16   \n",
       "3               16               16                7                7   \n",
       "4               22               10                2                9   \n",
       "\n",
       "   CTRL_PTMN_09_50  CTRL_PTMN_15_78  CTRL_PTMN_16_32  CTRL_SMTG_06_1516  \\\n",
       "0                8                8               31                 24   \n",
       "1                3                4                5                 11   \n",
       "2                5               14               12                  7   \n",
       "3               13               14               17                 19   \n",
       "4                1               11                7                  4   \n",
       "\n",
       "   CTRL_SMTG_09_1589  CTRL_SMTG_11_0311  CTRL_SMTG_11_0393  CTRL_SMTG_13_0038  \\\n",
       "0                 28                  6                 16                 26   \n",
       "1                 59                  3                  4                 10   \n",
       "2                 11                  5                  8                 14   \n",
       "3                 51                  7                 15                 18   \n",
       "4                 21                  1                 12                 24   \n",
       "\n",
       "   CTRL_SMTG_13_1226  CTRL_SMTG_14_0586  CTRL_SMTG_14_1018  CTRL_SUNI_00_38  \\\n",
       "0                  2                  6                 26               10   \n",
       "1                  8                  4                  4                6   \n",
       "2                 18                  5                 10               13   \n",
       "3                 18                  8                 12               13   \n",
       "4                  8                  6                 10                5   \n",
       "\n",
       "   CTRL_SUNI_01_31  CTRL_SUNI_03_15  CTRL_SUNI_03_39  CTRL_SUNI_03_41  \\\n",
       "0               12               18               26                4   \n",
       "1                9                2                9                8   \n",
       "2               16                5               27                3   \n",
       "3               21               16               30               17   \n",
       "4                8               10               24                6   \n",
       "\n",
       "   CTRL_SUNI_03_66  CTRL_SUNI_04_38  CTRL_SUNI_05_16  CTRL_SUNI_06_57  \\\n",
       "0               18               20                8                6   \n",
       "1               23               13                7                0   \n",
       "2               11               21                8                6   \n",
       "3               31               23               15               10   \n",
       "4               29               20                7               10   \n",
       "\n",
       "   CTRL_SUNI_07_28  CTRL_SUNI_09_35  CTRL_SUNI_09_50  CTRL_SUNI_15_78  \\\n",
       "0                5               12               36                2   \n",
       "1                4                2               16               11   \n",
       "2               21                7               13               26   \n",
       "3               24                5               25               28   \n",
       "4                7                2               15               11   \n",
       "\n",
       "   CTRL_SUNI_16_32  GBA1_CAUD_00_26  GBA1_CAUD_00_27  GBA1_CAUD_00_45  \\\n",
       "0               16                8               16                4   \n",
       "1               15               11               12               10   \n",
       "2               24                7               10               16   \n",
       "3               22               22               16               19   \n",
       "4               10               18                6                8   \n",
       "\n",
       "   GBA1_CAUD_03_43  GBA1_CAUD_03_51  GBA1_CAUD_04_37  GBA1_CAUD_07_41  \\\n",
       "0                8               14               15               12   \n",
       "1               14                4               10               10   \n",
       "2               18               10               21                2   \n",
       "3               26               18               36                8   \n",
       "4               15               11               13                5   \n",
       "\n",
       "   GBA1_CAUD_11_90  GBA1_CAUD_12_65  GBA1_CAUD_12_69  GBA1_CAUD_93_19  \\\n",
       "0                8               16               12               20   \n",
       "1                7                4                6                3   \n",
       "2               21                5                8               15   \n",
       "3               43               13               17                9   \n",
       "4               18                8                9               19   \n",
       "\n",
       "   GBA1_CAUD_94_31  GBA1_CAUD_97_52  GBA1_HIPP_03_43  GBA1_HIPP_11_90  \\\n",
       "0               34               22                8               12   \n",
       "1                8                7                4                4   \n",
       "2                6                8                8               13   \n",
       "3               21               13                9               24   \n",
       "4               16                3                5                2   \n",
       "\n",
       "   GBA1_HIPP_12_65  GBA1_HIPP_93_19  GBA1_HIPP_94_31  GBA1_HIPP_97_52  \\\n",
       "0                4               13               26               20   \n",
       "1                4                7                8                9   \n",
       "2                5                7               14               11   \n",
       "3               13               12               13               19   \n",
       "4                3               14               15                6   \n",
       "\n",
       "   GBA1_MDFG_00_26  GBA1_MDFG_00_27  GBA1_MDFG_00_45  GBA1_MDFG_03_43  \\\n",
       "0               20               18                0               20   \n",
       "1               11               15               10                6   \n",
       "2               25                9               13               13   \n",
       "3               31               24               12                6   \n",
       "4               24               14               12                7   \n",
       "\n",
       "   GBA1_MDFG_03_51  GBA1_MDFG_04_37  GBA1_MDFG_07_41  GBA1_MDFG_11_90  \\\n",
       "0                8                6               14               10   \n",
       "1                7               12               26               15   \n",
       "2               25               25               27               14   \n",
       "3               33               31               37               15   \n",
       "4               13               10               16                5   \n",
       "\n",
       "   GBA1_MDFG_12_65  GBA1_MDFG_12_69  GBA1_MDFG_13_22  GBA1_MDFG_93_19  \\\n",
       "0                8                2               12               26   \n",
       "1               17                0                0                4   \n",
       "2               32                5               16               22   \n",
       "3               34                5               14               22   \n",
       "4               26                2                5               10   \n",
       "\n",
       "   GBA1_MDFG_94_31  GBA1_MDFG_97_52  GBA1_MDTG_00_09  GBA1_MDTG_00_26  \\\n",
       "0               12               16               30               18   \n",
       "1               11                9                2                4   \n",
       "2                8                3               19               17   \n",
       "3               18               10               33               33   \n",
       "4                6                5               12               11   \n",
       "\n",
       "   GBA1_MDTG_00_27  GBA1_MDTG_00_45  GBA1_MDTG_03_43  GBA1_MDTG_03_51  \\\n",
       "0               16               21               16               38   \n",
       "1               18               14                6               19   \n",
       "2               28               35                8               35   \n",
       "3               40               46               14               40   \n",
       "4               21               20                9               16   \n",
       "\n",
       "   GBA1_MDTG_04_37  GBA1_MDTG_11_90  GBA1_MDTG_12_50  GBA1_MDTG_12_65  \\\n",
       "0               20                8                6               22   \n",
       "1               28               13               13               13   \n",
       "2               29               19               18               29   \n",
       "3               34               22               11               29   \n",
       "4               10                3               10               17   \n",
       "\n",
       "   GBA1_MDTG_12_69  GBA1_MDTG_13_22  GBA1_MDTG_93_19  GBA1_MDTG_94_31  \\\n",
       "0                4                2               15               50   \n",
       "1               10                3                7               15   \n",
       "2                9                5               16               31   \n",
       "3               11                3               31               61   \n",
       "4                8                4               12               31   \n",
       "\n",
       "   GBA1_MDTG_97_52  GBA1_PTMN_00_09  GBA1_PTMN_00_26  GBA1_PTMN_00_45  \\\n",
       "0               14                6               18                6   \n",
       "1                2                9                7                6   \n",
       "2               14               14               16                9   \n",
       "3               10               19               45               13   \n",
       "4                4                7               18                8   \n",
       "\n",
       "   GBA1_PTMN_03_43  GBA1_PTMN_03_51  GBA1_PTMN_04_37  GBA1_PTMN_07_41  \\\n",
       "0               16               26               10                4   \n",
       "1                7               12                6                2   \n",
       "2               10               27               17                4   \n",
       "3                9               61               14               10   \n",
       "4                4               41                9                7   \n",
       "\n",
       "   GBA1_PTMN_11_90  GBA1_PTMN_12_65  GBA1_PTMN_12_69  GBA1_PTMN_13_22  \\\n",
       "0                6               16               13               14   \n",
       "1                9                4                2                8   \n",
       "2                7               17               10                6   \n",
       "3               20               21               20               10   \n",
       "4               10                6                6                9   \n",
       "\n",
       "   GBA1_PTMN_93_19  GBA1_PTMN_94_31  GBA1_PTMN_97_52  GBA1_SUNI_00_26  \\\n",
       "0               11               16                4               21   \n",
       "1                1                5               11               38   \n",
       "2                8                8                2               33   \n",
       "3               11               18               19               38   \n",
       "4                8                7                7               18   \n",
       "\n",
       "   GBA1_SUNI_00_27  GBA1_SUNI_00_45  GBA1_SUNI_03_43  GBA1_SUNI_03_51  \\\n",
       "0                4                8               26               18   \n",
       "1                4                5                6                4   \n",
       "2                2               12               13                6   \n",
       "3                7                7               22               32   \n",
       "4                4                8                9                6   \n",
       "\n",
       "   GBA1_SUNI_04_37  GBA1_SUNI_07_41  GBA1_SUNI_11_90  GBA1_SUNI_12_50  \\\n",
       "0                0                8                6               16   \n",
       "1                8                6                6               12   \n",
       "2                3               12                8                8   \n",
       "3                5               19               12               20   \n",
       "4                1               13                9                5   \n",
       "\n",
       "   GBA1_SUNI_12_65  GBA1_SUNI_13_22  GBA1_SUNI_94_31  GBA1_SUNI_97_52  \\\n",
       "0               20                2               14               18   \n",
       "1                8                3               11                7   \n",
       "2               20                8                8                9   \n",
       "3               17                7               16                4   \n",
       "4               10                2               13                3   \n",
       "\n",
       "   LOAD_CAUD_07_0226  LOAD_CAUD_07_0664  LOAD_CAUD_07_1112  LOAD_CAUD_09_1166  \\\n",
       "0                 28                  8                 21                 12   \n",
       "1                  5                  3                  4                  7   \n",
       "2                  5                 10                 10                  9   \n",
       "3                 17                  5                 14                  9   \n",
       "4                  9                  5                  7                  4   \n",
       "\n",
       "   LOAD_CAUD_10_0188  LOAD_CAUD_10_1363  LOAD_CAUD_10_1770  LOAD_CAUD_10_1930  \\\n",
       "0                 24                 26                 16                 18   \n",
       "1                 11                  5                  2                 28   \n",
       "2                 18                  6                  0                  9   \n",
       "3                 17                 10                 10                 16   \n",
       "4                  2                  8                  8                  9   \n",
       "\n",
       "   LOAD_CAUD_10_2007  LOAD_CAUD_11_0352  LOAD_CAUD_11_0773  LOAD_CAUD_11_1475  \\\n",
       "0                 42                 36                 40                 24   \n",
       "1                  1                 10                 27                  4   \n",
       "2                 29                 14                 43                  5   \n",
       "3                 40                 15                 58                  2   \n",
       "4                 10                 20                 22                  0   \n",
       "\n",
       "   LOAD_CAUD_11_1686  LOAD_CAUD_13_0855  LOAD_CAUD_13_1300  LOAD_HIPP_06_0308  \\\n",
       "0                  8                 18                 35                 14   \n",
       "1                  6                  0                 12                  1   \n",
       "2                  3                  8                  9                  8   \n",
       "3                  7                  8                 15                 14   \n",
       "4                  4                  5                  9                  5   \n",
       "\n",
       "   LOAD_HIPP_07_0226  LOAD_HIPP_07_0664  LOAD_HIPP_07_0997  LOAD_HIPP_07_1112  \\\n",
       "0                  4                 12                 10                 12   \n",
       "1                  6                 19                  4                 11   \n",
       "2                  7                 20                  5                 21   \n",
       "3                 14                 33                  7                 40   \n",
       "4                 10                 13                  4                 18   \n",
       "\n",
       "   LOAD_HIPP_09_1166  LOAD_HIPP_10_0188  LOAD_HIPP_10_1363  LOAD_HIPP_10_1770  \\\n",
       "0                 21                  8                 18                 16   \n",
       "1                  5                  1                  3                  8   \n",
       "2                 13                 16                 13                  9   \n",
       "3                 26                 11                 25                  6   \n",
       "4                  9                 11                  4                  2   \n",
       "\n",
       "   LOAD_HIPP_11_0352  LOAD_HIPP_11_0773  LOAD_HIPP_11_1475  LOAD_HIPP_12_1181  \\\n",
       "0                 17                 14                 20                 32   \n",
       "1                  5                 10                  4                 14   \n",
       "2                 11                 11                  5                 13   \n",
       "3                 13                 18                  6                  5   \n",
       "4                 12                 13                  5                 14   \n",
       "\n",
       "   LOAD_HIPP_13_0855  LOAD_HIPP_13_1300  LOAD_PARL_06_0308  LOAD_PARL_07_0226  \\\n",
       "0                 44                 33                  4                  4   \n",
       "1                  1                 29                  3                 10   \n",
       "2                 16                  8                  9                  3   \n",
       "3                 33                 18                 11                  9   \n",
       "4                  7                 12                  9                  0   \n",
       "\n",
       "   LOAD_PARL_07_0664  LOAD_PARL_07_0997  LOAD_PARL_07_1112  LOAD_PARL_08_0789  \\\n",
       "0                 15                  0                 12                 14   \n",
       "1                  6                  1                 11                  8   \n",
       "2                 14                  1                  9                  2   \n",
       "3                 13                  5                 18                 12   \n",
       "4                 13                  2                 10                  7   \n",
       "\n",
       "   LOAD_PARL_09_1166  LOAD_PARL_10_0188  LOAD_PARL_10_0213  LOAD_PARL_10_1363  \\\n",
       "0                 12                 12                 22                 20   \n",
       "1                  8                  1                  8                  8   \n",
       "2                  8                  6                 24                  6   \n",
       "3                 16                  7                 16                  7   \n",
       "4                 10                  1                 10                  5   \n",
       "\n",
       "   LOAD_PARL_10_1770  LOAD_PARL_10_2007  LOAD_PARL_11_0352  LOAD_PARL_11_1686  \\\n",
       "0                 36                 14                  2                 10   \n",
       "1                  8                  4                  2                  7   \n",
       "2                 12                 10                 13                  1   \n",
       "3                 17                 24                 16                  6   \n",
       "4                  4                  9                  4                  1   \n",
       "\n",
       "   LOAD_PARL_12_1181  LOAD_PARL_13_0855  LOAD_PARL_13_1300  LOAD_SMTG_06_0308  \\\n",
       "0                 16                 20                 18                  0   \n",
       "1                  6                 13                 12                  9   \n",
       "2                  8                 26                 18                  5   \n",
       "3                 10                 30                 20                  7   \n",
       "4                  5                 18                 16                  0   \n",
       "\n",
       "   LOAD_SMTG_07_0226  LOAD_SMTG_07_1112  LOAD_SMTG_09_1166  LOAD_SMTG_10_0188  \\\n",
       "0                 12                  2                  5                 22   \n",
       "1                  3                  7                  6                  9   \n",
       "2                  9                 17                 12                 10   \n",
       "3                 22                 17                 24                 12   \n",
       "4                 10                 12                  7                  8   \n",
       "\n",
       "   LOAD_SMTG_10_0213  LOAD_SMTG_10_1770  LOAD_SMTG_10_1930  LOAD_SMTG_10_2007  \\\n",
       "0                 10                 20                 10                 12   \n",
       "1                  2                  6                  6                 12   \n",
       "2                 21                  6                  6                  9   \n",
       "3                 14                 11                 15                 27   \n",
       "4                 11                  4                  5                 12   \n",
       "\n",
       "   LOAD_SMTG_11_0352  LOAD_SMTG_11_1475  LOAD_SMTG_12_1181  LOAD_SMTG_13_0855  \\\n",
       "0                 12                 32                  4                  6   \n",
       "1                 10                  3                  5                  4   \n",
       "2                  7                 14                  4                  9   \n",
       "3                 26                  8                 23                  6   \n",
       "4                 15                  0                 14                  4   \n",
       "\n",
       "   LOAD_SMTG_13_1300  LOPD_CAUD_10_28  LOPD_CAUD_10_30  LOPD_CAUD_10_76  \\\n",
       "0                 18               10               12                2   \n",
       "1                  3                5                8                7   \n",
       "2                 18               28               10               15   \n",
       "3                 19               29               19               11   \n",
       "4                 14               23               10                5   \n",
       "\n",
       "   LOPD_CAUD_10_83  LOPD_CAUD_11_70  LOPD_CAUD_12_04  LOPD_CAUD_12_22  \\\n",
       "0               12               26                6               10   \n",
       "1                4               14                6               23   \n",
       "2                9               28                6               35   \n",
       "3               14               29                6               40   \n",
       "4                3               20                2               18   \n",
       "\n",
       "   LOPD_CAUD_12_29  LOPD_CAUD_12_42  LOPD_CAUD_13_05  LOPD_CAUD_13_11  \\\n",
       "0                2                6               29               18   \n",
       "1                2                1                8               11   \n",
       "2                1                5                6               19   \n",
       "3                3                6               17               31   \n",
       "4                3                1                5               19   \n",
       "\n",
       "   LOPD_CAUD_13_17  LOPD_CAUD_13_44  LOPD_CAUD_14_04  LOPD_HIPP_10_28  \\\n",
       "0               20               32               28                6   \n",
       "1                9               15               14                8   \n",
       "2                9               14               25               11   \n",
       "3                6               16               29               21   \n",
       "4                4               15               17               12   \n",
       "\n",
       "   LOPD_HIPP_10_83  LOPD_HIPP_11_70  LOPD_HIPP_12_42  LOPD_HIPP_13_05  \\\n",
       "0                5                4                6                4   \n",
       "1                2                0                1                2   \n",
       "2                5               13                3                1   \n",
       "3                1                8                8               10   \n",
       "4                2                6                2                4   \n",
       "\n",
       "   LOPD_HIPP_13_11  LOPD_HIPP_14_04  LOPD_MDFG_10_28  LOPD_MDFG_10_30  \\\n",
       "0               24               23                4                8   \n",
       "1                8               20               15                2   \n",
       "2               10               28               16                5   \n",
       "3               15               39               37               11   \n",
       "4               10               25               22                7   \n",
       "\n",
       "   LOPD_MDFG_10_76  LOPD_MDFG_10_83  LOPD_MDFG_11_70  LOPD_MDFG_12_04  \\\n",
       "0                8               20               22               16   \n",
       "1                9                3               22               12   \n",
       "2                9               25               54               16   \n",
       "3               23               19               50               35   \n",
       "4               13               11               25               13   \n",
       "\n",
       "   LOPD_MDFG_12_22  LOPD_MDFG_12_29  LOPD_MDFG_12_42  LOPD_MDFG_13_05  \\\n",
       "0                6                6                4               66   \n",
       "1               10                4                0               27   \n",
       "2               19               19                4               21   \n",
       "3               32               27                8               37   \n",
       "4                9               16                9               18   \n",
       "\n",
       "   LOPD_MDFG_13_11  LOPD_MDFG_13_17  LOPD_MDFG_13_44  LOPD_MDFG_14_04  \\\n",
       "0                6               10               26               42   \n",
       "1                7                5                6               29   \n",
       "2               10               12               22               29   \n",
       "3               11               19               18               31   \n",
       "4                6                7               18               15   \n",
       "\n",
       "   LOPD_MDTG_01_38  LOPD_MDTG_10_28  LOPD_MDTG_10_30  LOPD_MDTG_10_76  \\\n",
       "0                2               12                4               19   \n",
       "1                0                5                7                8   \n",
       "2                4               23               24               17   \n",
       "3                4               26               15               19   \n",
       "4                1                4                4               15   \n",
       "\n",
       "   LOPD_MDTG_10_83  LOPD_MDTG_11_70  LOPD_MDTG_12_22  LOPD_MDTG_12_42  \\\n",
       "0                6                8               10                2   \n",
       "1                5                6               11                0   \n",
       "2               10               16               12                5   \n",
       "3               17               18               24               12   \n",
       "4                8               14               14                1   \n",
       "\n",
       "   LOPD_MDTG_13_05  LOPD_MDTG_13_11  LOPD_MDTG_13_44  LOPD_MDTG_14_04  \\\n",
       "0               40               28               32               24   \n",
       "1                5               12               11                5   \n",
       "2               19               21                9               26   \n",
       "3               16               22               19               33   \n",
       "4                2                8               10               11   \n",
       "\n",
       "   LOPD_PTMN_10_28  LOPD_PTMN_10_30  LOPD_PTMN_10_76  LOPD_PTMN_10_83  \\\n",
       "0                2                4               16                8   \n",
       "1                7                7               11                0   \n",
       "2               23               10               20                8   \n",
       "3               20               27               24               14   \n",
       "4                9               17                7                9   \n",
       "\n",
       "   LOPD_PTMN_11_70  LOPD_PTMN_12_04  LOPD_PTMN_12_22  LOPD_PTMN_12_29  \\\n",
       "0               24               26               16                2   \n",
       "1               13               21                2                8   \n",
       "2               20               19                8               10   \n",
       "3               31               31               15               13   \n",
       "4                9               18               11                7   \n",
       "\n",
       "   LOPD_PTMN_12_42  LOPD_PTMN_13_05  LOPD_PTMN_13_11  LOPD_PTMN_13_17  \\\n",
       "0               12               17               18               12   \n",
       "1                0                8               10                4   \n",
       "2                5                6               11               11   \n",
       "3                6                8               23               20   \n",
       "4                4                6                6                8   \n",
       "\n",
       "   LOPD_PTMN_13_44  LOPD_PTMN_14_04  LOPD_SUNI_10_28  LOPD_SUNI_10_76  \\\n",
       "0                8               18                6               44   \n",
       "1               11               19                9               71   \n",
       "2                8               28               10               44   \n",
       "3               23               28               15               33   \n",
       "4                9               30                5               22   \n",
       "\n",
       "   LOPD_SUNI_10_83  LOPD_SUNI_11_70  LOPD_SUNI_12_04  LOPD_SUNI_12_22  \\\n",
       "0                6               42               12               22   \n",
       "1                7               48                4               11   \n",
       "2               15               62               11               26   \n",
       "3               30               52               15               42   \n",
       "4               15               32                6               13   \n",
       "\n",
       "   LOPD_SUNI_12_42  LOPD_SUNI_13_05  LOPD_SUNI_13_11  LOPD_SUNI_13_17  \\\n",
       "0                4               13               24                0   \n",
       "1                5                6               12                3   \n",
       "2               10                4               25                9   \n",
       "3               11               11               31               12   \n",
       "4                7                7               20                6   \n",
       "\n",
       "   LOPD_SUNI_13_44  LOPD_SUNI_14_04  LRRK_CAUD_04_10  LRRK_CAUD_10_37  \\\n",
       "0               15                8               10               12   \n",
       "1               10                4               12                0   \n",
       "2               17               11               16                5   \n",
       "3               16                8               17                2   \n",
       "4                9                4                8                7   \n",
       "\n",
       "   LRRK_CAUD_13_60  LRRK_HIPP_04_10  LRRK_MDFG_04_10  LRRK_MDFG_13_60  \\\n",
       "0                8                6                0                4   \n",
       "1                5                8               12                7   \n",
       "2               13               15               16               24   \n",
       "3               27               18               27               26   \n",
       "4               14                9               19                3   \n",
       "\n",
       "   LRRK_MDTG_01_39  LRRK_MDTG_04_10  LRRK_MDTG_10_37  LRRK_MDTG_13_60  \\\n",
       "0                3                2               14               18   \n",
       "1                1                5                2               14   \n",
       "2                1               19               10               21   \n",
       "3                1               20               26               25   \n",
       "4                5               16               11               18   \n",
       "\n",
       "   LRRK_PTMN_01_39  LRRK_PTMN_04_10  LRRK_PTMN_10_37  LRRK_PTMN_13_60  \\\n",
       "0                0                4               12                9   \n",
       "1                0                4                7                4   \n",
       "2                3               16                8                6   \n",
       "3                3               22               18               11   \n",
       "4                1               13                7                4   \n",
       "\n",
       "   LRRK_SUNI_04_10  LRRK_SUNI_10_37  \n",
       "0               22               26  \n",
       "1                8               18  \n",
       "2               20               13  \n",
       "3               26               24  \n",
       "4               23               15  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "(385725, 415)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "idr_counts = pd.read_csv('/mnt/lab_data2/annashch/alzheimers_parkinsons/adpd.atac.idr.counts.txt.gz', sep='\\t')\n",
    "counts_header = idr_counts.columns\n",
    "display(idr_counts.head())\n",
    "display(idr_counts.shape)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Intersect Counts Matrix with 10kb Expanded SNPs "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>new_start</th>\n",
       "      <th>new_end</th>\n",
       "      <th>rsid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39822150</td>\n",
       "      <td>39822151</td>\n",
       "      <td>rs34640847</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39822244</td>\n",
       "      <td>39822245</td>\n",
       "      <td>rs36015266</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39822259</td>\n",
       "      <td>39822260</td>\n",
       "      <td>rs61779808</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39822791</td>\n",
       "      <td>39822792</td>\n",
       "      <td>rs61779809</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>39825224</td>\n",
       "      <td>39825225</td>\n",
       "      <td>rs72666941</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr  new_start   new_end        rsid\n",
       "0  chr1   39822150  39822151  rs34640847\n",
       "1  chr1   39822244  39822245  rs36015266\n",
       "2  chr1   39822259  39822260  rs61779808\n",
       "3  chr1   39822791  39822792  rs61779809\n",
       "4  chr1   39825224  39825225  rs72666941"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "616"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "expand_by = 0\n",
    "counts_bed = pybedtools.BedTool.from_dataframe(idr_counts)\n",
    "expanded_snps = hg38_bed.copy(deep=True)\n",
    "expanded_snps['new_start'] = expanded_snps['start'].map(lambda x : max(0, x - expand_by))\n",
    "expanded_snps['new_end'] = expanded_snps['pos'].map(lambda x : x + expand_by)\n",
    "expanded_snps = expanded_snps[['chr', 'new_start', 'new_end', 'rsid']]\n",
    "display(expanded_snps.head())\n",
    "snps_bed = pybedtools.BedTool.from_dataframe(expanded_snps)\n",
    "intersect_bed = counts_bed.intersect(snps_bed, u=True, wa=True)\n",
    "intersect_counts = pybedtools.BedTool.to_dataframe(intersect_bed, header=None)\n",
    "intersect_counts.columns = counts_header\n",
    "final_peaks = intersect_counts['chrom'] + '_' + intersect_counts['start'].astype(str) + '_' + intersect_counts['end'].astype(str)\n",
    "final_peaks = list(final_peaks)\n",
    "display(len(final_peaks))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Calculate GC Content"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# ref_fasta = '/mnt/lab_data3/soumyak/refs/hg38/GRCh38_no_alt_analysis_set_GCA_000001405.15.fasta'\n",
    "# ref = pysam.FastaFile(ref_fasta)\n",
    "\n",
    "# with open('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/gc_content.txt', 'w') as outfile:\n",
    "#     for index,row in idr_counts.iterrows():                                         \n",
    "#         seq = ref.fetch(row['chrom'],int(row['start']),int(row['end']))\n",
    "#         seq = seq.upper()\n",
    "#         g = seq.count('G')\n",
    "#         c = seq.count('C')\n",
    "#         gc = (g + c) / len(seq)\n",
    "#         if index % 20000 == 0:\n",
    "#             print(index)\n",
    "#         outfile.write(str(gc) + '\\n')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create Region-Specific Counts Matrices"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CAUD\n",
      "['CTRH_CAUD_07_1144', 'CTRH_CAUD_07_1287', 'CTRH_CAUD_08_0712', 'CTRH_CAUD_10_0867', 'CTRH_CAUD_13_0962', 'CTRH_CAUD_14_0380', 'CTRH_CAUD_14_0513', 'CTRH_CAUD_14_1383', 'CTRH_CAUD_15_1023', 'CTRH_CAUD_15_1025', 'CTRL_CAUD_06_0615', 'CTRL_CAUD_06_1516', 'CTRL_CAUD_09_1589', 'CTRL_CAUD_11_0393', 'CTRL_CAUD_13_0038', 'CTRL_CAUD_13_1226', 'CTRL_CAUD_14_0586', 'CTRL_CAUD_14_0941', 'CTRL_CAUD_14_1018', 'CTRL_CAUD_00_38', 'CTRL_CAUD_01_31', 'CTRL_CAUD_03_15', 'CTRL_CAUD_03_39', 'CTRL_CAUD_03_41', 'CTRL_CAUD_03_66', 'CTRL_CAUD_04_38', 'CTRL_CAUD_05_16', 'CTRL_CAUD_08_90', 'CTRL_CAUD_09_35', 'CTRL_CAUD_15_78', 'CTRL_CAUD_16_10', 'CTRL_CAUD_16_32']\n",
      "HIPP\n",
      "['CTRH_HIPP_07_1058', 'CTRH_HIPP_07_1144', 'CTRH_HIPP_07_1287', 'CTRH_HIPP_08_0298', 'CTRH_HIPP_10_0867', 'CTRH_HIPP_13_0962', 'CTRH_HIPP_14_0380', 'CTRH_HIPP_14_0513', 'CTRH_HIPP_14_1383', 'CTRH_HIPP_15_1023', 'CTRH_HIPP_15_1025', 'CTRL_HIPP_06_1516', 'CTRL_HIPP_11_0393', 'CTRL_HIPP_13_0038', 'CTRL_HIPP_13_1226', 'CTRL_HIPP_14_0586', 'CTRL_HIPP_14_0941', 'CTRL_HIPP_14_1018', 'CTRL_HIPP_00_38', 'CTRL_HIPP_04_38', 'CTRL_HIPP_05_16', 'CTRL_HIPP_09_35']\n",
      "MDFG\n",
      "['CTRL_MDFG_00_38', 'CTRL_MDFG_01_37', 'CTRL_MDFG_03_39', 'CTRL_MDFG_03_41', 'CTRL_MDFG_03_66', 'CTRL_MDFG_04_38', 'CTRL_MDFG_05_16', 'CTRL_MDFG_09_35', 'CTRL_MDFG_09_50', 'CTRL_MDFG_15_78', 'CTRL_MDFG_16_10', 'CTRL_MDFG_16_32']\n",
      "MDTG\n",
      "['CTRL_MDTG_00_38', 'CTRL_MDTG_03_15', 'CTRL_MDTG_03_39', 'CTRL_MDTG_03_41', 'CTRL_MDTG_03_66', 'CTRL_MDTG_04_38', 'CTRL_MDTG_05_16', 'CTRL_MDTG_09_35', 'CTRL_MDTG_09_50', 'CTRL_MDTG_15_78', 'CTRL_MDTG_16_10', 'CTRL_MDTG_16_32']\n",
      "PTMN\n",
      "['CTRL_PTMN_00_38', 'CTRL_PTMN_03_39', 'CTRL_PTMN_03_41', 'CTRL_PTMN_03_66', 'CTRL_PTMN_04_38', 'CTRL_PTMN_05_16', 'CTRL_PTMN_07_28', 'CTRL_PTMN_09_35', 'CTRL_PTMN_09_50', 'CTRL_PTMN_15_78', 'CTRL_PTMN_16_32']\n",
      "SUNI\n",
      "['CTRL_SUNI_00_38', 'CTRL_SUNI_01_31', 'CTRL_SUNI_03_15', 'CTRL_SUNI_03_39', 'CTRL_SUNI_03_41', 'CTRL_SUNI_03_66', 'CTRL_SUNI_04_38', 'CTRL_SUNI_05_16', 'CTRL_SUNI_06_57', 'CTRL_SUNI_07_28', 'CTRL_SUNI_09_35', 'CTRL_SUNI_09_50', 'CTRL_SUNI_15_78', 'CTRL_SUNI_16_32']\n",
      "SMTG\n",
      "['CTRH_SMTG_07_1058', 'CTRH_SMTG_07_1144', 'CTRH_SMTG_07_1287', 'CTRH_SMTG_08_0712', 'CTRH_SMTG_10_0867', 'CTRH_SMTG_13_0962', 'CTRH_SMTG_14_0380', 'CTRH_SMTG_14_0513', 'CTRH_SMTG_14_1383', 'CTRH_SMTG_15_1023', 'CTRH_SMTG_15_1025', 'CTRL_SMTG_06_1516', 'CTRL_SMTG_09_1589', 'CTRL_SMTG_11_0311', 'CTRL_SMTG_11_0393', 'CTRL_SMTG_13_0038', 'CTRL_SMTG_13_1226', 'CTRL_SMTG_14_0586', 'CTRL_SMTG_14_1018']\n",
      "PARL\n",
      "['CTRH_PARL_07_1058', 'CTRH_PARL_07_1144', 'CTRH_PARL_07_1287', 'CTRH_PARL_08_0298', 'CTRH_PARL_10_0867', 'CTRH_PARL_13_0962', 'CTRH_PARL_14_0380', 'CTRH_PARL_14_0513', 'CTRH_PARL_14_1383', 'CTRH_PARL_15_1023', 'CTRH_PARL_15_1025', 'CTRL_PARL_09_1589', 'CTRL_PARL_11_0311', 'CTRL_PARL_11_0393', 'CTRL_PARL_13_0038', 'CTRL_PARL_13_1226', 'CTRL_PARL_14_0586', 'CTRL_PARL_14_1018']\n"
     ]
    }
   ],
   "source": [
    "for reg in multi_bam_regs:\n",
    "    print(reg)\n",
    "    with open('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/bam_lists/' + reg + '_ctrl_bams.txt') as infile:\n",
    "        reg_bams = [i.strip() for i in infile.readlines()]\n",
    "    bam_names = [bam.split('/')[10] for bam in reg_bams]\n",
    "    print(bam_names)\n",
    "    reg_counts = idr_counts.copy(deep=True)\n",
    "    reg_counts['index'] = reg_counts['chrom'] + '_' + reg_counts['start'].astype(str) + '_' + reg_counts['end'].astype(str)\n",
    "    reg_counts = reg_counts[['index'] + bam_names]\n",
    "    reg_counts.to_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/'\n",
    "                        + reg + '/idr_counts.txt', index=False, header=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Generate Offsets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "# cd /home/users/soumyak/rasqual <- Run in Sherlock\n",
    "# bash /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/scripts/make_offsets.sh"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create Chromosome-Specific Inputs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "chroms = ['chr' + str(i) for i in range(1, 23)]\n",
    "chroms.append('chrX')\n",
    "chroms.append('chrY')\n",
    "# print(chroms)\n",
    "# for reg in multi_bam_regs:\n",
    "#     print(reg)\n",
    "#     reg_counts = pd.read_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/'\n",
    "#                         + reg + '/idr_counts.txt', header=None, sep='\\t')\n",
    "#     reg_offset = pd.read_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/'\n",
    "#                         + reg + '/offset.txt',header=None, sep='\\t')\n",
    "#     for chrom in chroms:\n",
    "#         if not os.path.isdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/'\n",
    "#                              + reg + '/' + chrom):\n",
    "#             os.mkdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/'\n",
    "#                      + reg + '/' + chrom)\n",
    "#         print(chrom)\n",
    "#         reg_chrom_counts = reg_counts.loc[reg_counts[0].apply(lambda x : x.startswith(chrom + '_'))]\n",
    "#         reg_chrom_offset = reg_offset.loc[reg_offset[0].apply(lambda x : x.startswith(chrom + '_'))]\n",
    "#         reg_chrom_counts.to_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/'\n",
    "#                                  + reg + '/' + chrom + '/Y.txt', index=False, header=False, sep='\\t')\n",
    "#         reg_chrom_offset.to_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/'\n",
    "#                                  + reg + '/' + chrom + '/K.txt', index=False, header=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Generate Binaries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# cd /home/users/soumyak/rasqual <- Run in Sherlock\n",
    "# bash /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/scripts/make_binary.sh"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get Genotyped ASVCFs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CAUD\n",
      "HIPP\n",
      "MDFG\n",
      "MDTG\n",
      "PTMN\n",
      "SUNI\n",
      "SMTG\n",
      "PARL\n"
     ]
    }
   ],
   "source": [
    "for reg in multi_bam_regs:\n",
    "    print(reg)\n",
    "    with open('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/bam_lists/' + reg + '_ctrl_bams.txt') as infile:\n",
    "        reg_bams = [i.strip() for i in infile.readlines()]\n",
    "    asvcf = pd.read_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' + reg + '/asvcf.gz',\n",
    "                        sep='\\t', skiprows=1, header=None)\n",
    "    bam_names = [bam.split('/')[10] for bam in reg_bams]\n",
    "    bam_patients = [name.split('_')[2] + '_' + name.split('_')[3] for name in bam_names]\n",
    "    header = ['chrom', 'pos', 'rsid', 'ref', 'alt', 'score', 'qc', 'info', 'format'] + bam_patients\n",
    "    asvcf.columns = header\n",
    "    asvcf.set_index('rsid', inplace=True)\n",
    "    #print(asvcf.head())\n",
    "    for patient in bam_patients:\n",
    "        if os.path.isfile('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/quasar/output/'\n",
    "                            + patient + '/genotypes.txt'):\n",
    "            genotype = pd.read_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/quasar/output/'\n",
    "                                   + patient + '/genotypes.txt', sep='\\t')\n",
    "        genotype = genotype[['rsID', 'map.g0', 'map.g1', 'map.g2']]\n",
    "        for index,row in genotype.iterrows():\n",
    "            if row['map.g1'] >= row['map.g0']:\n",
    "                if row['map.g1'] >= row['map.g2']:\n",
    "                    asvcf.at[row['rsID'], patient] = asvcf.at[row['rsID'], patient].replace('./.', '0/1')\n",
    "                else:\n",
    "                    asvcf.at[row['rsID'], patient] = asvcf.at[row['rsID'], patient].replace('./.', '1/1')\n",
    "            else:\n",
    "                if row['map.g0'] >= row['map.g2']:\n",
    "                    asvcf.at[row['rsID'], patient] = asvcf.at[row['rsID'], patient].replace('./.', '0/0')\n",
    "                else:\n",
    "                    asvcf.at[row['rsID'], patient] = asvcf.at[row['rsID'], patient].replace('./.', '1/1')\n",
    "    asvcf.reset_index(inplace=True)\n",
    "    asvcf = asvcf[header]\n",
    "    #print(asvcf.head())\n",
    "    asvcf.to_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' + reg + '/genotyped_asvcf.txt',\n",
    "                    sep='\\t', index=False)\n",
    "    for chrom in chroms:\n",
    "        if not os.path.isdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/'\n",
    "                             + reg + '/' + chrom):\n",
    "            os.mkdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/'\n",
    "                     + reg + '/' + chrom)\n",
    "        if not os.path.isdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/'\n",
    "                             + reg + '/' + chrom):\n",
    "            os.mkdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/'\n",
    "                     + reg + '/' + chrom)\n",
    "        #print(chrom)\n",
    "        chrom_asvcf = asvcf.loc[asvcf['chrom'] == chrom]\n",
    "        chrom_asvcf.to_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/'\n",
    "                                 + reg + '/' + chrom + '/asvcf', index=False, header=False, sep='\\t')\n",
    "        bgzip_cmd = 'bgzip -f /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' + reg + '/' + chrom + '/asvcf'\n",
    "        ! {bgzip_cmd}\n",
    "        tabix_cmd = 'tabix -f -p vcf /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' + reg + '/' + chrom + '/asvcf.gz'\n",
    "        ! {tabix_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CAUD\n",
      "HIPP\n",
      "MDFG\n",
      "MDTG\n",
      "PTMN\n",
      "SUNI\n",
      "SMTG\n",
      "PARL\n"
     ]
    }
   ],
   "source": [
    "for reg in multi_bam_regs:\n",
    "    print(reg)\n",
    "    for chrom in chroms:\n",
    "        #print(chrom)\n",
    "        if not os.path.isdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/scripts/'\n",
    "                             + reg + '/' + chrom):\n",
    "            os.mkdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/scripts/'\n",
    "                     + reg + '/' + chrom)\n",
    "        if not os.path.isdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/logs/'\n",
    "                             + reg + '/' + chrom):\n",
    "            os.mkdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/logs/'\n",
    "                     + reg + '/' + chrom)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Generate RASQUAL commands"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CAUD\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "HIPP\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "MDFG\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "MDTG\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "PTMN\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "SUNI\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "SMTG\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "PARL\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n"
     ]
    }
   ],
   "source": [
    "chroms_noXY = ['chr' + str(i) for i in range(1, 23)]\n",
    "for reg in multi_bam_regs:\n",
    "    print(reg)\n",
    "    reg_counts = pd.read_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/'\n",
    "                            + reg + '/idr_counts.txt', header=None, sep='\\t')\n",
    "    for chrom in chroms_noXY:\n",
    "        print(chrom)\n",
    "        asvcf = pd.read_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/'\n",
    "                            + reg + '/' + chrom + '/asvcf.gz', header=None, sep='\\t')\n",
    "        reg_chrom_counts = reg_counts.loc[reg_counts[0].apply(lambda x : x.startswith(chrom + '_'))]\n",
    "        #print(asvcf.head())\n",
    "        #print(reg_chrom_counts.head())\n",
    "        samples_n = reg_counts.shape[1] - 1\n",
    "        #print(\"Samples (n): \", samples_n)\n",
    "        with open('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/scripts/'\n",
    "                     + reg + '/' + chrom + '/run_rasqual.sh', 'w') as outfile, \\\n",
    "             open('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/scripts/'\n",
    "                     + reg + '/' + chrom + '/run_rasqual_permutation.sh', 'w') as outfile_perm:\n",
    "            outfile.write('#!/bin/bash\\n')\n",
    "            outfile_perm.write('#!/bin/bash\\n')\n",
    "            for ind,feature in reg_chrom_counts.iterrows():\n",
    "                #print(ind)\n",
    "                if feature[0] in final_peaks:\n",
    "                    feat_j = ind + 1\n",
    "                    feat_n = feature[0]\n",
    "                    #print(feat_j)\n",
    "                    feature_start = int(feature[0].split('_')[1])\n",
    "                    feature_end = int(feature[0].split('_')[2])\n",
    "                    testing_start = max(0, feature_start - expand_by)\n",
    "                    testing_end = feature_end + expand_by\n",
    "                    feature_snps = 0\n",
    "                    testing_snps = 0\n",
    "                    for index2,snp in asvcf.iterrows():\n",
    "                        if snp[1] >= feature_start and snp[1] <= feature_end:\n",
    "                            testing_snps += 1\n",
    "                            feature_snps += 1\n",
    "                        elif snp[1] >= testing_start and snp[1] <= testing_end:\n",
    "                            testing_snps += 1\n",
    "                    assert expand_by == 0\n",
    "                    assert feature_snps == testing_snps\n",
    "                    rasqual_cmd = 'tabix /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' \\\n",
    "                                    + reg + '/' + chrom + '/asvcf.gz ' + chrom + ':' + str(testing_start) + '-' + str(testing_end) \\\n",
    "                                    + ' | /home/users/soumyak/rasqual/bin/rasqual -y /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' \\\n",
    "                                    + reg + '/idr_counts.bin -k /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' \\\n",
    "                                    + reg + '/offset.bin -n ' + str(samples_n) + ' -j ' + str(feat_j) + ' -l ' + str(testing_snps) + ' -m ' \\\n",
    "                                    + str(feature_snps) + ' -s ' + str(feature_start) + ' -e ' + str(feature_end) + ' -f ' + feat_n\n",
    "                    rasqual_cmd_r = 'tabix /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' \\\n",
    "                                    + reg + '/' + chrom + '/asvcf.gz ' + chrom + ':' + str(testing_start) + '-' + str(testing_end) \\\n",
    "                                    + ' | /home/users/soumyak/rasqual/bin/rasqual -y /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' \\\n",
    "                                    + reg + '/idr_counts.bin -k /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' \\\n",
    "                                    + reg + '/offset.bin -n ' + str(samples_n) + ' -j ' + str(feat_j) + ' -l ' + str(testing_snps) + ' -m ' \\\n",
    "                                    + str(feature_snps) + ' -s ' + str(feature_start) + ' -e ' + str(feature_end) + ' -f ' + feat_n + ' -r'\n",
    "                    outfile.write(rasqual_cmd + '\\n')\n",
    "                    outfile_perm.write(rasqual_cmd_r + '\\n')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Concatenate RASQUAL outputs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CAUD\n",
      "HIPP\n",
      "MDFG\n",
      "MDTG\n",
      "PTMN\n",
      "SUNI\n",
      "SMTG\n",
      "PARL\n"
     ]
    }
   ],
   "source": [
    "for reg in multi_bam_regs:\n",
    "    print(reg)\n",
    "    ! cat /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/{reg}/chr*/rasqual_output.txt | sort -k11,11rn > /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/{reg}/all_{reg}_output.txt\n",
    "    ! cat /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/{reg}/chr*/rasqual_output_permutation.txt | sort -k11,11rn > /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/{reg}/all_{reg}_output_permutation.txt    \n",
    "! cat /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/*/all_*_output.txt | sort -k11,11rn> /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/all_regions_output.txt\n",
    "! cat /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/*/all_*_output_permutation.txt | sort -k11,11rn> /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/all_regions_output_permutation.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get P-values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CAUD\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>feature</th>\n",
       "      <th>rsid</th>\n",
       "      <th>chrom</th>\n",
       "      <th>snp_pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>allele_freq</th>\n",
       "      <th>HWE_chi_square_statistic</th>\n",
       "      <th>imputation_quality_score</th>\n",
       "      <th>log10_bh_qvalue</th>\n",
       "      <th>chi_square_statistic</th>\n",
       "      <th>effect_size</th>\n",
       "      <th>mapping_error_rate</th>\n",
       "      <th>ref_allele_mapping_bias</th>\n",
       "      <th>overdispersion</th>\n",
       "      <th>snp_id_within_region</th>\n",
       "      <th>num_feature_snps</th>\n",
       "      <th>num_testing_snps</th>\n",
       "      <th>num_iterations_null_hypothesis</th>\n",
       "      <th>null_iterations_alt_hypothesis</th>\n",
       "      <th>random_location_of_ties</th>\n",
       "      <th>log_likelihood_null_hypothesis</th>\n",
       "      <th>convergence_status</th>\n",
       "      <th>squared_correlation_fsnps</th>\n",
       "      <th>squared_correlation_rsnps</th>\n",
       "      <th>in_feature</th>\n",
       "      <th>region</th>\n",
       "      <th>raw_pvalue</th>\n",
       "      <th>cis_window_qvalue</th>\n",
       "      <th>null_cis_qvalue</th>\n",
       "      <th>genomewide_bh_qvalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr17_78431572_78433595</td>\n",
       "      <td>rs7212019</td>\n",
       "      <td>chr17</td>\n",
       "      <td>78432472</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.171875</td>\n",
       "      <td>1.378426</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-7.673398</td>\n",
       "      <td>34.864044</td>\n",
       "      <td>0.764919</td>\n",
       "      <td>0.054194</td>\n",
       "      <td>0.143244</td>\n",
       "      <td>2.808754</td>\n",
       "      <td>2.0</td>\n",
       "      <td>8</td>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>6</td>\n",
       "      <td>78432472</td>\n",
       "      <td>16.662091</td>\n",
       "      <td>0</td>\n",
       "      <td>0.993437</td>\n",
       "      <td>0.990011</td>\n",
       "      <td>True</td>\n",
       "      <td>CAUD</td>\n",
       "      <td>3.535502e-09</td>\n",
       "      <td>2.121301e-08</td>\n",
       "      <td>0.002443</td>\n",
       "      <td>0.000011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr17_78431572_78433595</td>\n",
       "      <td>rs11077365</td>\n",
       "      <td>chr17</td>\n",
       "      <td>78432412</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.937500</td>\n",
       "      <td>1.378426</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-5.686070</td>\n",
       "      <td>24.651913</td>\n",
       "      <td>0.778447</td>\n",
       "      <td>0.037254</td>\n",
       "      <td>0.127387</td>\n",
       "      <td>2.468079</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>78432472</td>\n",
       "      <td>11.668557</td>\n",
       "      <td>0</td>\n",
       "      <td>0.992228</td>\n",
       "      <td>0.981582</td>\n",
       "      <td>True</td>\n",
       "      <td>CAUD</td>\n",
       "      <td>6.867655e-07</td>\n",
       "      <td>2.060297e-06</td>\n",
       "      <td>0.005189</td>\n",
       "      <td>0.000398</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr17_45702992_45704386</td>\n",
       "      <td>rs17762308</td>\n",
       "      <td>chr17</td>\n",
       "      <td>45703582</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.359375</td>\n",
       "      <td>0.888889</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-5.638462</td>\n",
       "      <td>25.425502</td>\n",
       "      <td>0.719547</td>\n",
       "      <td>0.079023</td>\n",
       "      <td>0.338486</td>\n",
       "      <td>2.867342</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>45703582</td>\n",
       "      <td>11.654617</td>\n",
       "      <td>0</td>\n",
       "      <td>0.989611</td>\n",
       "      <td>0.986383</td>\n",
       "      <td>True</td>\n",
       "      <td>CAUD</td>\n",
       "      <td>4.597995e-07</td>\n",
       "      <td>2.298997e-06</td>\n",
       "      <td>0.005189</td>\n",
       "      <td>0.000398</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr17_45818390_45819780</td>\n",
       "      <td>rs62057109</td>\n",
       "      <td>chr17</td>\n",
       "      <td>45819271</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.218750</td>\n",
       "      <td>0.235102</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-5.436042</td>\n",
       "      <td>24.878265</td>\n",
       "      <td>0.705236</td>\n",
       "      <td>0.039329</td>\n",
       "      <td>0.295675</td>\n",
       "      <td>2.962826</td>\n",
       "      <td>4.0</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>8</td>\n",
       "      <td>9</td>\n",
       "      <td>45819271</td>\n",
       "      <td>11.402730</td>\n",
       "      <td>0</td>\n",
       "      <td>0.985955</td>\n",
       "      <td>0.993407</td>\n",
       "      <td>True</td>\n",
       "      <td>CAUD</td>\n",
       "      <td>6.106707e-07</td>\n",
       "      <td>3.664024e-06</td>\n",
       "      <td>0.005189</td>\n",
       "      <td>0.000475</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr17_78431572_78433595</td>\n",
       "      <td>rs692329</td>\n",
       "      <td>chr17</td>\n",
       "      <td>78431841</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.703125</td>\n",
       "      <td>2.492678</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-5.167449</td>\n",
       "      <td>21.576268</td>\n",
       "      <td>0.703714</td>\n",
       "      <td>0.059994</td>\n",
       "      <td>0.179475</td>\n",
       "      <td>2.233529</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>78432472</td>\n",
       "      <td>9.971584</td>\n",
       "      <td>0</td>\n",
       "      <td>0.993179</td>\n",
       "      <td>0.992335</td>\n",
       "      <td>True</td>\n",
       "      <td>CAUD</td>\n",
       "      <td>3.400333e-06</td>\n",
       "      <td>6.800666e-06</td>\n",
       "      <td>0.010112</td>\n",
       "      <td>0.000664</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   feature        rsid  chrom   snp_pos ref alt  allele_freq  \\\n",
       "0  chr17_78431572_78433595   rs7212019  chr17  78432472   G   A     0.171875   \n",
       "3  chr17_78431572_78433595  rs11077365  chr17  78432412   C   A     0.937500   \n",
       "1  chr17_45702992_45704386  rs17762308  chr17  45703582   T   C     0.359375   \n",
       "2  chr17_45818390_45819780  rs62057109  chr17  45819271   T   C     0.218750   \n",
       "5  chr17_78431572_78433595    rs692329  chr17  78431841   C   T     0.703125   \n",
       "\n",
       "   HWE_chi_square_statistic  imputation_quality_score  log10_bh_qvalue  \\\n",
       "0                  1.378426                       1.0        -7.673398   \n",
       "3                  1.378426                       1.0        -5.686070   \n",
       "1                  0.888889                       1.0        -5.638462   \n",
       "2                  0.235102                       1.0        -5.436042   \n",
       "5                  2.492678                       1.0        -5.167449   \n",
       "\n",
       "   chi_square_statistic  effect_size  mapping_error_rate  \\\n",
       "0             34.864044     0.764919            0.054194   \n",
       "3             24.651913     0.778447            0.037254   \n",
       "1             25.425502     0.719547            0.079023   \n",
       "2             24.878265     0.705236            0.039329   \n",
       "5             21.576268     0.703714            0.059994   \n",
       "\n",
       "   ref_allele_mapping_bias  overdispersion  snp_id_within_region  \\\n",
       "0                 0.143244        2.808754                   2.0   \n",
       "3                 0.127387        2.468079                   1.0   \n",
       "1                 0.338486        2.867342                   0.0   \n",
       "2                 0.295675        2.962826                   4.0   \n",
       "5                 0.179475        2.233529                   0.0   \n",
       "\n",
       "   num_feature_snps  num_testing_snps  num_iterations_null_hypothesis  \\\n",
       "0                 8                 6                               7   \n",
       "3                 8                 6                               6   \n",
       "1                 5                 5                               6   \n",
       "2                 6                 6                               8   \n",
       "5                 8                 6                               6   \n",
       "\n",
       "   null_iterations_alt_hypothesis  random_location_of_ties  \\\n",
       "0                               6                 78432472   \n",
       "3                               6                 78432472   \n",
       "1                               6                 45703582   \n",
       "2                               9                 45819271   \n",
       "5                               6                 78432472   \n",
       "\n",
       "   log_likelihood_null_hypothesis  convergence_status  \\\n",
       "0                       16.662091                   0   \n",
       "3                       11.668557                   0   \n",
       "1                       11.654617                   0   \n",
       "2                       11.402730                   0   \n",
       "5                        9.971584                   0   \n",
       "\n",
       "   squared_correlation_fsnps  squared_correlation_rsnps  in_feature region  \\\n",
       "0                   0.993437                   0.990011        True   CAUD   \n",
       "3                   0.992228                   0.981582        True   CAUD   \n",
       "1                   0.989611                   0.986383        True   CAUD   \n",
       "2                   0.985955                   0.993407        True   CAUD   \n",
       "5                   0.993179                   0.992335        True   CAUD   \n",
       "\n",
       "     raw_pvalue  cis_window_qvalue  null_cis_qvalue  genomewide_bh_qvalue  \n",
       "0  3.535502e-09       2.121301e-08         0.002443              0.000011  \n",
       "3  6.867655e-07       2.060297e-06         0.005189              0.000398  \n",
       "1  4.597995e-07       2.298997e-06         0.005189              0.000398  \n",
       "2  6.106707e-07       3.664024e-06         0.005189              0.000475  \n",
       "5  3.400333e-06       6.800666e-06         0.010112              0.000664  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "(519, 31)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "HIPP\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>feature</th>\n",
       "      <th>rsid</th>\n",
       "      <th>chrom</th>\n",
       "      <th>snp_pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>allele_freq</th>\n",
       "      <th>HWE_chi_square_statistic</th>\n",
       "      <th>imputation_quality_score</th>\n",
       "      <th>log10_bh_qvalue</th>\n",
       "      <th>chi_square_statistic</th>\n",
       "      <th>effect_size</th>\n",
       "      <th>mapping_error_rate</th>\n",
       "      <th>ref_allele_mapping_bias</th>\n",
       "      <th>overdispersion</th>\n",
       "      <th>snp_id_within_region</th>\n",
       "      <th>num_feature_snps</th>\n",
       "      <th>num_testing_snps</th>\n",
       "      <th>num_iterations_null_hypothesis</th>\n",
       "      <th>null_iterations_alt_hypothesis</th>\n",
       "      <th>random_location_of_ties</th>\n",
       "      <th>log_likelihood_null_hypothesis</th>\n",
       "      <th>convergence_status</th>\n",
       "      <th>squared_correlation_fsnps</th>\n",
       "      <th>squared_correlation_rsnps</th>\n",
       "      <th>in_feature</th>\n",
       "      <th>region</th>\n",
       "      <th>raw_pvalue</th>\n",
       "      <th>cis_window_qvalue</th>\n",
       "      <th>null_cis_qvalue</th>\n",
       "      <th>genomewide_bh_qvalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr17_78431572_78433595</td>\n",
       "      <td>rs7212019</td>\n",
       "      <td>chr17</td>\n",
       "      <td>78432472</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.181818</td>\n",
       "      <td>1.086420</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-8.792368</td>\n",
       "      <td>39.097073</td>\n",
       "      <td>0.791280</td>\n",
       "      <td>0.060590</td>\n",
       "      <td>0.161570</td>\n",
       "      <td>3.462957</td>\n",
       "      <td>2.0</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>78432472</td>\n",
       "      <td>18.747591</td>\n",
       "      <td>0</td>\n",
       "      <td>0.993585</td>\n",
       "      <td>0.991026</td>\n",
       "      <td>True</td>\n",
       "      <td>HIPP</td>\n",
       "      <td>4.032481e-10</td>\n",
       "      <td>1.612992e-09</td>\n",
       "      <td>0.001692</td>\n",
       "      <td>8.500469e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr17_78431572_78433595</td>\n",
       "      <td>rs691346</td>\n",
       "      <td>chr17</td>\n",
       "      <td>78433056</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.727273</td>\n",
       "      <td>3.414141</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-5.049101</td>\n",
       "      <td>21.053873</td>\n",
       "      <td>0.710199</td>\n",
       "      <td>0.069978</td>\n",
       "      <td>0.226901</td>\n",
       "      <td>2.324247</td>\n",
       "      <td>6.0</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>78432472</td>\n",
       "      <td>9.743852</td>\n",
       "      <td>0</td>\n",
       "      <td>0.993090</td>\n",
       "      <td>0.990593</td>\n",
       "      <td>True</td>\n",
       "      <td>HIPP</td>\n",
       "      <td>4.465493e-06</td>\n",
       "      <td>8.930985e-06</td>\n",
       "      <td>0.001692</td>\n",
       "      <td>2.353315e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr17_78431572_78433595</td>\n",
       "      <td>rs11077365</td>\n",
       "      <td>chr17</td>\n",
       "      <td>78432412</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.818182</td>\n",
       "      <td>1.086420</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-4.347109</td>\n",
       "      <td>17.195408</td>\n",
       "      <td>0.744396</td>\n",
       "      <td>0.045380</td>\n",
       "      <td>0.173180</td>\n",
       "      <td>2.657648</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>6</td>\n",
       "      <td>78432472</td>\n",
       "      <td>7.900580</td>\n",
       "      <td>0</td>\n",
       "      <td>0.990224</td>\n",
       "      <td>0.990035</td>\n",
       "      <td>True</td>\n",
       "      <td>HIPP</td>\n",
       "      <td>3.372506e-05</td>\n",
       "      <td>4.496674e-05</td>\n",
       "      <td>0.001692</td>\n",
       "      <td>7.899158e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr17_78431572_78433595</td>\n",
       "      <td>rs692329</td>\n",
       "      <td>chr17</td>\n",
       "      <td>78431841</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.750000</td>\n",
       "      <td>2.148438</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-3.366332</td>\n",
       "      <td>12.396246</td>\n",
       "      <td>0.673225</td>\n",
       "      <td>0.068775</td>\n",
       "      <td>0.231607</td>\n",
       "      <td>2.102971</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>78432472</td>\n",
       "      <td>5.513500</td>\n",
       "      <td>0</td>\n",
       "      <td>0.993048</td>\n",
       "      <td>0.993161</td>\n",
       "      <td>True</td>\n",
       "      <td>HIPP</td>\n",
       "      <td>4.301978e-04</td>\n",
       "      <td>4.301978e-04</td>\n",
       "      <td>0.001692</td>\n",
       "      <td>5.667855e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr7_65960407_65960696</td>\n",
       "      <td>rs62470932</td>\n",
       "      <td>chr7</td>\n",
       "      <td>65960530</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.340909</td>\n",
       "      <td>3.327160</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-2.922870</td>\n",
       "      <td>10.499070</td>\n",
       "      <td>0.306683</td>\n",
       "      <td>0.000059</td>\n",
       "      <td>0.595304</td>\n",
       "      <td>11.227217</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>65960530</td>\n",
       "      <td>4.589529</td>\n",
       "      <td>0</td>\n",
       "      <td>0.896464</td>\n",
       "      <td>0.896464</td>\n",
       "      <td>True</td>\n",
       "      <td>HIPP</td>\n",
       "      <td>1.194346e-03</td>\n",
       "      <td>1.194346e-03</td>\n",
       "      <td>0.003558</td>\n",
       "      <td>1.258841e-01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   feature        rsid  chrom   snp_pos ref alt  allele_freq  \\\n",
       "0  chr17_78431572_78433595   rs7212019  chr17  78432472   G   A     0.181818   \n",
       "1  chr17_78431572_78433595    rs691346  chr17  78433056   G   T     0.727273   \n",
       "2  chr17_78431572_78433595  rs11077365  chr17  78432412   C   A     0.818182   \n",
       "3  chr17_78431572_78433595    rs692329  chr17  78431841   C   T     0.750000   \n",
       "4   chr7_65960407_65960696  rs62470932   chr7  65960530   C   T     0.340909   \n",
       "\n",
       "   HWE_chi_square_statistic  imputation_quality_score  log10_bh_qvalue  \\\n",
       "0                  1.086420                       1.0        -8.792368   \n",
       "1                  3.414141                       1.0        -5.049101   \n",
       "2                  1.086420                       1.0        -4.347109   \n",
       "3                  2.148438                       1.0        -3.366332   \n",
       "4                  3.327160                       1.0        -2.922870   \n",
       "\n",
       "   chi_square_statistic  effect_size  mapping_error_rate  \\\n",
       "0             39.097073     0.791280            0.060590   \n",
       "1             21.053873     0.710199            0.069978   \n",
       "2             17.195408     0.744396            0.045380   \n",
       "3             12.396246     0.673225            0.068775   \n",
       "4             10.499070     0.306683            0.000059   \n",
       "\n",
       "   ref_allele_mapping_bias  overdispersion  snp_id_within_region  \\\n",
       "0                 0.161570        3.462957                   2.0   \n",
       "1                 0.226901        2.324247                   6.0   \n",
       "2                 0.173180        2.657648                   1.0   \n",
       "3                 0.231607        2.102971                   0.0   \n",
       "4                 0.595304       11.227217                   0.0   \n",
       "\n",
       "   num_feature_snps  num_testing_snps  num_iterations_null_hypothesis  \\\n",
       "0                 8                 4                               6   \n",
       "1                 8                 4                               6   \n",
       "2                 8                 4                               8   \n",
       "3                 8                 4                               6   \n",
       "4                 1                 1                               5   \n",
       "\n",
       "   null_iterations_alt_hypothesis  random_location_of_ties  \\\n",
       "0                               6                 78432472   \n",
       "1                               6                 78432472   \n",
       "2                               6                 78432472   \n",
       "3                               6                 78432472   \n",
       "4                               5                 65960530   \n",
       "\n",
       "   log_likelihood_null_hypothesis  convergence_status  \\\n",
       "0                       18.747591                   0   \n",
       "1                        9.743852                   0   \n",
       "2                        7.900580                   0   \n",
       "3                        5.513500                   0   \n",
       "4                        4.589529                   0   \n",
       "\n",
       "   squared_correlation_fsnps  squared_correlation_rsnps  in_feature region  \\\n",
       "0                   0.993585                   0.991026        True   HIPP   \n",
       "1                   0.993090                   0.990593        True   HIPP   \n",
       "2                   0.990224                   0.990035        True   HIPP   \n",
       "3                   0.993048                   0.993161        True   HIPP   \n",
       "4                   0.896464                   0.896464        True   HIPP   \n",
       "\n",
       "     raw_pvalue  cis_window_qvalue  null_cis_qvalue  genomewide_bh_qvalue  \n",
       "0  4.032481e-10       1.612992e-09         0.001692          8.500469e-07  \n",
       "1  4.465493e-06       8.930985e-06         0.001692          2.353315e-03  \n",
       "2  3.372506e-05       4.496674e-05         0.001692          7.899158e-03  \n",
       "3  4.301978e-04       4.301978e-04         0.001692          5.667855e-02  \n",
       "4  1.194346e-03       1.194346e-03         0.003558          1.258841e-01  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "(527, 31)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MDFG\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>feature</th>\n",
       "      <th>rsid</th>\n",
       "      <th>chrom</th>\n",
       "      <th>snp_pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>allele_freq</th>\n",
       "      <th>HWE_chi_square_statistic</th>\n",
       "      <th>imputation_quality_score</th>\n",
       "      <th>log10_bh_qvalue</th>\n",
       "      <th>chi_square_statistic</th>\n",
       "      <th>effect_size</th>\n",
       "      <th>mapping_error_rate</th>\n",
       "      <th>ref_allele_mapping_bias</th>\n",
       "      <th>overdispersion</th>\n",
       "      <th>snp_id_within_region</th>\n",
       "      <th>num_feature_snps</th>\n",
       "      <th>num_testing_snps</th>\n",
       "      <th>num_iterations_null_hypothesis</th>\n",
       "      <th>null_iterations_alt_hypothesis</th>\n",
       "      <th>random_location_of_ties</th>\n",
       "      <th>log_likelihood_null_hypothesis</th>\n",
       "      <th>convergence_status</th>\n",
       "      <th>squared_correlation_fsnps</th>\n",
       "      <th>squared_correlation_rsnps</th>\n",
       "      <th>in_feature</th>\n",
       "      <th>region</th>\n",
       "      <th>raw_pvalue</th>\n",
       "      <th>cis_window_qvalue</th>\n",
       "      <th>null_cis_qvalue</th>\n",
       "      <th>genomewide_bh_qvalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr8_22599221_22600284</td>\n",
       "      <td>rs755934</td>\n",
       "      <td>chr8</td>\n",
       "      <td>22599692</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.375000</td>\n",
       "      <td>0.148148</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-2.385009</td>\n",
       "      <td>8.229762</td>\n",
       "      <td>0.394382</td>\n",
       "      <td>0.000146</td>\n",
       "      <td>0.498644</td>\n",
       "      <td>40.121462</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>22599692</td>\n",
       "      <td>2.729497</td>\n",
       "      <td>0</td>\n",
       "      <td>0.996045</td>\n",
       "      <td>0.996045</td>\n",
       "      <td>True</td>\n",
       "      <td>MDFG</td>\n",
       "      <td>0.004121</td>\n",
       "      <td>0.004121</td>\n",
       "      <td>0.024660</td>\n",
       "      <td>0.190429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr8_22622374_22623929</td>\n",
       "      <td>rs11780207</td>\n",
       "      <td>chr8</td>\n",
       "      <td>22623006</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.458333</td>\n",
       "      <td>2.958189</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-2.352835</td>\n",
       "      <td>9.359037</td>\n",
       "      <td>0.392856</td>\n",
       "      <td>0.000048</td>\n",
       "      <td>0.494381</td>\n",
       "      <td>29.332559</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>22623006</td>\n",
       "      <td>3.137108</td>\n",
       "      <td>0</td>\n",
       "      <td>0.996551</td>\n",
       "      <td>0.997301</td>\n",
       "      <td>True</td>\n",
       "      <td>MDFG</td>\n",
       "      <td>0.002219</td>\n",
       "      <td>0.004438</td>\n",
       "      <td>0.017543</td>\n",
       "      <td>0.190429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr17_45825137_45827395</td>\n",
       "      <td>rs62057154</td>\n",
       "      <td>chr17</td>\n",
       "      <td>45827244</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>4.687500</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-2.225874</td>\n",
       "      <td>8.504661</td>\n",
       "      <td>0.300795</td>\n",
       "      <td>0.042145</td>\n",
       "      <td>0.321162</td>\n",
       "      <td>6.965365</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>45827244</td>\n",
       "      <td>3.412369</td>\n",
       "      <td>0</td>\n",
       "      <td>0.991781</td>\n",
       "      <td>0.994368</td>\n",
       "      <td>True</td>\n",
       "      <td>MDFG</td>\n",
       "      <td>0.003542</td>\n",
       "      <td>0.005945</td>\n",
       "      <td>0.024660</td>\n",
       "      <td>0.190429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr17_45825137_45827395</td>\n",
       "      <td>rs62057148</td>\n",
       "      <td>chr17</td>\n",
       "      <td>45826119</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.875000</td>\n",
       "      <td>7.647765</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-2.225874</td>\n",
       "      <td>8.300645</td>\n",
       "      <td>0.309707</td>\n",
       "      <td>0.042193</td>\n",
       "      <td>0.321536</td>\n",
       "      <td>5.765476</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>45827244</td>\n",
       "      <td>3.386116</td>\n",
       "      <td>0</td>\n",
       "      <td>0.991024</td>\n",
       "      <td>0.987951</td>\n",
       "      <td>True</td>\n",
       "      <td>MDFG</td>\n",
       "      <td>0.003963</td>\n",
       "      <td>0.005945</td>\n",
       "      <td>0.024660</td>\n",
       "      <td>0.190429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr11_86067972_86070115</td>\n",
       "      <td>rs2458500</td>\n",
       "      <td>chr11</td>\n",
       "      <td>86068268</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.833333</td>\n",
       "      <td>0.480000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-2.112704</td>\n",
       "      <td>7.615858</td>\n",
       "      <td>0.662747</td>\n",
       "      <td>0.002169</td>\n",
       "      <td>0.343247</td>\n",
       "      <td>11.983433</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>86068255</td>\n",
       "      <td>2.791775</td>\n",
       "      <td>0</td>\n",
       "      <td>0.685511</td>\n",
       "      <td>0.678352</td>\n",
       "      <td>True</td>\n",
       "      <td>MDFG</td>\n",
       "      <td>0.005786</td>\n",
       "      <td>0.007714</td>\n",
       "      <td>0.088001</td>\n",
       "      <td>0.190429</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    feature        rsid  chrom   snp_pos ref alt  allele_freq  \\\n",
       "3    chr8_22599221_22600284    rs755934   chr8  22599692   T   G     0.375000   \n",
       "0    chr8_22622374_22623929  rs11780207   chr8  22623006   G   A     0.458333   \n",
       "1   chr17_45825137_45827395  rs62057154  chr17  45827244   C   T     0.333333   \n",
       "2   chr17_45825137_45827395  rs62057148  chr17  45826119   G   C     0.875000   \n",
       "18  chr11_86067972_86070115   rs2458500  chr11  86068268   A   G     0.833333   \n",
       "\n",
       "    HWE_chi_square_statistic  imputation_quality_score  log10_bh_qvalue  \\\n",
       "3                   0.148148                       1.0        -2.385009   \n",
       "0                   2.958189                       1.0        -2.352835   \n",
       "1                   4.687500                       1.0        -2.225874   \n",
       "2                   7.647765                       1.0        -2.225874   \n",
       "18                  0.480000                       1.0        -2.112704   \n",
       "\n",
       "    chi_square_statistic  effect_size  mapping_error_rate  \\\n",
       "3               8.229762     0.394382            0.000146   \n",
       "0               9.359037     0.392856            0.000048   \n",
       "1               8.504661     0.300795            0.042145   \n",
       "2               8.300645     0.309707            0.042193   \n",
       "18              7.615858     0.662747            0.002169   \n",
       "\n",
       "    ref_allele_mapping_bias  overdispersion  snp_id_within_region  \\\n",
       "3                  0.498644       40.121462                   0.0   \n",
       "0                  0.494381       29.332559                   1.0   \n",
       "1                  0.321162        6.965365                   1.0   \n",
       "2                  0.321536        5.765476                   0.0   \n",
       "18                 0.343247       11.983433                   2.0   \n",
       "\n",
       "    num_feature_snps  num_testing_snps  num_iterations_null_hypothesis  \\\n",
       "3                  1                 1                               5   \n",
       "0                  2                 2                               5   \n",
       "1                  3                 3                               6   \n",
       "2                  3                 3                               5   \n",
       "18                 4                 4                               7   \n",
       "\n",
       "    null_iterations_alt_hypothesis  random_location_of_ties  \\\n",
       "3                                5                 22599692   \n",
       "0                                5                 22623006   \n",
       "1                                5                 45827244   \n",
       "2                                5                 45827244   \n",
       "18                               7                 86068255   \n",
       "\n",
       "    log_likelihood_null_hypothesis  convergence_status  \\\n",
       "3                         2.729497                   0   \n",
       "0                         3.137108                   0   \n",
       "1                         3.412369                   0   \n",
       "2                         3.386116                   0   \n",
       "18                        2.791775                   0   \n",
       "\n",
       "    squared_correlation_fsnps  squared_correlation_rsnps  in_feature region  \\\n",
       "3                    0.996045                   0.996045        True   MDFG   \n",
       "0                    0.996551                   0.997301        True   MDFG   \n",
       "1                    0.991781                   0.994368        True   MDFG   \n",
       "2                    0.991024                   0.987951        True   MDFG   \n",
       "18                   0.685511                   0.678352        True   MDFG   \n",
       "\n",
       "    raw_pvalue  cis_window_qvalue  null_cis_qvalue  genomewide_bh_qvalue  \n",
       "3     0.004121           0.004121         0.024660              0.190429  \n",
       "0     0.002219           0.004438         0.017543              0.190429  \n",
       "1     0.003542           0.005945         0.024660              0.190429  \n",
       "2     0.003963           0.005945         0.024660              0.190429  \n",
       "18    0.005786           0.007714         0.088001              0.190429  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "(553, 31)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MDTG\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>feature</th>\n",
       "      <th>rsid</th>\n",
       "      <th>chrom</th>\n",
       "      <th>snp_pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>allele_freq</th>\n",
       "      <th>HWE_chi_square_statistic</th>\n",
       "      <th>imputation_quality_score</th>\n",
       "      <th>log10_bh_qvalue</th>\n",
       "      <th>chi_square_statistic</th>\n",
       "      <th>effect_size</th>\n",
       "      <th>mapping_error_rate</th>\n",
       "      <th>ref_allele_mapping_bias</th>\n",
       "      <th>overdispersion</th>\n",
       "      <th>snp_id_within_region</th>\n",
       "      <th>num_feature_snps</th>\n",
       "      <th>num_testing_snps</th>\n",
       "      <th>num_iterations_null_hypothesis</th>\n",
       "      <th>null_iterations_alt_hypothesis</th>\n",
       "      <th>random_location_of_ties</th>\n",
       "      <th>log_likelihood_null_hypothesis</th>\n",
       "      <th>convergence_status</th>\n",
       "      <th>squared_correlation_fsnps</th>\n",
       "      <th>squared_correlation_rsnps</th>\n",
       "      <th>in_feature</th>\n",
       "      <th>region</th>\n",
       "      <th>raw_pvalue</th>\n",
       "      <th>cis_window_qvalue</th>\n",
       "      <th>null_cis_qvalue</th>\n",
       "      <th>genomewide_bh_qvalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr11_60251518_60251763</td>\n",
       "      <td>rs636341</td>\n",
       "      <td>chr11</td>\n",
       "      <td>60251688</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.750000</td>\n",
       "      <td>2.958189</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-4.614461</td>\n",
       "      <td>17.831440</td>\n",
       "      <td>0.287161</td>\n",
       "      <td>0.000079</td>\n",
       "      <td>0.377744</td>\n",
       "      <td>19.795544</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>60251688</td>\n",
       "      <td>7.626432</td>\n",
       "      <td>0</td>\n",
       "      <td>0.996875</td>\n",
       "      <td>0.997271</td>\n",
       "      <td>True</td>\n",
       "      <td>MDTG</td>\n",
       "      <td>0.000024</td>\n",
       "      <td>0.000024</td>\n",
       "      <td>0.029677</td>\n",
       "      <td>0.004054</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr11_60251518_60251763</td>\n",
       "      <td>rs636317</td>\n",
       "      <td>chr11</td>\n",
       "      <td>60251677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.750000</td>\n",
       "      <td>2.958189</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-4.614461</td>\n",
       "      <td>17.818867</td>\n",
       "      <td>0.287154</td>\n",
       "      <td>0.000079</td>\n",
       "      <td>0.377505</td>\n",
       "      <td>19.789926</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>60251688</td>\n",
       "      <td>7.620118</td>\n",
       "      <td>0</td>\n",
       "      <td>0.997225</td>\n",
       "      <td>0.997193</td>\n",
       "      <td>True</td>\n",
       "      <td>MDTG</td>\n",
       "      <td>0.000024</td>\n",
       "      <td>0.000024</td>\n",
       "      <td>0.028981</td>\n",
       "      <td>0.004054</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr17_78431572_78433595</td>\n",
       "      <td>rs7212019</td>\n",
       "      <td>chr17</td>\n",
       "      <td>78432472</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.166667</td>\n",
       "      <td>0.480000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-4.157169</td>\n",
       "      <td>17.630667</td>\n",
       "      <td>0.723166</td>\n",
       "      <td>0.036605</td>\n",
       "      <td>0.183028</td>\n",
       "      <td>3.364151</td>\n",
       "      <td>6.0</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>78432472</td>\n",
       "      <td>8.120604</td>\n",
       "      <td>0</td>\n",
       "      <td>0.992652</td>\n",
       "      <td>0.990770</td>\n",
       "      <td>True</td>\n",
       "      <td>MDTG</td>\n",
       "      <td>0.000027</td>\n",
       "      <td>0.000070</td>\n",
       "      <td>0.028981</td>\n",
       "      <td>0.004054</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr17_78431572_78433595</td>\n",
       "      <td>rs691362</td>\n",
       "      <td>chr17</td>\n",
       "      <td>78432944</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.166667</td>\n",
       "      <td>0.480000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-4.157169</td>\n",
       "      <td>17.627564</td>\n",
       "      <td>0.723161</td>\n",
       "      <td>0.036603</td>\n",
       "      <td>0.183028</td>\n",
       "      <td>3.363916</td>\n",
       "      <td>9.0</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>78432472</td>\n",
       "      <td>8.119062</td>\n",
       "      <td>0</td>\n",
       "      <td>0.992666</td>\n",
       "      <td>0.990261</td>\n",
       "      <td>True</td>\n",
       "      <td>MDTG</td>\n",
       "      <td>0.000027</td>\n",
       "      <td>0.000070</td>\n",
       "      <td>0.028981</td>\n",
       "      <td>0.004054</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr17_78431572_78433595</td>\n",
       "      <td>rs72907463</td>\n",
       "      <td>chr17</td>\n",
       "      <td>78432579</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.166667</td>\n",
       "      <td>0.480000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-4.157169</td>\n",
       "      <td>17.627417</td>\n",
       "      <td>0.723160</td>\n",
       "      <td>0.036602</td>\n",
       "      <td>0.183030</td>\n",
       "      <td>3.363953</td>\n",
       "      <td>8.0</td>\n",
       "      <td>18</td>\n",
       "      <td>18</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>78432472</td>\n",
       "      <td>8.119033</td>\n",
       "      <td>0</td>\n",
       "      <td>0.992666</td>\n",
       "      <td>0.990462</td>\n",
       "      <td>True</td>\n",
       "      <td>MDTG</td>\n",
       "      <td>0.000027</td>\n",
       "      <td>0.000070</td>\n",
       "      <td>0.028981</td>\n",
       "      <td>0.004054</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   feature        rsid  chrom   snp_pos ref alt  allele_freq  \\\n",
       "0  chr11_60251518_60251763    rs636341  chr11  60251688   A   C     0.750000   \n",
       "1  chr11_60251518_60251763    rs636317  chr11  60251677   C   T     0.750000   \n",
       "2  chr17_78431572_78433595   rs7212019  chr17  78432472   G   A     0.166667   \n",
       "3  chr17_78431572_78433595    rs691362  chr17  78432944   T   C     0.166667   \n",
       "4  chr17_78431572_78433595  rs72907463  chr17  78432579   G   A     0.166667   \n",
       "\n",
       "   HWE_chi_square_statistic  imputation_quality_score  log10_bh_qvalue  \\\n",
       "0                  2.958189                       1.0        -4.614461   \n",
       "1                  2.958189                       1.0        -4.614461   \n",
       "2                  0.480000                       1.0        -4.157169   \n",
       "3                  0.480000                       1.0        -4.157169   \n",
       "4                  0.480000                       1.0        -4.157169   \n",
       "\n",
       "   chi_square_statistic  effect_size  mapping_error_rate  \\\n",
       "0             17.831440     0.287161            0.000079   \n",
       "1             17.818867     0.287154            0.000079   \n",
       "2             17.630667     0.723166            0.036605   \n",
       "3             17.627564     0.723161            0.036603   \n",
       "4             17.627417     0.723160            0.036602   \n",
       "\n",
       "   ref_allele_mapping_bias  overdispersion  snp_id_within_region  \\\n",
       "0                 0.377744       19.795544                   1.0   \n",
       "1                 0.377505       19.789926                   0.0   \n",
       "2                 0.183028        3.364151                   6.0   \n",
       "3                 0.183028        3.363916                   9.0   \n",
       "4                 0.183030        3.363953                   8.0   \n",
       "\n",
       "   num_feature_snps  num_testing_snps  num_iterations_null_hypothesis  \\\n",
       "0                 2                 2                               5   \n",
       "1                 2                 2                               4   \n",
       "2                18                18                               5   \n",
       "3                18                18                               5   \n",
       "4                18                18                               5   \n",
       "\n",
       "   null_iterations_alt_hypothesis  random_location_of_ties  \\\n",
       "0                               4                 60251688   \n",
       "1                               4                 60251688   \n",
       "2                               5                 78432472   \n",
       "3                               5                 78432472   \n",
       "4                               5                 78432472   \n",
       "\n",
       "   log_likelihood_null_hypothesis  convergence_status  \\\n",
       "0                        7.626432                   0   \n",
       "1                        7.620118                   0   \n",
       "2                        8.120604                   0   \n",
       "3                        8.119062                   0   \n",
       "4                        8.119033                   0   \n",
       "\n",
       "   squared_correlation_fsnps  squared_correlation_rsnps  in_feature region  \\\n",
       "0                   0.996875                   0.997271        True   MDTG   \n",
       "1                   0.997225                   0.997193        True   MDTG   \n",
       "2                   0.992652                   0.990770        True   MDTG   \n",
       "3                   0.992666                   0.990261        True   MDTG   \n",
       "4                   0.992666                   0.990462        True   MDTG   \n",
       "\n",
       "   raw_pvalue  cis_window_qvalue  null_cis_qvalue  genomewide_bh_qvalue  \n",
       "0    0.000024           0.000024         0.029677              0.004054  \n",
       "1    0.000024           0.000024         0.028981              0.004054  \n",
       "2    0.000027           0.000070         0.028981              0.004054  \n",
       "3    0.000027           0.000070         0.028981              0.004054  \n",
       "4    0.000027           0.000070         0.028981              0.004054  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "(524, 31)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PTMN\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>feature</th>\n",
       "      <th>rsid</th>\n",
       "      <th>chrom</th>\n",
       "      <th>snp_pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>allele_freq</th>\n",
       "      <th>HWE_chi_square_statistic</th>\n",
       "      <th>imputation_quality_score</th>\n",
       "      <th>log10_bh_qvalue</th>\n",
       "      <th>chi_square_statistic</th>\n",
       "      <th>effect_size</th>\n",
       "      <th>mapping_error_rate</th>\n",
       "      <th>ref_allele_mapping_bias</th>\n",
       "      <th>overdispersion</th>\n",
       "      <th>snp_id_within_region</th>\n",
       "      <th>num_feature_snps</th>\n",
       "      <th>num_testing_snps</th>\n",
       "      <th>num_iterations_null_hypothesis</th>\n",
       "      <th>null_iterations_alt_hypothesis</th>\n",
       "      <th>random_location_of_ties</th>\n",
       "      <th>log_likelihood_null_hypothesis</th>\n",
       "      <th>convergence_status</th>\n",
       "      <th>squared_correlation_fsnps</th>\n",
       "      <th>squared_correlation_rsnps</th>\n",
       "      <th>in_feature</th>\n",
       "      <th>region</th>\n",
       "      <th>raw_pvalue</th>\n",
       "      <th>cis_window_qvalue</th>\n",
       "      <th>null_cis_qvalue</th>\n",
       "      <th>genomewide_bh_qvalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr11_60251518_60251763</td>\n",
       "      <td>rs636317</td>\n",
       "      <td>chr11</td>\n",
       "      <td>60251677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.818182</td>\n",
       "      <td>2.272727</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-4.343199</td>\n",
       "      <td>16.632233</td>\n",
       "      <td>0.251451</td>\n",
       "      <td>0.000382</td>\n",
       "      <td>0.482460</td>\n",
       "      <td>21.296903</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>60251677</td>\n",
       "      <td>7.212099</td>\n",
       "      <td>0</td>\n",
       "      <td>0.996979</td>\n",
       "      <td>0.996979</td>\n",
       "      <td>True</td>\n",
       "      <td>PTMN</td>\n",
       "      <td>0.000045</td>\n",
       "      <td>0.000045</td>\n",
       "      <td>0.030643</td>\n",
       "      <td>0.012513</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr17_45825137_45827395</td>\n",
       "      <td>rs62057154</td>\n",
       "      <td>chr17</td>\n",
       "      <td>45827244</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.318182</td>\n",
       "      <td>1.517551</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-3.997813</td>\n",
       "      <td>16.552042</td>\n",
       "      <td>0.236428</td>\n",
       "      <td>0.023228</td>\n",
       "      <td>0.425026</td>\n",
       "      <td>8.262196</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>45827244</td>\n",
       "      <td>7.408472</td>\n",
       "      <td>0</td>\n",
       "      <td>0.992381</td>\n",
       "      <td>0.994588</td>\n",
       "      <td>True</td>\n",
       "      <td>PTMN</td>\n",
       "      <td>0.000047</td>\n",
       "      <td>0.000101</td>\n",
       "      <td>0.016011</td>\n",
       "      <td>0.012513</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr17_45825137_45827395</td>\n",
       "      <td>rs62057147</td>\n",
       "      <td>chr17</td>\n",
       "      <td>45825932</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.909091</td>\n",
       "      <td>3.227431</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-3.997813</td>\n",
       "      <td>15.781771</td>\n",
       "      <td>0.242174</td>\n",
       "      <td>0.024419</td>\n",
       "      <td>0.412235</td>\n",
       "      <td>8.062102</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>45827244</td>\n",
       "      <td>7.022824</td>\n",
       "      <td>0</td>\n",
       "      <td>0.991537</td>\n",
       "      <td>0.996260</td>\n",
       "      <td>True</td>\n",
       "      <td>PTMN</td>\n",
       "      <td>0.000071</td>\n",
       "      <td>0.000101</td>\n",
       "      <td>0.030718</td>\n",
       "      <td>0.012513</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr17_45825137_45827395</td>\n",
       "      <td>rs1912151</td>\n",
       "      <td>chr17</td>\n",
       "      <td>45825578</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.909091</td>\n",
       "      <td>3.227431</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-3.997813</td>\n",
       "      <td>15.670806</td>\n",
       "      <td>0.242677</td>\n",
       "      <td>0.024599</td>\n",
       "      <td>0.413606</td>\n",
       "      <td>8.004409</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>45827244</td>\n",
       "      <td>6.974272</td>\n",
       "      <td>0</td>\n",
       "      <td>0.993062</td>\n",
       "      <td>0.990183</td>\n",
       "      <td>True</td>\n",
       "      <td>PTMN</td>\n",
       "      <td>0.000075</td>\n",
       "      <td>0.000101</td>\n",
       "      <td>0.030718</td>\n",
       "      <td>0.012513</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr17_45639355_45639798</td>\n",
       "      <td>rs413778</td>\n",
       "      <td>chr17</td>\n",
       "      <td>45639519</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.727273</td>\n",
       "      <td>2.090072</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-3.751973</td>\n",
       "      <td>14.060507</td>\n",
       "      <td>0.748280</td>\n",
       "      <td>0.016716</td>\n",
       "      <td>0.514541</td>\n",
       "      <td>15.421235</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>45639519</td>\n",
       "      <td>5.765139</td>\n",
       "      <td>0</td>\n",
       "      <td>0.987050</td>\n",
       "      <td>0.987050</td>\n",
       "      <td>True</td>\n",
       "      <td>PTMN</td>\n",
       "      <td>0.000177</td>\n",
       "      <td>0.000177</td>\n",
       "      <td>0.030718</td>\n",
       "      <td>0.017631</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   feature        rsid  chrom   snp_pos ref alt  allele_freq  \\\n",
       "0  chr11_60251518_60251763    rs636317  chr11  60251677   C   T     0.818182   \n",
       "1  chr17_45825137_45827395  rs62057154  chr17  45827244   C   T     0.318182   \n",
       "2  chr17_45825137_45827395  rs62057147  chr17  45825932   G   A     0.909091   \n",
       "3  chr17_45825137_45827395   rs1912151  chr17  45825578   C   T     0.909091   \n",
       "4  chr17_45639355_45639798    rs413778  chr17  45639519   A   G     0.727273   \n",
       "\n",
       "   HWE_chi_square_statistic  imputation_quality_score  log10_bh_qvalue  \\\n",
       "0                  2.272727                       1.0        -4.343199   \n",
       "1                  1.517551                       1.0        -3.997813   \n",
       "2                  3.227431                       1.0        -3.997813   \n",
       "3                  3.227431                       1.0        -3.997813   \n",
       "4                  2.090072                       1.0        -3.751973   \n",
       "\n",
       "   chi_square_statistic  effect_size  mapping_error_rate  \\\n",
       "0             16.632233     0.251451            0.000382   \n",
       "1             16.552042     0.236428            0.023228   \n",
       "2             15.781771     0.242174            0.024419   \n",
       "3             15.670806     0.242677            0.024599   \n",
       "4             14.060507     0.748280            0.016716   \n",
       "\n",
       "   ref_allele_mapping_bias  overdispersion  snp_id_within_region  \\\n",
       "0                 0.482460       21.296903                   0.0   \n",
       "1                 0.425026        8.262196                   4.0   \n",
       "2                 0.412235        8.062102                   1.0   \n",
       "3                 0.413606        8.004409                   0.0   \n",
       "4                 0.514541       15.421235                   0.0   \n",
       "\n",
       "   num_feature_snps  num_testing_snps  num_iterations_null_hypothesis  \\\n",
       "0                 1                 1                               5   \n",
       "1                 5                 4                               6   \n",
       "2                 5                 4                               6   \n",
       "3                 5                 4                               6   \n",
       "4                 1                 1                               6   \n",
       "\n",
       "   null_iterations_alt_hypothesis  random_location_of_ties  \\\n",
       "0                               5                 60251677   \n",
       "1                               6                 45827244   \n",
       "2                               6                 45827244   \n",
       "3                               6                 45827244   \n",
       "4                               6                 45639519   \n",
       "\n",
       "   log_likelihood_null_hypothesis  convergence_status  \\\n",
       "0                        7.212099                   0   \n",
       "1                        7.408472                   0   \n",
       "2                        7.022824                   0   \n",
       "3                        6.974272                   0   \n",
       "4                        5.765139                   0   \n",
       "\n",
       "   squared_correlation_fsnps  squared_correlation_rsnps  in_feature region  \\\n",
       "0                   0.996979                   0.996979        True   PTMN   \n",
       "1                   0.992381                   0.994588        True   PTMN   \n",
       "2                   0.991537                   0.996260        True   PTMN   \n",
       "3                   0.993062                   0.990183        True   PTMN   \n",
       "4                   0.987050                   0.987050        True   PTMN   \n",
       "\n",
       "   raw_pvalue  cis_window_qvalue  null_cis_qvalue  genomewide_bh_qvalue  \n",
       "0    0.000045           0.000045         0.030643              0.012513  \n",
       "1    0.000047           0.000101         0.016011              0.012513  \n",
       "2    0.000071           0.000101         0.030718              0.012513  \n",
       "3    0.000075           0.000101         0.030718              0.012513  \n",
       "4    0.000177           0.000177         0.030718              0.017631  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "(498, 31)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SUNI\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>feature</th>\n",
       "      <th>rsid</th>\n",
       "      <th>chrom</th>\n",
       "      <th>snp_pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>allele_freq</th>\n",
       "      <th>HWE_chi_square_statistic</th>\n",
       "      <th>imputation_quality_score</th>\n",
       "      <th>log10_bh_qvalue</th>\n",
       "      <th>chi_square_statistic</th>\n",
       "      <th>effect_size</th>\n",
       "      <th>mapping_error_rate</th>\n",
       "      <th>ref_allele_mapping_bias</th>\n",
       "      <th>overdispersion</th>\n",
       "      <th>snp_id_within_region</th>\n",
       "      <th>num_feature_snps</th>\n",
       "      <th>num_testing_snps</th>\n",
       "      <th>num_iterations_null_hypothesis</th>\n",
       "      <th>null_iterations_alt_hypothesis</th>\n",
       "      <th>random_location_of_ties</th>\n",
       "      <th>log_likelihood_null_hypothesis</th>\n",
       "      <th>convergence_status</th>\n",
       "      <th>squared_correlation_fsnps</th>\n",
       "      <th>squared_correlation_rsnps</th>\n",
       "      <th>in_feature</th>\n",
       "      <th>region</th>\n",
       "      <th>raw_pvalue</th>\n",
       "      <th>cis_window_qvalue</th>\n",
       "      <th>null_cis_qvalue</th>\n",
       "      <th>genomewide_bh_qvalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr17_45894052_45896463</td>\n",
       "      <td>rs11575895</td>\n",
       "      <td>chr17</td>\n",
       "      <td>45894419</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.357143</td>\n",
       "      <td>0.062222</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-2.561455</td>\n",
       "      <td>11.480721</td>\n",
       "      <td>0.671862</td>\n",
       "      <td>0.010848</td>\n",
       "      <td>0.301631</td>\n",
       "      <td>7.377854</td>\n",
       "      <td>3.0</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>9</td>\n",
       "      <td>10</td>\n",
       "      <td>45894419</td>\n",
       "      <td>4.588737</td>\n",
       "      <td>0</td>\n",
       "      <td>0.978662</td>\n",
       "      <td>0.869372</td>\n",
       "      <td>True</td>\n",
       "      <td>SUNI</td>\n",
       "      <td>0.000703</td>\n",
       "      <td>0.002745</td>\n",
       "      <td>0.043506</td>\n",
       "      <td>0.099234</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr17_45894052_45896463</td>\n",
       "      <td>rs62056779</td>\n",
       "      <td>chr17</td>\n",
       "      <td>45894571</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.357143</td>\n",
       "      <td>0.062222</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-2.561455</td>\n",
       "      <td>11.467030</td>\n",
       "      <td>0.671855</td>\n",
       "      <td>0.010850</td>\n",
       "      <td>0.301618</td>\n",
       "      <td>7.376435</td>\n",
       "      <td>4.0</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>9</td>\n",
       "      <td>10</td>\n",
       "      <td>45894419</td>\n",
       "      <td>4.582022</td>\n",
       "      <td>0</td>\n",
       "      <td>0.978731</td>\n",
       "      <td>0.869331</td>\n",
       "      <td>True</td>\n",
       "      <td>SUNI</td>\n",
       "      <td>0.000708</td>\n",
       "      <td>0.002745</td>\n",
       "      <td>0.034931</td>\n",
       "      <td>0.099234</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr17_45894052_45896463</td>\n",
       "      <td>rs80233201</td>\n",
       "      <td>chr17</td>\n",
       "      <td>45894115</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.357143</td>\n",
       "      <td>0.062222</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-2.561455</td>\n",
       "      <td>10.719334</td>\n",
       "      <td>0.674705</td>\n",
       "      <td>0.010953</td>\n",
       "      <td>0.298847</td>\n",
       "      <td>7.208121</td>\n",
       "      <td>1.0</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>9</td>\n",
       "      <td>10</td>\n",
       "      <td>45894419</td>\n",
       "      <td>4.214025</td>\n",
       "      <td>0</td>\n",
       "      <td>0.978907</td>\n",
       "      <td>0.869343</td>\n",
       "      <td>True</td>\n",
       "      <td>SUNI</td>\n",
       "      <td>0.001060</td>\n",
       "      <td>0.002745</td>\n",
       "      <td>0.043506</td>\n",
       "      <td>0.099234</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr17_45894052_45896463</td>\n",
       "      <td>rs111972148</td>\n",
       "      <td>chr17</td>\n",
       "      <td>45895755</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.357143</td>\n",
       "      <td>0.062222</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-2.561455</td>\n",
       "      <td>10.718434</td>\n",
       "      <td>0.674692</td>\n",
       "      <td>0.010955</td>\n",
       "      <td>0.298866</td>\n",
       "      <td>7.209797</td>\n",
       "      <td>8.0</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>9</td>\n",
       "      <td>10</td>\n",
       "      <td>45894419</td>\n",
       "      <td>4.213470</td>\n",
       "      <td>0</td>\n",
       "      <td>0.978911</td>\n",
       "      <td>0.868927</td>\n",
       "      <td>True</td>\n",
       "      <td>SUNI</td>\n",
       "      <td>0.001061</td>\n",
       "      <td>0.002745</td>\n",
       "      <td>0.047231</td>\n",
       "      <td>0.099234</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr17_45894052_45896463</td>\n",
       "      <td>rs74548327</td>\n",
       "      <td>chr17</td>\n",
       "      <td>45895714</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.642857</td>\n",
       "      <td>0.884383</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-2.561455</td>\n",
       "      <td>10.297340</td>\n",
       "      <td>0.666734</td>\n",
       "      <td>0.011023</td>\n",
       "      <td>0.315520</td>\n",
       "      <td>6.779840</td>\n",
       "      <td>7.0</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>10</td>\n",
       "      <td>10</td>\n",
       "      <td>45894419</td>\n",
       "      <td>3.985714</td>\n",
       "      <td>0</td>\n",
       "      <td>0.977652</td>\n",
       "      <td>0.887832</td>\n",
       "      <td>True</td>\n",
       "      <td>SUNI</td>\n",
       "      <td>0.001332</td>\n",
       "      <td>0.002745</td>\n",
       "      <td>0.047231</td>\n",
       "      <td>0.099234</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   feature         rsid  chrom   snp_pos ref alt  allele_freq  \\\n",
       "1  chr17_45894052_45896463   rs11575895  chr17  45894419   A   G     0.357143   \n",
       "2  chr17_45894052_45896463   rs62056779  chr17  45894571   C   A     0.357143   \n",
       "3  chr17_45894052_45896463   rs80233201  chr17  45894115   T   C     0.357143   \n",
       "4  chr17_45894052_45896463  rs111972148  chr17  45895755   G   C     0.357143   \n",
       "5  chr17_45894052_45896463   rs74548327  chr17  45895714   A   G     0.642857   \n",
       "\n",
       "   HWE_chi_square_statistic  imputation_quality_score  log10_bh_qvalue  \\\n",
       "1                  0.062222                       1.0        -2.561455   \n",
       "2                  0.062222                       1.0        -2.561455   \n",
       "3                  0.062222                       1.0        -2.561455   \n",
       "4                  0.062222                       1.0        -2.561455   \n",
       "5                  0.884383                       1.0        -2.561455   \n",
       "\n",
       "   chi_square_statistic  effect_size  mapping_error_rate  \\\n",
       "1             11.480721     0.671862            0.010848   \n",
       "2             11.467030     0.671855            0.010850   \n",
       "3             10.719334     0.674705            0.010953   \n",
       "4             10.718434     0.674692            0.010955   \n",
       "5             10.297340     0.666734            0.011023   \n",
       "\n",
       "   ref_allele_mapping_bias  overdispersion  snp_id_within_region  \\\n",
       "1                 0.301631        7.377854                   3.0   \n",
       "2                 0.301618        7.376435                   4.0   \n",
       "3                 0.298847        7.208121                   1.0   \n",
       "4                 0.298866        7.209797                   8.0   \n",
       "5                 0.315520        6.779840                   7.0   \n",
       "\n",
       "   num_feature_snps  num_testing_snps  num_iterations_null_hypothesis  \\\n",
       "1                13                13                               9   \n",
       "2                13                13                               9   \n",
       "3                13                13                               9   \n",
       "4                13                13                               9   \n",
       "5                13                13                              10   \n",
       "\n",
       "   null_iterations_alt_hypothesis  random_location_of_ties  \\\n",
       "1                              10                 45894419   \n",
       "2                              10                 45894419   \n",
       "3                              10                 45894419   \n",
       "4                              10                 45894419   \n",
       "5                              10                 45894419   \n",
       "\n",
       "   log_likelihood_null_hypothesis  convergence_status  \\\n",
       "1                        4.588737                   0   \n",
       "2                        4.582022                   0   \n",
       "3                        4.214025                   0   \n",
       "4                        4.213470                   0   \n",
       "5                        3.985714                   0   \n",
       "\n",
       "   squared_correlation_fsnps  squared_correlation_rsnps  in_feature region  \\\n",
       "1                   0.978662                   0.869372        True   SUNI   \n",
       "2                   0.978731                   0.869331        True   SUNI   \n",
       "3                   0.978907                   0.869343        True   SUNI   \n",
       "4                   0.978911                   0.868927        True   SUNI   \n",
       "5                   0.977652                   0.887832        True   SUNI   \n",
       "\n",
       "   raw_pvalue  cis_window_qvalue  null_cis_qvalue  genomewide_bh_qvalue  \n",
       "1    0.000703           0.002745         0.043506              0.099234  \n",
       "2    0.000708           0.002745         0.034931              0.099234  \n",
       "3    0.001060           0.002745         0.043506              0.099234  \n",
       "4    0.001061           0.002745         0.047231              0.099234  \n",
       "5    0.001332           0.002745         0.047231              0.099234  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "(401, 31)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SMTG\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>feature</th>\n",
       "      <th>rsid</th>\n",
       "      <th>chrom</th>\n",
       "      <th>snp_pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>allele_freq</th>\n",
       "      <th>HWE_chi_square_statistic</th>\n",
       "      <th>imputation_quality_score</th>\n",
       "      <th>log10_bh_qvalue</th>\n",
       "      <th>chi_square_statistic</th>\n",
       "      <th>effect_size</th>\n",
       "      <th>mapping_error_rate</th>\n",
       "      <th>ref_allele_mapping_bias</th>\n",
       "      <th>overdispersion</th>\n",
       "      <th>snp_id_within_region</th>\n",
       "      <th>num_feature_snps</th>\n",
       "      <th>num_testing_snps</th>\n",
       "      <th>num_iterations_null_hypothesis</th>\n",
       "      <th>null_iterations_alt_hypothesis</th>\n",
       "      <th>random_location_of_ties</th>\n",
       "      <th>log_likelihood_null_hypothesis</th>\n",
       "      <th>convergence_status</th>\n",
       "      <th>squared_correlation_fsnps</th>\n",
       "      <th>squared_correlation_rsnps</th>\n",
       "      <th>in_feature</th>\n",
       "      <th>region</th>\n",
       "      <th>raw_pvalue</th>\n",
       "      <th>cis_window_qvalue</th>\n",
       "      <th>null_cis_qvalue</th>\n",
       "      <th>genomewide_bh_qvalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr11_60251518_60251763</td>\n",
       "      <td>rs636317</td>\n",
       "      <td>chr11</td>\n",
       "      <td>60251677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.947368</td>\n",
       "      <td>0.120222</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-2.646028</td>\n",
       "      <td>9.325969</td>\n",
       "      <td>0.349554</td>\n",
       "      <td>0.023827</td>\n",
       "      <td>0.454270</td>\n",
       "      <td>14.676850</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>60251677</td>\n",
       "      <td>3.764374</td>\n",
       "      <td>0</td>\n",
       "      <td>0.990307</td>\n",
       "      <td>0.990307</td>\n",
       "      <td>True</td>\n",
       "      <td>SMTG</td>\n",
       "      <td>0.002259</td>\n",
       "      <td>0.002259</td>\n",
       "      <td>0.034640</td>\n",
       "      <td>0.280309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr17_45702992_45704386</td>\n",
       "      <td>rs17762308</td>\n",
       "      <td>chr17</td>\n",
       "      <td>45703582</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.342105</td>\n",
       "      <td>0.824653</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-2.609435</td>\n",
       "      <td>11.141087</td>\n",
       "      <td>0.684378</td>\n",
       "      <td>0.051434</td>\n",
       "      <td>0.236782</td>\n",
       "      <td>4.800978</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>45703582</td>\n",
       "      <td>4.751464</td>\n",
       "      <td>0</td>\n",
       "      <td>0.988776</td>\n",
       "      <td>0.984820</td>\n",
       "      <td>True</td>\n",
       "      <td>SMTG</td>\n",
       "      <td>0.000844</td>\n",
       "      <td>0.002458</td>\n",
       "      <td>0.011683</td>\n",
       "      <td>0.280309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr17_45702992_45704386</td>\n",
       "      <td>rs62056879</td>\n",
       "      <td>chr17</td>\n",
       "      <td>45704060</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.342105</td>\n",
       "      <td>0.824653</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-2.609435</td>\n",
       "      <td>9.891001</td>\n",
       "      <td>0.668710</td>\n",
       "      <td>0.052973</td>\n",
       "      <td>0.242037</td>\n",
       "      <td>5.232670</td>\n",
       "      <td>2.0</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>8</td>\n",
       "      <td>45703582</td>\n",
       "      <td>4.025779</td>\n",
       "      <td>0</td>\n",
       "      <td>0.985551</td>\n",
       "      <td>0.973879</td>\n",
       "      <td>True</td>\n",
       "      <td>SMTG</td>\n",
       "      <td>0.001661</td>\n",
       "      <td>0.002458</td>\n",
       "      <td>0.047964</td>\n",
       "      <td>0.280309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr17_45702992_45704386</td>\n",
       "      <td>rs968028</td>\n",
       "      <td>chr17</td>\n",
       "      <td>45703739</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.342105</td>\n",
       "      <td>0.824653</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-2.609435</td>\n",
       "      <td>9.699255</td>\n",
       "      <td>0.668376</td>\n",
       "      <td>0.052029</td>\n",
       "      <td>0.243765</td>\n",
       "      <td>4.841157</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>45703582</td>\n",
       "      <td>3.975581</td>\n",
       "      <td>0</td>\n",
       "      <td>0.986465</td>\n",
       "      <td>0.976028</td>\n",
       "      <td>True</td>\n",
       "      <td>SMTG</td>\n",
       "      <td>0.001843</td>\n",
       "      <td>0.002458</td>\n",
       "      <td>0.032534</td>\n",
       "      <td>0.280309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr17_45870069_45870629</td>\n",
       "      <td>rs56398500</td>\n",
       "      <td>chr17</td>\n",
       "      <td>45870122</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.342105</td>\n",
       "      <td>0.824653</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-2.486283</td>\n",
       "      <td>9.923435</td>\n",
       "      <td>0.349655</td>\n",
       "      <td>0.133834</td>\n",
       "      <td>0.209595</td>\n",
       "      <td>34.113632</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>45870122</td>\n",
       "      <td>3.698863</td>\n",
       "      <td>0</td>\n",
       "      <td>0.989065</td>\n",
       "      <td>0.985043</td>\n",
       "      <td>True</td>\n",
       "      <td>SMTG</td>\n",
       "      <td>0.001632</td>\n",
       "      <td>0.003264</td>\n",
       "      <td>0.020303</td>\n",
       "      <td>0.280309</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   feature        rsid  chrom   snp_pos ref alt  allele_freq  \\\n",
       "4  chr11_60251518_60251763    rs636317  chr11  60251677   C   T     0.947368   \n",
       "0  chr17_45702992_45704386  rs17762308  chr17  45703582   T   C     0.342105   \n",
       "2  chr17_45702992_45704386  rs62056879  chr17  45704060   T   C     0.342105   \n",
       "3  chr17_45702992_45704386    rs968028  chr17  45703739   A   G     0.342105   \n",
       "1  chr17_45870069_45870629  rs56398500  chr17  45870122   A   G     0.342105   \n",
       "\n",
       "   HWE_chi_square_statistic  imputation_quality_score  log10_bh_qvalue  \\\n",
       "4                  0.120222                       1.0        -2.646028   \n",
       "0                  0.824653                       1.0        -2.609435   \n",
       "2                  0.824653                       1.0        -2.609435   \n",
       "3                  0.824653                       1.0        -2.609435   \n",
       "1                  0.824653                       1.0        -2.486283   \n",
       "\n",
       "   chi_square_statistic  effect_size  mapping_error_rate  \\\n",
       "4              9.325969     0.349554            0.023827   \n",
       "0             11.141087     0.684378            0.051434   \n",
       "2              9.891001     0.668710            0.052973   \n",
       "3              9.699255     0.668376            0.052029   \n",
       "1              9.923435     0.349655            0.133834   \n",
       "\n",
       "   ref_allele_mapping_bias  overdispersion  snp_id_within_region  \\\n",
       "4                 0.454270       14.676850                   0.0   \n",
       "0                 0.236782        4.800978                   0.0   \n",
       "2                 0.242037        5.232670                   2.0   \n",
       "3                 0.243765        4.841157                   1.0   \n",
       "1                 0.209595       34.113632                   0.0   \n",
       "\n",
       "   num_feature_snps  num_testing_snps  num_iterations_null_hypothesis  \\\n",
       "4                 1                 1                               5   \n",
       "0                 5                 4                               8   \n",
       "2                 5                 4                              10   \n",
       "3                 5                 4                               8   \n",
       "1                 2                 2                               7   \n",
       "\n",
       "   null_iterations_alt_hypothesis  random_location_of_ties  \\\n",
       "4                               5                 60251677   \n",
       "0                               8                 45703582   \n",
       "2                               8                 45703582   \n",
       "3                               8                 45703582   \n",
       "1                               7                 45870122   \n",
       "\n",
       "   log_likelihood_null_hypothesis  convergence_status  \\\n",
       "4                        3.764374                   0   \n",
       "0                        4.751464                   0   \n",
       "2                        4.025779                   0   \n",
       "3                        3.975581                   0   \n",
       "1                        3.698863                   0   \n",
       "\n",
       "   squared_correlation_fsnps  squared_correlation_rsnps  in_feature region  \\\n",
       "4                   0.990307                   0.990307        True   SMTG   \n",
       "0                   0.988776                   0.984820        True   SMTG   \n",
       "2                   0.985551                   0.973879        True   SMTG   \n",
       "3                   0.986465                   0.976028        True   SMTG   \n",
       "1                   0.989065                   0.985043        True   SMTG   \n",
       "\n",
       "   raw_pvalue  cis_window_qvalue  null_cis_qvalue  genomewide_bh_qvalue  \n",
       "4    0.002259           0.002259         0.034640              0.280309  \n",
       "0    0.000844           0.002458         0.011683              0.280309  \n",
       "2    0.001661           0.002458         0.047964              0.280309  \n",
       "3    0.001843           0.002458         0.032534              0.280309  \n",
       "1    0.001632           0.003264         0.020303              0.280309  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "(498, 31)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PARL\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>feature</th>\n",
       "      <th>rsid</th>\n",
       "      <th>chrom</th>\n",
       "      <th>snp_pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>allele_freq</th>\n",
       "      <th>HWE_chi_square_statistic</th>\n",
       "      <th>imputation_quality_score</th>\n",
       "      <th>log10_bh_qvalue</th>\n",
       "      <th>chi_square_statistic</th>\n",
       "      <th>effect_size</th>\n",
       "      <th>mapping_error_rate</th>\n",
       "      <th>ref_allele_mapping_bias</th>\n",
       "      <th>overdispersion</th>\n",
       "      <th>snp_id_within_region</th>\n",
       "      <th>num_feature_snps</th>\n",
       "      <th>num_testing_snps</th>\n",
       "      <th>num_iterations_null_hypothesis</th>\n",
       "      <th>null_iterations_alt_hypothesis</th>\n",
       "      <th>random_location_of_ties</th>\n",
       "      <th>log_likelihood_null_hypothesis</th>\n",
       "      <th>convergence_status</th>\n",
       "      <th>squared_correlation_fsnps</th>\n",
       "      <th>squared_correlation_rsnps</th>\n",
       "      <th>in_feature</th>\n",
       "      <th>region</th>\n",
       "      <th>raw_pvalue</th>\n",
       "      <th>cis_window_qvalue</th>\n",
       "      <th>null_cis_qvalue</th>\n",
       "      <th>genomewide_bh_qvalue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr17_45825137_45827395</td>\n",
       "      <td>rs62057155</td>\n",
       "      <td>chr17</td>\n",
       "      <td>45827307</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>1.655026</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-2.621835</td>\n",
       "      <td>13.396173</td>\n",
       "      <td>0.716110</td>\n",
       "      <td>0.130264</td>\n",
       "      <td>0.123697</td>\n",
       "      <td>2.667632</td>\n",
       "      <td>13.0</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>45827307</td>\n",
       "      <td>6.089785</td>\n",
       "      <td>0</td>\n",
       "      <td>0.990664</td>\n",
       "      <td>0.990264</td>\n",
       "      <td>True</td>\n",
       "      <td>PARL</td>\n",
       "      <td>0.000252</td>\n",
       "      <td>0.002389</td>\n",
       "      <td>0.008219</td>\n",
       "      <td>0.249884</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr17_45825137_45827395</td>\n",
       "      <td>rs62057154</td>\n",
       "      <td>chr17</td>\n",
       "      <td>45827244</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>1.655026</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-2.621835</td>\n",
       "      <td>11.757083</td>\n",
       "      <td>0.706438</td>\n",
       "      <td>0.132745</td>\n",
       "      <td>0.123332</td>\n",
       "      <td>2.540562</td>\n",
       "      <td>12.0</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>45827307</td>\n",
       "      <td>5.310940</td>\n",
       "      <td>0</td>\n",
       "      <td>0.990610</td>\n",
       "      <td>0.989764</td>\n",
       "      <td>True</td>\n",
       "      <td>PARL</td>\n",
       "      <td>0.000606</td>\n",
       "      <td>0.002389</td>\n",
       "      <td>0.045270</td>\n",
       "      <td>0.249884</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr17_45825137_45827395</td>\n",
       "      <td>rs17763050</td>\n",
       "      <td>chr17</td>\n",
       "      <td>45825970</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.055556</td>\n",
       "      <td>0.062284</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-2.621835</td>\n",
       "      <td>11.541501</td>\n",
       "      <td>0.720707</td>\n",
       "      <td>0.127308</td>\n",
       "      <td>0.125002</td>\n",
       "      <td>3.177627</td>\n",
       "      <td>5.0</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>45827307</td>\n",
       "      <td>5.226442</td>\n",
       "      <td>0</td>\n",
       "      <td>0.990575</td>\n",
       "      <td>0.970495</td>\n",
       "      <td>True</td>\n",
       "      <td>PARL</td>\n",
       "      <td>0.000681</td>\n",
       "      <td>0.002389</td>\n",
       "      <td>0.045270</td>\n",
       "      <td>0.249884</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr17_45825137_45827395</td>\n",
       "      <td>rs62057148</td>\n",
       "      <td>chr17</td>\n",
       "      <td>45826119</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.055556</td>\n",
       "      <td>0.062284</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-2.621835</td>\n",
       "      <td>11.536333</td>\n",
       "      <td>0.720665</td>\n",
       "      <td>0.127318</td>\n",
       "      <td>0.125014</td>\n",
       "      <td>3.174356</td>\n",
       "      <td>6.0</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>45827307</td>\n",
       "      <td>5.223527</td>\n",
       "      <td>0</td>\n",
       "      <td>0.990600</td>\n",
       "      <td>0.969942</td>\n",
       "      <td>True</td>\n",
       "      <td>PARL</td>\n",
       "      <td>0.000682</td>\n",
       "      <td>0.002389</td>\n",
       "      <td>0.045270</td>\n",
       "      <td>0.249884</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr8_22599221_22600284</td>\n",
       "      <td>rs755934</td>\n",
       "      <td>chr8</td>\n",
       "      <td>22599692</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.305556</td>\n",
       "      <td>0.571478</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-2.607441</td>\n",
       "      <td>9.163267</td>\n",
       "      <td>0.373230</td>\n",
       "      <td>0.007269</td>\n",
       "      <td>0.554673</td>\n",
       "      <td>22.269325</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>22599692</td>\n",
       "      <td>3.459936</td>\n",
       "      <td>0</td>\n",
       "      <td>0.993740</td>\n",
       "      <td>0.993740</td>\n",
       "      <td>True</td>\n",
       "      <td>PARL</td>\n",
       "      <td>0.002469</td>\n",
       "      <td>0.002469</td>\n",
       "      <td>0.076945</td>\n",
       "      <td>0.249884</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   feature        rsid  chrom   snp_pos ref alt  allele_freq  \\\n",
       "0  chr17_45825137_45827395  rs62057155  chr17  45827307   C   T     0.333333   \n",
       "1  chr17_45825137_45827395  rs62057154  chr17  45827244   C   T     0.333333   \n",
       "2  chr17_45825137_45827395  rs17763050  chr17  45825970   G   A     0.055556   \n",
       "3  chr17_45825137_45827395  rs62057148  chr17  45826119   G   C     0.055556   \n",
       "4   chr8_22599221_22600284    rs755934   chr8  22599692   T   G     0.305556   \n",
       "\n",
       "   HWE_chi_square_statistic  imputation_quality_score  log10_bh_qvalue  \\\n",
       "0                  1.655026                       1.0        -2.621835   \n",
       "1                  1.655026                       1.0        -2.621835   \n",
       "2                  0.062284                       1.0        -2.621835   \n",
       "3                  0.062284                       1.0        -2.621835   \n",
       "4                  0.571478                       1.0        -2.607441   \n",
       "\n",
       "   chi_square_statistic  effect_size  mapping_error_rate  \\\n",
       "0             13.396173     0.716110            0.130264   \n",
       "1             11.757083     0.706438            0.132745   \n",
       "2             11.541501     0.720707            0.127308   \n",
       "3             11.536333     0.720665            0.127318   \n",
       "4              9.163267     0.373230            0.007269   \n",
       "\n",
       "   ref_allele_mapping_bias  overdispersion  snp_id_within_region  \\\n",
       "0                 0.123697        2.667632                  13.0   \n",
       "1                 0.123332        2.540562                  12.0   \n",
       "2                 0.125002        3.177627                   5.0   \n",
       "3                 0.125014        3.174356                   6.0   \n",
       "4                 0.554673       22.269325                   0.0   \n",
       "\n",
       "   num_feature_snps  num_testing_snps  num_iterations_null_hypothesis  \\\n",
       "0                14                14                               5   \n",
       "1                14                14                               5   \n",
       "2                14                14                               5   \n",
       "3                14                14                               6   \n",
       "4                 1                 1                               4   \n",
       "\n",
       "   null_iterations_alt_hypothesis  random_location_of_ties  \\\n",
       "0                               5                 45827307   \n",
       "1                               5                 45827307   \n",
       "2                               5                 45827307   \n",
       "3                               5                 45827307   \n",
       "4                               4                 22599692   \n",
       "\n",
       "   log_likelihood_null_hypothesis  convergence_status  \\\n",
       "0                        6.089785                   0   \n",
       "1                        5.310940                   0   \n",
       "2                        5.226442                   0   \n",
       "3                        5.223527                   0   \n",
       "4                        3.459936                   0   \n",
       "\n",
       "   squared_correlation_fsnps  squared_correlation_rsnps  in_feature region  \\\n",
       "0                   0.990664                   0.990264        True   PARL   \n",
       "1                   0.990610                   0.989764        True   PARL   \n",
       "2                   0.990575                   0.970495        True   PARL   \n",
       "3                   0.990600                   0.969942        True   PARL   \n",
       "4                   0.993740                   0.993740        True   PARL   \n",
       "\n",
       "   raw_pvalue  cis_window_qvalue  null_cis_qvalue  genomewide_bh_qvalue  \n",
       "0    0.000252           0.002389         0.008219              0.249884  \n",
       "1    0.000606           0.002389         0.045270              0.249884  \n",
       "2    0.000681           0.002389         0.045270              0.249884  \n",
       "3    0.000682           0.002389         0.045270              0.249884  \n",
       "4    0.002469           0.002469         0.076945              0.249884  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "(506, 31)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for region in multi_bam_regs:\n",
    "    print(region)\n",
    "    region_output = pd.read_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/'\n",
    "                                    + region + '/all_' + region + '_output.txt', sep='\\t', header=None)\n",
    "    region_output.columns = ['feature', 'rsid', 'chrom', 'snp_pos', 'ref', 'alt', 'allele_freq', 'HWE_chi_square_statistic',\n",
    "                             'imputation_quality_score', 'log10_bh_qvalue', 'chi_square_statistic', 'effect_size', 'mapping_error_rate',\n",
    "                            'ref_allele_mapping_bias', 'overdispersion', 'snp_id_within_region', 'num_feature_snps', 'num_testing_snps',\n",
    "                            'num_iterations_null_hypothesis', 'null_iterations_alt_hypothesis', 'random_location_of_ties',\n",
    "                             'log_likelihood_null_hypothesis', 'convergence_status', 'squared_correlation_fsnps', 'squared_correlation_rsnps']\n",
    "    region_output = region_output.loc[region_output['rsid'] != 'SKIPPED']\n",
    "    in_feature = []\n",
    "    for ind,test in region_output.iterrows():\n",
    "        start = int(test['feature'].split('_')[1])\n",
    "        end = int(test['feature'].split('_')[2])\n",
    "        if int(test['snp_pos']) >= start and int(test['snp_pos']) <= end:\n",
    "            in_feature.append(True)\n",
    "        else:\n",
    "            in_feature.append(False)\n",
    "    region_output['in_feature'] = in_feature\n",
    "    region_output['region'] = [region for i in range(len(region_output))]\n",
    "    region_output['raw_pvalue'] = region_output['chi_square_statistic'].map(lambda x : (1 - stats.chi2.cdf(float(x), 1)))\n",
    "    region_output['cis_window_qvalue'] = 10**region_output['log10_bh_qvalue']\n",
    "    \n",
    "    region_permutation = pd.read_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/'\n",
    "                                    + region + '/all_' + region + '_output_permutation.txt', sep='\\t', header=None)\n",
    "    region_permutation.columns = ['feature', 'rsid', 'chrom', 'snp_pos', 'ref', 'alt', 'allele_freq', 'HWE_chi_square_statistic',\n",
    "                             'imputation_quality_score', 'log10_bh_qvalue', 'chi_square_statistic', 'effect_size', 'mapping_error_rate',\n",
    "                            'ref_allele_mapping_bias', 'overdispersion', 'snp_id_within_region', 'num_feature_snps', 'num_testing_snps',\n",
    "                            'num_iterations_null_hypothesis', 'null_iterations_alt_hypothesis', 'random_location_of_ties',\n",
    "                             'log_likelihood_null_hypothesis', 'convergence_status', 'squared_correlation_fsnps', 'squared_correlation_rsnps']\n",
    "    region_permutation = region_permutation.loc[region_permutation['rsid'] != 'SKIPPED']\n",
    "    region_permutation['raw_pvalue'] = region_permutation['chi_square_statistic'].map(lambda x : (1 - stats.chi2.cdf(float(x), 1)))\n",
    "    region_permutation['cis_window_qvalue'] = 10**region_permutation['log10_bh_qvalue']\n",
    "    \n",
    "    region_output['null_cis_qvalue'] = region_permutation['cis_window_qvalue']\n",
    "    region_output.sort_values(by='cis_window_qvalue', inplace=True)\n",
    "    region_output['genomewide_bh_qvalue'] = sm.stats.fdrcorrection(list(region_output['cis_window_qvalue']))[1]\n",
    "    region_output.to_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/'\n",
    "                            + region + '/final_' + region + '_output.txt', sep='\\t', index=False)\n",
    "    \n",
    "    display(region_output.head())\n",
    "    display(region_output.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "# run run_permutation_rasqual.ipynb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CAUD\n",
      "HIPP\n",
      "MDFG\n",
      "MDTG\n",
      "PTMN\n",
      "SUNI\n",
      "SMTG\n",
      "PARL\n"
     ]
    }
   ],
   "source": [
    "with pd.ExcelWriter('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/'\n",
    "                    + 'rasqual_output.xlsx') as writer:  \n",
    "    for region in multi_bam_regs:\n",
    "        print(region)\n",
    "        region_output = pd.read_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/'\n",
    "                                    + region + '/final_' + region + '_output.txt', sep='\\t')\n",
    "        region_output.to_excel(writer, sheet_name=region, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "grep: warning: GREP_OPTIONS is deprecated; please use an alias or script\n"
     ]
    }
   ],
   "source": [
    "! cat /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/*/final_*_output.txt | grep -v ^feature > /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/final_regions_output.txt\n",
    "! cat /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/*/final_*_output_permutation.txt | grep -v ^feature > /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/final_regions_output_permutation.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'All Significant SNPs: '"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "(209, 34)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "'Unique Significant SNPs: '"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "(116, 34)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ryan_bias_snps = ['rs1237999','rs3755519','rs6781790','rs55682376','rs12119232','rs147889095',\n",
    "                'rs3104768','rs58392387','rs17631303','rs17762308','rs56327054','rs12150223',\n",
    "                'rs2532307','rs72914882','rs72914893','rs72914895','rs600834','rs600850','rs11077365',\n",
    "                'rs72907463','rs691362','rs691346','rs691331','rs691328','rs691317','rs6501212',\n",
    "                'rs76516995','rs10182292','rs744373','rs12493578']\n",
    "in_original = []\n",
    "in_original_sig = []\n",
    "combined_output = pd.read_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/final_regions_output.txt', sep='\\t', header=None)\n",
    "display(combined_output.head())\n",
    "combined_output.columns = ['feature', 'rsid', 'chrom', 'snp_pos', 'ref', 'alt', 'allele_freq', 'HWE_chi_square_statistic',\n",
    "                             'imputation_quality_score', 'log10_bh_qvalue', 'chi_square_statistic', 'effect_size', 'mapping_error_rate',\n",
    "                            'ref_allele_mapping_bias', 'overdispersion', 'snp_id_within_region', 'num_feature_snps', 'num_testing_snps',\n",
    "                            'num_iterations_null_hypothesis', 'null_iterations_alt_hypothesis', 'random_location_of_ties',\n",
    "                             'log_likelihood_null_hypothesis', 'convergence_status', 'squared_correlation_fsnps', 'squared_correlation_rsnps',\n",
    "                              'in_feature', 'region', 'raw_pvalue', 'cis_window_qvalue', 'null_cis_qvalue', 'genomewide_bh_qvalue',\n",
    "                               'permutation_significant', 'fdr10_threshold']\n",
    "combined_output.sort_values(by=['rsid','cis_window_qvalue'], inplace=True)\n",
    "sig_snps = combined_output.loc[combined_output['permutation_significant'] == True]\n",
    "for ind,val in combined_output.iterrows():\n",
    "    if val['rsid'] in ryan_bias_snps:\n",
    "        in_original.append(True)\n",
    "    else:\n",
    "        in_original.append(False)\n",
    "combined_output['in_original'] = in_original\n",
    "\n",
    "for ind,val in sig_snps.iterrows():\n",
    "    if val['rsid'] in ryan_bias_snps:\n",
    "        in_original_sig.append(True)\n",
    "    else:\n",
    "        in_original_sig.append(False)\n",
    "sig_snps['in_original'] = in_original_sig\n",
    "unique_sig_snps = sig_snps.drop_duplicates(subset='rsid')\n",
    "unique_sig_snps.sort_values(by=['cis_window_qvalue'], inplace=True)\n",
    "display('All Significant SNPs: ', sig_snps.shape)\n",
    "display('Unique Significant SNPs: ', unique_sig_snps.shape)\n",
    "sig_snps.to_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/rasqual_significant_snps.csv', sep='\\t', index=False)\n",
    "unique_sig_snps.to_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/unique_rasqual_significant_snps.csv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Plot P-Value Distributions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "# for region in multi_bam_regs:\n",
    "#     print(region)\n",
    "#     region_output = pd.read_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/'\n",
    "#                                     + region + '/final_' + region + '_output.txt', sep='\\t')\n",
    "#     print(len(region_output))\n",
    "#     display(p9.ggplot(data=region_output, mapping=p9.aes(x='raw_pvalue')) + p9.geom_histogram(bins=20) + p9.labs(title='Raw P-value: ' + region))\n",
    "#     display(p9.ggplot(data=region_output, mapping=p9.aes(x='cis_window_qvalue')) + p9.geom_histogram(bins=20) + p9.labs(title='Q-value: ' + region))\n",
    "#     display(p9.ggplot(data=region_output, mapping=p9.aes(x='null_cis_qvalue')) + p9.geom_histogram(bins=20) + p9.labs(title='Null Q-value: ' + region))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>r2_with_ld_tag</th>\n",
       "      <th>ld_tag_chr</th>\n",
       "      <th>ld_tag_pos</th>\n",
       "      <th>source_gwas</th>\n",
       "      <th>source</th>\n",
       "      <th>snp_id</th>\n",
       "      <th>locus_num</th>\n",
       "      <th>number_ld_buddies</th>\n",
       "      <th>ld_tag_locus</th>\n",
       "      <th>rsid</th>\n",
       "      <th>chrom_hg19</th>\n",
       "      <th>snp_pos_hg19</th>\n",
       "      <th>file</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>noneffect_allele</th>\n",
       "      <th>direction</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>has_coloc</th>\n",
       "      <th>direct_atac_overlap_narrow_tissue_regions</th>\n",
       "      <th>containing_atac_tissues_narrow_tissue_regions</th>\n",
       "      <th>nearest_atac_tissue_narrow_tissue_regions</th>\n",
       "      <th>start_narrow_tissue_regions</th>\n",
       "      <th>end_narrow_tissue_regions</th>\n",
       "      <th>dist_narrow_tissue_regions</th>\n",
       "      <th>direct_atac_overlap_broad_tissue_regions</th>\n",
       "      <th>containing_atac_tissues_broad_tissue_regions</th>\n",
       "      <th>nearest_atac_tissue_broad_tissue_regions</th>\n",
       "      <th>start_broad_tissue_regions</th>\n",
       "      <th>end_broad_tissue_regions</th>\n",
       "      <th>dist_broad_tissue_regions</th>\n",
       "      <th>direct_atac_overlap_single_cell</th>\n",
       "      <th>containing_atac_tissues_single_cell</th>\n",
       "      <th>nearest_atac_tissue_single_cell</th>\n",
       "      <th>start_single_cell</th>\n",
       "      <th>end_single_cell</th>\n",
       "      <th>dist_single_cell</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10</td>\n",
       "      <td>102207833</td>\n",
       "      <td>0.805869</td>\n",
       "      <td>10</td>\n",
       "      <td>102255522</td>\n",
       "      <td>Nalls_23andMe</td>\n",
       "      <td>LD</td>\n",
       "      <td>10_102207833</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>10_102255522</td>\n",
       "      <td>rs10883717</td>\n",
       "      <td>10.0</td>\n",
       "      <td>103967590.0</td>\n",
       "      <td>/users/mgloud/projects/ad-pd/data/gwas/prepare...</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>0.014434</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>CAUD</td>\n",
       "      <td>102225620</td>\n",
       "      <td>102226734</td>\n",
       "      <td>17780.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>ALLO</td>\n",
       "      <td>102225838</td>\n",
       "      <td>102226054</td>\n",
       "      <td>18005.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>microglia</td>\n",
       "      <td>102213693</td>\n",
       "      <td>102214072</td>\n",
       "      <td>5860.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10</td>\n",
       "      <td>102245653</td>\n",
       "      <td>0.987490</td>\n",
       "      <td>10</td>\n",
       "      <td>102255522</td>\n",
       "      <td>Nalls_23andMe</td>\n",
       "      <td>LD</td>\n",
       "      <td>10_102245653</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>10_102255522</td>\n",
       "      <td>rs2296887</td>\n",
       "      <td>10.0</td>\n",
       "      <td>104005410.0</td>\n",
       "      <td>/users/mgloud/projects/ad-pd/data/gwas/prepare...</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>0.006977</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>CAUD,HIPP,MDFG,PARL,PTMN,SMTG,SUNI</td>\n",
       "      <td>CAUD,HIPP,MDFG,PARL,PTMN,SMTG,SUNI</td>\n",
       "      <td>102244764</td>\n",
       "      <td>102246025</td>\n",
       "      <td>258.5</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>ALLO</td>\n",
       "      <td>102245836</td>\n",
       "      <td>102246040</td>\n",
       "      <td>183.0</td>\n",
       "      <td>True</td>\n",
       "      <td>astrocytes,doublets,excitatory_neurons,inhibit...</td>\n",
       "      <td>astrocytes,doublets,excitatory_neurons,inhibit...</td>\n",
       "      <td>102244973</td>\n",
       "      <td>102245978</td>\n",
       "      <td>177.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10</td>\n",
       "      <td>102250385</td>\n",
       "      <td>0.809348</td>\n",
       "      <td>10</td>\n",
       "      <td>102255522</td>\n",
       "      <td>Nalls_23andMe</td>\n",
       "      <td>LD</td>\n",
       "      <td>10_102250385</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>10_102255522</td>\n",
       "      <td>rs7913281</td>\n",
       "      <td>10.0</td>\n",
       "      <td>104010142.0</td>\n",
       "      <td>/users/mgloud/projects/ad-pd/data/gwas/prepare...</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>+</td>\n",
       "      <td>0.012859</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>CAUD</td>\n",
       "      <td>102244791</td>\n",
       "      <td>102246164</td>\n",
       "      <td>4221.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>ALLO</td>\n",
       "      <td>102245836</td>\n",
       "      <td>102246040</td>\n",
       "      <td>4345.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>opcs</td>\n",
       "      <td>102247774</td>\n",
       "      <td>102248243</td>\n",
       "      <td>2142.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10</td>\n",
       "      <td>102251214</td>\n",
       "      <td>0.809348</td>\n",
       "      <td>10</td>\n",
       "      <td>102255522</td>\n",
       "      <td>Nalls_23andMe</td>\n",
       "      <td>LD</td>\n",
       "      <td>10_102251214</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>10_102255522</td>\n",
       "      <td>rs10883720</td>\n",
       "      <td>10.0</td>\n",
       "      <td>104010971.0</td>\n",
       "      <td>/users/mgloud/projects/ad-pd/data/gwas/prepare...</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>0.012617</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>CAUD</td>\n",
       "      <td>102244791</td>\n",
       "      <td>102246164</td>\n",
       "      <td>5050.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>ALLO</td>\n",
       "      <td>102245836</td>\n",
       "      <td>102246040</td>\n",
       "      <td>5174.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>opcs</td>\n",
       "      <td>102247774</td>\n",
       "      <td>102248243</td>\n",
       "      <td>2971.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10</td>\n",
       "      <td>102255522</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>10</td>\n",
       "      <td>102255522</td>\n",
       "      <td>Nalls_23andMe</td>\n",
       "      <td>Nalls-Chang</td>\n",
       "      <td>10_102255522</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>10_102255522</td>\n",
       "      <td>rs10748818</td>\n",
       "      <td>10.0</td>\n",
       "      <td>104015279.0</td>\n",
       "      <td>/users/mgloud/projects/ad-pd/data/gwas/prepare...</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>+</td>\n",
       "      <td>0.006335</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>CAUD</td>\n",
       "      <td>102244791</td>\n",
       "      <td>102246164</td>\n",
       "      <td>9358.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>ALLO</td>\n",
       "      <td>102245836</td>\n",
       "      <td>102246040</td>\n",
       "      <td>9482.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>inhibitory_neurons</td>\n",
       "      <td>102255548</td>\n",
       "      <td>102256050</td>\n",
       "      <td>26.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chr        pos  r2_with_ld_tag  ld_tag_chr  ld_tag_pos    source_gwas  \\\n",
       "0   10  102207833        0.805869          10   102255522  Nalls_23andMe   \n",
       "1   10  102245653        0.987490          10   102255522  Nalls_23andMe   \n",
       "2   10  102250385        0.809348          10   102255522  Nalls_23andMe   \n",
       "3   10  102251214        0.809348          10   102255522  Nalls_23andMe   \n",
       "4   10  102255522        1.000000          10   102255522  Nalls_23andMe   \n",
       "\n",
       "        source        snp_id  locus_num  number_ld_buddies  ld_tag_locus  \\\n",
       "0           LD  10_102207833          1                  9  10_102255522   \n",
       "1           LD  10_102245653          1                  9  10_102255522   \n",
       "2           LD  10_102250385          1                  9  10_102255522   \n",
       "3           LD  10_102251214          1                  9  10_102255522   \n",
       "4  Nalls-Chang  10_102255522          1                  9  10_102255522   \n",
       "\n",
       "         rsid  chrom_hg19  snp_pos_hg19  \\\n",
       "0  rs10883717        10.0   103967590.0   \n",
       "1   rs2296887        10.0   104005410.0   \n",
       "2   rs7913281        10.0   104010142.0   \n",
       "3  rs10883720        10.0   104010971.0   \n",
       "4  rs10748818        10.0   104015279.0   \n",
       "\n",
       "                                                file effect_allele  \\\n",
       "0  /users/mgloud/projects/ad-pd/data/gwas/prepare...             T   \n",
       "1  /users/mgloud/projects/ad-pd/data/gwas/prepare...             T   \n",
       "2  /users/mgloud/projects/ad-pd/data/gwas/prepare...             G   \n",
       "3  /users/mgloud/projects/ad-pd/data/gwas/prepare...             G   \n",
       "4  /users/mgloud/projects/ad-pd/data/gwas/prepare...             G   \n",
       "\n",
       "  noneffect_allele direction    pvalue  has_coloc  \\\n",
       "0                C         -  0.014434      False   \n",
       "1                C         -  0.006977      False   \n",
       "2                A         +  0.012859      False   \n",
       "3                C         -  0.012617      False   \n",
       "4                A         +  0.006335      False   \n",
       "\n",
       "   direct_atac_overlap_narrow_tissue_regions  \\\n",
       "0                                      False   \n",
       "1                                       True   \n",
       "2                                      False   \n",
       "3                                      False   \n",
       "4                                      False   \n",
       "\n",
       "  containing_atac_tissues_narrow_tissue_regions  \\\n",
       "0                                          none   \n",
       "1            CAUD,HIPP,MDFG,PARL,PTMN,SMTG,SUNI   \n",
       "2                                          none   \n",
       "3                                          none   \n",
       "4                                          none   \n",
       "\n",
       "  nearest_atac_tissue_narrow_tissue_regions  start_narrow_tissue_regions  \\\n",
       "0                                      CAUD                    102225620   \n",
       "1        CAUD,HIPP,MDFG,PARL,PTMN,SMTG,SUNI                    102244764   \n",
       "2                                      CAUD                    102244791   \n",
       "3                                      CAUD                    102244791   \n",
       "4                                      CAUD                    102244791   \n",
       "\n",
       "   end_narrow_tissue_regions  dist_narrow_tissue_regions  \\\n",
       "0                  102226734                     17780.0   \n",
       "1                  102246025                       258.5   \n",
       "2                  102246164                      4221.0   \n",
       "3                  102246164                      5050.0   \n",
       "4                  102246164                      9358.0   \n",
       "\n",
       "   direct_atac_overlap_broad_tissue_regions  \\\n",
       "0                                     False   \n",
       "1                                     False   \n",
       "2                                     False   \n",
       "3                                     False   \n",
       "4                                     False   \n",
       "\n",
       "  containing_atac_tissues_broad_tissue_regions  \\\n",
       "0                                         none   \n",
       "1                                         none   \n",
       "2                                         none   \n",
       "3                                         none   \n",
       "4                                         none   \n",
       "\n",
       "  nearest_atac_tissue_broad_tissue_regions  start_broad_tissue_regions  \\\n",
       "0                                     ALLO                   102225838   \n",
       "1                                     ALLO                   102245836   \n",
       "2                                     ALLO                   102245836   \n",
       "3                                     ALLO                   102245836   \n",
       "4                                     ALLO                   102245836   \n",
       "\n",
       "   end_broad_tissue_regions  dist_broad_tissue_regions  \\\n",
       "0                 102226054                    18005.0   \n",
       "1                 102246040                      183.0   \n",
       "2                 102246040                     4345.0   \n",
       "3                 102246040                     5174.0   \n",
       "4                 102246040                     9482.0   \n",
       "\n",
       "   direct_atac_overlap_single_cell  \\\n",
       "0                            False   \n",
       "1                             True   \n",
       "2                            False   \n",
       "3                            False   \n",
       "4                            False   \n",
       "\n",
       "                 containing_atac_tissues_single_cell  \\\n",
       "0                                               none   \n",
       "1  astrocytes,doublets,excitatory_neurons,inhibit...   \n",
       "2                                               none   \n",
       "3                                               none   \n",
       "4                                               none   \n",
       "\n",
       "                     nearest_atac_tissue_single_cell  start_single_cell  \\\n",
       "0                                          microglia          102213693   \n",
       "1  astrocytes,doublets,excitatory_neurons,inhibit...          102244973   \n",
       "2                                               opcs          102247774   \n",
       "3                                               opcs          102247774   \n",
       "4                                 inhibitory_neurons          102255548   \n",
       "\n",
       "   end_single_cell  dist_single_cell  \n",
       "0        102214072            5860.0  \n",
       "1        102245978             177.5  \n",
       "2        102248243            2142.0  \n",
       "3        102248243            2971.0  \n",
       "4        102256050              26.0  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "(13773, 38)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "adpd_stage3 = pd.read_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/snps_final/191121_ld_buddies_table_stage3.tsv', sep='\\t')\n",
    "display(adpd_stage3.head())\n",
    "display(adpd_stage3.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>feature</th>\n",
       "      <th>rsid</th>\n",
       "      <th>chrom</th>\n",
       "      <th>snp_pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>allele_freq</th>\n",
       "      <th>effect_size</th>\n",
       "      <th>mapping_error_rate</th>\n",
       "      <th>ref_allele_mapping_bias</th>\n",
       "      <th>overdispersion</th>\n",
       "      <th>snp_id_within_region</th>\n",
       "      <th>num_feature_snps</th>\n",
       "      <th>num_testing_snps</th>\n",
       "      <th>in_feature</th>\n",
       "      <th>region</th>\n",
       "      <th>raw_pvalue</th>\n",
       "      <th>cis_window_qvalue</th>\n",
       "      <th>genomewide_bh_qvalue</th>\n",
       "      <th>permutation_significant</th>\n",
       "      <th>fdr10_threshold</th>\n",
       "      <th>in_original</th>\n",
       "      <th>r2_with_ld_tag</th>\n",
       "      <th>ld_tag_chr</th>\n",
       "      <th>ld_tag_pos</th>\n",
       "      <th>source_gwas</th>\n",
       "      <th>source</th>\n",
       "      <th>locus_num</th>\n",
       "      <th>number_ld_buddies</th>\n",
       "      <th>ld_tag_locus</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>noneffect_allele</th>\n",
       "      <th>direction</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>has_coloc</th>\n",
       "      <th>direct_atac_overlap_narrow_tissue_regions</th>\n",
       "      <th>containing_atac_tissues_narrow_tissue_regions</th>\n",
       "      <th>nearest_atac_tissue_narrow_tissue_regions</th>\n",
       "      <th>start_narrow_tissue_regions</th>\n",
       "      <th>end_narrow_tissue_regions</th>\n",
       "      <th>dist_narrow_tissue_regions</th>\n",
       "      <th>direct_atac_overlap_broad_tissue_regions</th>\n",
       "      <th>containing_atac_tissues_broad_tissue_regions</th>\n",
       "      <th>nearest_atac_tissue_broad_tissue_regions</th>\n",
       "      <th>start_broad_tissue_regions</th>\n",
       "      <th>end_broad_tissue_regions</th>\n",
       "      <th>dist_broad_tissue_regions</th>\n",
       "      <th>direct_atac_overlap_single_cell</th>\n",
       "      <th>containing_atac_tissues_single_cell</th>\n",
       "      <th>nearest_atac_tissue_single_cell</th>\n",
       "      <th>start_single_cell</th>\n",
       "      <th>end_single_cell</th>\n",
       "      <th>dist_single_cell</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1_161185016_161186504</td>\n",
       "      <td>rs11585858</td>\n",
       "      <td>chr1</td>\n",
       "      <td>161186243</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.552632</td>\n",
       "      <td>0.382894</td>\n",
       "      <td>0.000046</td>\n",
       "      <td>0.441369</td>\n",
       "      <td>40.915012</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>SMTG</td>\n",
       "      <td>0.009947</td>\n",
       "      <td>0.019893</td>\n",
       "      <td>0.586346</td>\n",
       "      <td>True</td>\n",
       "      <td>0.032534</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>161186243</td>\n",
       "      <td>Alzheimers_Jansen_2018</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>129</td>\n",
       "      <td>2</td>\n",
       "      <td>1_161186243</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>+</td>\n",
       "      <td>5.580000e-10</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>HIPP,MDFG,PARL,SMTG</td>\n",
       "      <td>HIPP,MDFG,PARL,SMTG</td>\n",
       "      <td>161185040</td>\n",
       "      <td>161186430</td>\n",
       "      <td>422.0</td>\n",
       "      <td>True</td>\n",
       "      <td>ALLO</td>\n",
       "      <td>ALLO</td>\n",
       "      <td>161185897</td>\n",
       "      <td>161186291</td>\n",
       "      <td>149.0</td>\n",
       "      <td>True</td>\n",
       "      <td>microglia</td>\n",
       "      <td>microglia</td>\n",
       "      <td>161185232</td>\n",
       "      <td>161186486</td>\n",
       "      <td>384.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1_232528515_232530038</td>\n",
       "      <td>rs10797576</td>\n",
       "      <td>chr1</td>\n",
       "      <td>232528865</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.281250</td>\n",
       "      <td>0.359946</td>\n",
       "      <td>0.000028</td>\n",
       "      <td>0.581322</td>\n",
       "      <td>27.696111</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>CAUD</td>\n",
       "      <td>0.000556</td>\n",
       "      <td>0.000556</td>\n",
       "      <td>0.016024</td>\n",
       "      <td>True</td>\n",
       "      <td>0.005189</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>232528865</td>\n",
       "      <td>Chang_23andMe_Parkinsons</td>\n",
       "      <td>Nalls-Chang</td>\n",
       "      <td>22</td>\n",
       "      <td>14</td>\n",
       "      <td>1_232528865</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>+</td>\n",
       "      <td>7.480000e-04</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>CAUD,PARL,PTMN,SMTG</td>\n",
       "      <td>CAUD,PARL,PTMN,SMTG</td>\n",
       "      <td>232528497</td>\n",
       "      <td>232530050</td>\n",
       "      <td>287.5</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>STRI</td>\n",
       "      <td>232528568</td>\n",
       "      <td>232528768</td>\n",
       "      <td>97.0</td>\n",
       "      <td>True</td>\n",
       "      <td>excitatory_neurons,inhibitory_neurons,neurons_...</td>\n",
       "      <td>excitatory_neurons,inhibitory_neurons,neurons_...</td>\n",
       "      <td>232528552</td>\n",
       "      <td>232529236</td>\n",
       "      <td>29.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1_232528515_232530038</td>\n",
       "      <td>rs10797576</td>\n",
       "      <td>chr1</td>\n",
       "      <td>232528865</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.281250</td>\n",
       "      <td>0.359946</td>\n",
       "      <td>0.000028</td>\n",
       "      <td>0.581322</td>\n",
       "      <td>27.696111</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>CAUD</td>\n",
       "      <td>0.000556</td>\n",
       "      <td>0.000556</td>\n",
       "      <td>0.016024</td>\n",
       "      <td>True</td>\n",
       "      <td>0.005189</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>232528865</td>\n",
       "      <td>Nalls_23andMe</td>\n",
       "      <td>Nalls-Chang</td>\n",
       "      <td>22</td>\n",
       "      <td>14</td>\n",
       "      <td>1_232528865</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>+</td>\n",
       "      <td>7.480000e-04</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>CAUD,PARL,PTMN,SMTG</td>\n",
       "      <td>CAUD,PARL,PTMN,SMTG</td>\n",
       "      <td>232528497</td>\n",
       "      <td>232530050</td>\n",
       "      <td>287.5</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>STRI</td>\n",
       "      <td>232528568</td>\n",
       "      <td>232528768</td>\n",
       "      <td>97.0</td>\n",
       "      <td>True</td>\n",
       "      <td>excitatory_neurons,inhibitory_neurons,neurons_...</td>\n",
       "      <td>excitatory_neurons,inhibitory_neurons,neurons_...</td>\n",
       "      <td>232528552</td>\n",
       "      <td>232529236</td>\n",
       "      <td>29.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>163</th>\n",
       "      <td>chr11_60251518_60251763</td>\n",
       "      <td>rs636317</td>\n",
       "      <td>chr11</td>\n",
       "      <td>60251677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.833333</td>\n",
       "      <td>0.387836</td>\n",
       "      <td>0.007736</td>\n",
       "      <td>0.491990</td>\n",
       "      <td>18.930226</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>MDFG</td>\n",
       "      <td>0.008743</td>\n",
       "      <td>0.008745</td>\n",
       "      <td>0.190429</td>\n",
       "      <td>True</td>\n",
       "      <td>0.024660</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>11</td>\n",
       "      <td>60251677</td>\n",
       "      <td>Alzheimers_Jansen_2018</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>11</td>\n",
       "      <td>385</td>\n",
       "      <td>11_60251677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>2.830000e-13</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>MDFG</td>\n",
       "      <td>60264747</td>\n",
       "      <td>60266009</td>\n",
       "      <td>13070.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>ISOC</td>\n",
       "      <td>60265062</td>\n",
       "      <td>60265262</td>\n",
       "      <td>13385.0</td>\n",
       "      <td>True</td>\n",
       "      <td>excitatory_neurons,microglia</td>\n",
       "      <td>excitatory_neurons,microglia</td>\n",
       "      <td>60251481</td>\n",
       "      <td>60252738</td>\n",
       "      <td>432.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164</th>\n",
       "      <td>chr11_60251518_60251763</td>\n",
       "      <td>rs636317</td>\n",
       "      <td>chr11</td>\n",
       "      <td>60251677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.833333</td>\n",
       "      <td>0.387836</td>\n",
       "      <td>0.007736</td>\n",
       "      <td>0.491990</td>\n",
       "      <td>18.930226</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>MDFG</td>\n",
       "      <td>0.008743</td>\n",
       "      <td>0.008745</td>\n",
       "      <td>0.190429</td>\n",
       "      <td>True</td>\n",
       "      <td>0.024660</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>11</td>\n",
       "      <td>60251677</td>\n",
       "      <td>Alzheimers_Kunkle_2019</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>11</td>\n",
       "      <td>261</td>\n",
       "      <td>11_60251677</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>+</td>\n",
       "      <td>5.910000e-15</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>MDFG</td>\n",
       "      <td>60264747</td>\n",
       "      <td>60266009</td>\n",
       "      <td>13070.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>ISOC</td>\n",
       "      <td>60265062</td>\n",
       "      <td>60265262</td>\n",
       "      <td>13385.0</td>\n",
       "      <td>True</td>\n",
       "      <td>excitatory_neurons,microglia</td>\n",
       "      <td>excitatory_neurons,microglia</td>\n",
       "      <td>60251481</td>\n",
       "      <td>60252738</td>\n",
       "      <td>432.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      feature        rsid  chrom    snp_pos ref alt  \\\n",
       "18   chr1_161185016_161186504  rs11585858   chr1  161186243   C   A   \n",
       "0    chr1_232528515_232530038  rs10797576   chr1  232528865   C   T   \n",
       "1    chr1_232528515_232530038  rs10797576   chr1  232528865   C   T   \n",
       "163   chr11_60251518_60251763    rs636317  chr11   60251677   C   T   \n",
       "164   chr11_60251518_60251763    rs636317  chr11   60251677   C   T   \n",
       "\n",
       "     allele_freq  effect_size  mapping_error_rate  ref_allele_mapping_bias  \\\n",
       "18      0.552632     0.382894            0.000046                 0.441369   \n",
       "0       0.281250     0.359946            0.000028                 0.581322   \n",
       "1       0.281250     0.359946            0.000028                 0.581322   \n",
       "163     0.833333     0.387836            0.007736                 0.491990   \n",
       "164     0.833333     0.387836            0.007736                 0.491990   \n",
       "\n",
       "     overdispersion  snp_id_within_region  num_feature_snps  num_testing_snps  \\\n",
       "18        40.915012                     1                 2                 2   \n",
       "0         27.696111                     0                 1                 1   \n",
       "1         27.696111                     0                 1                 1   \n",
       "163       18.930226                     0                 2                 2   \n",
       "164       18.930226                     0                 2                 2   \n",
       "\n",
       "     in_feature region  raw_pvalue  cis_window_qvalue  genomewide_bh_qvalue  \\\n",
       "18         True   SMTG    0.009947           0.019893              0.586346   \n",
       "0          True   CAUD    0.000556           0.000556              0.016024   \n",
       "1          True   CAUD    0.000556           0.000556              0.016024   \n",
       "163        True   MDFG    0.008743           0.008745              0.190429   \n",
       "164        True   MDFG    0.008743           0.008745              0.190429   \n",
       "\n",
       "     permutation_significant  fdr10_threshold  in_original  r2_with_ld_tag  \\\n",
       "18                      True         0.032534        False             1.0   \n",
       "0                       True         0.005189        False             1.0   \n",
       "1                       True         0.005189        False             1.0   \n",
       "163                     True         0.024660        False             1.0   \n",
       "164                     True         0.024660        False             1.0   \n",
       "\n",
       "     ld_tag_chr  ld_tag_pos               source_gwas       source  locus_num  \\\n",
       "18            1   161186243    Alzheimers_Jansen_2018         GWAS        129   \n",
       "0             1   232528865  Chang_23andMe_Parkinsons  Nalls-Chang         22   \n",
       "1             1   232528865             Nalls_23andMe  Nalls-Chang         22   \n",
       "163          11    60251677    Alzheimers_Jansen_2018         GWAS         11   \n",
       "164          11    60251677    Alzheimers_Kunkle_2019         GWAS         11   \n",
       "\n",
       "     number_ld_buddies ld_tag_locus effect_allele noneffect_allele direction  \\\n",
       "18                   2  1_161186243             A                C         +   \n",
       "0                   14  1_232528865             T                C         +   \n",
       "1                   14  1_232528865             T                C         +   \n",
       "163                385  11_60251677             C                T         -   \n",
       "164                261  11_60251677             T                C         +   \n",
       "\n",
       "           pvalue  has_coloc  direct_atac_overlap_narrow_tissue_regions  \\\n",
       "18   5.580000e-10       True                                       True   \n",
       "0    7.480000e-04      False                                       True   \n",
       "1    7.480000e-04      False                                       True   \n",
       "163  2.830000e-13      False                                      False   \n",
       "164  5.910000e-15      False                                      False   \n",
       "\n",
       "    containing_atac_tissues_narrow_tissue_regions  \\\n",
       "18                            HIPP,MDFG,PARL,SMTG   \n",
       "0                             CAUD,PARL,PTMN,SMTG   \n",
       "1                             CAUD,PARL,PTMN,SMTG   \n",
       "163                                          none   \n",
       "164                                          none   \n",
       "\n",
       "    nearest_atac_tissue_narrow_tissue_regions  start_narrow_tissue_regions  \\\n",
       "18                        HIPP,MDFG,PARL,SMTG                    161185040   \n",
       "0                         CAUD,PARL,PTMN,SMTG                    232528497   \n",
       "1                         CAUD,PARL,PTMN,SMTG                    232528497   \n",
       "163                                      MDFG                     60264747   \n",
       "164                                      MDFG                     60264747   \n",
       "\n",
       "     end_narrow_tissue_regions  dist_narrow_tissue_regions  \\\n",
       "18                   161186430                       422.0   \n",
       "0                    232530050                       287.5   \n",
       "1                    232530050                       287.5   \n",
       "163                   60266009                     13070.0   \n",
       "164                   60266009                     13070.0   \n",
       "\n",
       "     direct_atac_overlap_broad_tissue_regions  \\\n",
       "18                                       True   \n",
       "0                                       False   \n",
       "1                                       False   \n",
       "163                                     False   \n",
       "164                                     False   \n",
       "\n",
       "    containing_atac_tissues_broad_tissue_regions  \\\n",
       "18                                          ALLO   \n",
       "0                                           none   \n",
       "1                                           none   \n",
       "163                                         none   \n",
       "164                                         none   \n",
       "\n",
       "    nearest_atac_tissue_broad_tissue_regions  start_broad_tissue_regions  \\\n",
       "18                                      ALLO                   161185897   \n",
       "0                                       STRI                   232528568   \n",
       "1                                       STRI                   232528568   \n",
       "163                                     ISOC                    60265062   \n",
       "164                                     ISOC                    60265062   \n",
       "\n",
       "     end_broad_tissue_regions  dist_broad_tissue_regions  \\\n",
       "18                  161186291                      149.0   \n",
       "0                   232528768                       97.0   \n",
       "1                   232528768                       97.0   \n",
       "163                  60265262                    13385.0   \n",
       "164                  60265262                    13385.0   \n",
       "\n",
       "     direct_atac_overlap_single_cell  \\\n",
       "18                              True   \n",
       "0                               True   \n",
       "1                               True   \n",
       "163                             True   \n",
       "164                             True   \n",
       "\n",
       "                   containing_atac_tissues_single_cell  \\\n",
       "18                                           microglia   \n",
       "0    excitatory_neurons,inhibitory_neurons,neurons_...   \n",
       "1    excitatory_neurons,inhibitory_neurons,neurons_...   \n",
       "163                       excitatory_neurons,microglia   \n",
       "164                       excitatory_neurons,microglia   \n",
       "\n",
       "                       nearest_atac_tissue_single_cell  start_single_cell  \\\n",
       "18                                           microglia          161185232   \n",
       "0    excitatory_neurons,inhibitory_neurons,neurons_...          232528552   \n",
       "1    excitatory_neurons,inhibitory_neurons,neurons_...          232528552   \n",
       "163                       excitatory_neurons,microglia           60251481   \n",
       "164                       excitatory_neurons,microglia           60251481   \n",
       "\n",
       "     end_single_cell  dist_single_cell  \n",
       "18         161186486             384.0  \n",
       "0          232529236              29.0  \n",
       "1          232529236              29.0  \n",
       "163         60252738             432.5  \n",
       "164         60252738             432.5  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "(276, 53)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "trimmed_sig_snps = sig_snps[['feature', 'rsid', 'chrom', 'snp_pos', 'ref', 'alt', 'allele_freq',\n",
    "                            'effect_size', 'mapping_error_rate', 'ref_allele_mapping_bias', 'overdispersion',\n",
    "                            'snp_id_within_region', 'num_feature_snps', 'num_testing_snps',\n",
    "                            'in_feature', 'region', 'raw_pvalue', 'cis_window_qvalue', 'genomewide_bh_qvalue',\n",
    "                            'permutation_significant', 'fdr10_threshold', 'in_original']]\n",
    "trimmed_adpd_stage3 = adpd_stage3[['rsid', 'r2_with_ld_tag', 'ld_tag_chr', 'ld_tag_pos', 'source_gwas', 'source',\n",
    "                                    'locus_num', 'number_ld_buddies', 'ld_tag_locus', 'effect_allele', 'noneffect_allele', 'direction',\n",
    "                                    'pvalue', 'has_coloc', 'direct_atac_overlap_narrow_tissue_regions',\n",
    "                                    'containing_atac_tissues_narrow_tissue_regions',\n",
    "                                    'nearest_atac_tissue_narrow_tissue_regions',\n",
    "                                    'start_narrow_tissue_regions',\n",
    "                                    'end_narrow_tissue_regions',\n",
    "                                    'dist_narrow_tissue_regions',\n",
    "                                    'direct_atac_overlap_broad_tissue_regions',\n",
    "                                    'containing_atac_tissues_broad_tissue_regions',\n",
    "                                    'nearest_atac_tissue_broad_tissue_regions',\n",
    "                                    'start_broad_tissue_regions',\n",
    "                                    'end_broad_tissue_regions',\n",
    "                                    'dist_broad_tissue_regions',\n",
    "                                    'direct_atac_overlap_single_cell',\n",
    "                                    'containing_atac_tissues_single_cell',\n",
    "                                    'nearest_atac_tissue_single_cell',\n",
    "                                    'start_single_cell',\n",
    "                                    'end_single_cell',\n",
    "                                    'dist_single_cell']]\n",
    "trimmed_adpd_stage3.drop_duplicates(inplace=True)\n",
    "trim_sig_merge = trimmed_sig_snps.merge(trimmed_adpd_stage3, on='rsid')\n",
    "trim_sig_merge.sort_values(by=['chrom', 'snp_pos', 'rsid', 'feature', 'region'], inplace=True)\n",
    "display(trim_sig_merge.head())\n",
    "display(trim_sig_merge.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>feature</th>\n",
       "      <th>rsid</th>\n",
       "      <th>chrom</th>\n",
       "      <th>snp_pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>allele_freq</th>\n",
       "      <th>effect_size</th>\n",
       "      <th>mapping_error_rate</th>\n",
       "      <th>ref_allele_mapping_bias</th>\n",
       "      <th>overdispersion</th>\n",
       "      <th>snp_id_within_region</th>\n",
       "      <th>num_feature_snps</th>\n",
       "      <th>num_testing_snps</th>\n",
       "      <th>in_feature</th>\n",
       "      <th>region</th>\n",
       "      <th>raw_pvalue</th>\n",
       "      <th>cis_window_qvalue</th>\n",
       "      <th>genomewide_bh_qvalue</th>\n",
       "      <th>permutation_significant</th>\n",
       "      <th>fdr10_threshold</th>\n",
       "      <th>in_original</th>\n",
       "      <th>r2_with_ld_tag</th>\n",
       "      <th>ld_tag_chr</th>\n",
       "      <th>ld_tag_pos</th>\n",
       "      <th>source_gwas</th>\n",
       "      <th>source</th>\n",
       "      <th>locus_num</th>\n",
       "      <th>number_ld_buddies</th>\n",
       "      <th>ld_tag_locus</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>noneffect_allele</th>\n",
       "      <th>direction</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>has_coloc</th>\n",
       "      <th>direct_atac_overlap_narrow_tissue_regions</th>\n",
       "      <th>containing_atac_tissues_narrow_tissue_regions</th>\n",
       "      <th>nearest_atac_tissue_narrow_tissue_regions</th>\n",
       "      <th>start_narrow_tissue_regions</th>\n",
       "      <th>end_narrow_tissue_regions</th>\n",
       "      <th>dist_narrow_tissue_regions</th>\n",
       "      <th>direct_atac_overlap_broad_tissue_regions</th>\n",
       "      <th>containing_atac_tissues_broad_tissue_regions</th>\n",
       "      <th>nearest_atac_tissue_broad_tissue_regions</th>\n",
       "      <th>start_broad_tissue_regions</th>\n",
       "      <th>end_broad_tissue_regions</th>\n",
       "      <th>dist_broad_tissue_regions</th>\n",
       "      <th>direct_atac_overlap_single_cell</th>\n",
       "      <th>containing_atac_tissues_single_cell</th>\n",
       "      <th>nearest_atac_tissue_single_cell</th>\n",
       "      <th>start_single_cell</th>\n",
       "      <th>end_single_cell</th>\n",
       "      <th>dist_single_cell</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1_161185016_161186504</td>\n",
       "      <td>rs11585858</td>\n",
       "      <td>chr1</td>\n",
       "      <td>161186243</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.552632</td>\n",
       "      <td>0.382894</td>\n",
       "      <td>0.000046</td>\n",
       "      <td>0.441369</td>\n",
       "      <td>40.915012</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>SMTG</td>\n",
       "      <td>0.009947</td>\n",
       "      <td>0.019893</td>\n",
       "      <td>0.586346</td>\n",
       "      <td>True</td>\n",
       "      <td>0.032534</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>161186243</td>\n",
       "      <td>Alzheimers_Jansen_2018</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>129</td>\n",
       "      <td>2</td>\n",
       "      <td>1_161186243</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>+</td>\n",
       "      <td>5.580000e-10</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>HIPP,MDFG,PARL,SMTG</td>\n",
       "      <td>HIPP,MDFG,PARL,SMTG</td>\n",
       "      <td>161185040</td>\n",
       "      <td>161186430</td>\n",
       "      <td>422.0</td>\n",
       "      <td>True</td>\n",
       "      <td>ALLO</td>\n",
       "      <td>ALLO</td>\n",
       "      <td>161185897</td>\n",
       "      <td>161186291</td>\n",
       "      <td>149.0</td>\n",
       "      <td>True</td>\n",
       "      <td>microglia</td>\n",
       "      <td>microglia</td>\n",
       "      <td>161185232</td>\n",
       "      <td>161186486</td>\n",
       "      <td>384.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1_232528515_232530038</td>\n",
       "      <td>rs10797576</td>\n",
       "      <td>chr1</td>\n",
       "      <td>232528865</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.281250</td>\n",
       "      <td>0.359946</td>\n",
       "      <td>0.000028</td>\n",
       "      <td>0.581322</td>\n",
       "      <td>27.696111</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>CAUD</td>\n",
       "      <td>0.000556</td>\n",
       "      <td>0.000556</td>\n",
       "      <td>0.016024</td>\n",
       "      <td>True</td>\n",
       "      <td>0.005189</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>232528865</td>\n",
       "      <td>Chang_23andMe_Parkinsons</td>\n",
       "      <td>Nalls-Chang</td>\n",
       "      <td>22</td>\n",
       "      <td>14</td>\n",
       "      <td>1_232528865</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>+</td>\n",
       "      <td>7.480000e-04</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>CAUD,PARL,PTMN,SMTG</td>\n",
       "      <td>CAUD,PARL,PTMN,SMTG</td>\n",
       "      <td>232528497</td>\n",
       "      <td>232530050</td>\n",
       "      <td>287.5</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>STRI</td>\n",
       "      <td>232528568</td>\n",
       "      <td>232528768</td>\n",
       "      <td>97.0</td>\n",
       "      <td>True</td>\n",
       "      <td>excitatory_neurons,inhibitory_neurons,neurons_...</td>\n",
       "      <td>excitatory_neurons,inhibitory_neurons,neurons_...</td>\n",
       "      <td>232528552</td>\n",
       "      <td>232529236</td>\n",
       "      <td>29.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>163</th>\n",
       "      <td>chr11_60251518_60251763</td>\n",
       "      <td>rs636317</td>\n",
       "      <td>chr11</td>\n",
       "      <td>60251677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.833333</td>\n",
       "      <td>0.387836</td>\n",
       "      <td>0.007736</td>\n",
       "      <td>0.491990</td>\n",
       "      <td>18.930226</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>MDFG</td>\n",
       "      <td>0.008743</td>\n",
       "      <td>0.008745</td>\n",
       "      <td>0.190429</td>\n",
       "      <td>True</td>\n",
       "      <td>0.024660</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>11</td>\n",
       "      <td>60251677</td>\n",
       "      <td>Alzheimers_Jansen_2018</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>11</td>\n",
       "      <td>385</td>\n",
       "      <td>11_60251677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>2.830000e-13</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>MDFG</td>\n",
       "      <td>60264747</td>\n",
       "      <td>60266009</td>\n",
       "      <td>13070.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>ISOC</td>\n",
       "      <td>60265062</td>\n",
       "      <td>60265262</td>\n",
       "      <td>13385.0</td>\n",
       "      <td>True</td>\n",
       "      <td>excitatory_neurons,microglia</td>\n",
       "      <td>excitatory_neurons,microglia</td>\n",
       "      <td>60251481</td>\n",
       "      <td>60252738</td>\n",
       "      <td>432.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>chr11_60251518_60251763</td>\n",
       "      <td>rs636317</td>\n",
       "      <td>chr11</td>\n",
       "      <td>60251677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.750000</td>\n",
       "      <td>0.287154</td>\n",
       "      <td>0.000079</td>\n",
       "      <td>0.377505</td>\n",
       "      <td>19.789926</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>MDTG</td>\n",
       "      <td>0.000024</td>\n",
       "      <td>0.000024</td>\n",
       "      <td>0.004054</td>\n",
       "      <td>True</td>\n",
       "      <td>0.025700</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>11</td>\n",
       "      <td>60251677</td>\n",
       "      <td>Alzheimers_Jansen_2018</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>11</td>\n",
       "      <td>385</td>\n",
       "      <td>11_60251677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>2.830000e-13</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>MDFG</td>\n",
       "      <td>60264747</td>\n",
       "      <td>60266009</td>\n",
       "      <td>13070.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>ISOC</td>\n",
       "      <td>60265062</td>\n",
       "      <td>60265262</td>\n",
       "      <td>13385.0</td>\n",
       "      <td>True</td>\n",
       "      <td>excitatory_neurons,microglia</td>\n",
       "      <td>excitatory_neurons,microglia</td>\n",
       "      <td>60251481</td>\n",
       "      <td>60252738</td>\n",
       "      <td>432.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>154</th>\n",
       "      <td>chr11_60251518_60251763</td>\n",
       "      <td>rs636317</td>\n",
       "      <td>chr11</td>\n",
       "      <td>60251677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.818182</td>\n",
       "      <td>0.251451</td>\n",
       "      <td>0.000382</td>\n",
       "      <td>0.482460</td>\n",
       "      <td>21.296903</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>PTMN</td>\n",
       "      <td>0.000045</td>\n",
       "      <td>0.000045</td>\n",
       "      <td>0.012513</td>\n",
       "      <td>True</td>\n",
       "      <td>0.030718</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>11</td>\n",
       "      <td>60251677</td>\n",
       "      <td>Alzheimers_Jansen_2018</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>11</td>\n",
       "      <td>385</td>\n",
       "      <td>11_60251677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>2.830000e-13</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>MDFG</td>\n",
       "      <td>60264747</td>\n",
       "      <td>60266009</td>\n",
       "      <td>13070.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>ISOC</td>\n",
       "      <td>60265062</td>\n",
       "      <td>60265262</td>\n",
       "      <td>13385.0</td>\n",
       "      <td>True</td>\n",
       "      <td>excitatory_neurons,microglia</td>\n",
       "      <td>excitatory_neurons,microglia</td>\n",
       "      <td>60251481</td>\n",
       "      <td>60252738</td>\n",
       "      <td>432.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      feature        rsid  chrom    snp_pos ref alt  \\\n",
       "18   chr1_161185016_161186504  rs11585858   chr1  161186243   C   A   \n",
       "0    chr1_232528515_232530038  rs10797576   chr1  232528865   C   T   \n",
       "163   chr11_60251518_60251763    rs636317  chr11   60251677   C   T   \n",
       "151   chr11_60251518_60251763    rs636317  chr11   60251677   C   T   \n",
       "154   chr11_60251518_60251763    rs636317  chr11   60251677   C   T   \n",
       "\n",
       "     allele_freq  effect_size  mapping_error_rate  ref_allele_mapping_bias  \\\n",
       "18      0.552632     0.382894            0.000046                 0.441369   \n",
       "0       0.281250     0.359946            0.000028                 0.581322   \n",
       "163     0.833333     0.387836            0.007736                 0.491990   \n",
       "151     0.750000     0.287154            0.000079                 0.377505   \n",
       "154     0.818182     0.251451            0.000382                 0.482460   \n",
       "\n",
       "     overdispersion  snp_id_within_region  num_feature_snps  num_testing_snps  \\\n",
       "18        40.915012                     1                 2                 2   \n",
       "0         27.696111                     0                 1                 1   \n",
       "163       18.930226                     0                 2                 2   \n",
       "151       19.789926                     0                 2                 2   \n",
       "154       21.296903                     0                 1                 1   \n",
       "\n",
       "     in_feature region  raw_pvalue  cis_window_qvalue  genomewide_bh_qvalue  \\\n",
       "18         True   SMTG    0.009947           0.019893              0.586346   \n",
       "0          True   CAUD    0.000556           0.000556              0.016024   \n",
       "163        True   MDFG    0.008743           0.008745              0.190429   \n",
       "151        True   MDTG    0.000024           0.000024              0.004054   \n",
       "154        True   PTMN    0.000045           0.000045              0.012513   \n",
       "\n",
       "     permutation_significant  fdr10_threshold  in_original  r2_with_ld_tag  \\\n",
       "18                      True         0.032534        False             1.0   \n",
       "0                       True         0.005189        False             1.0   \n",
       "163                     True         0.024660        False             1.0   \n",
       "151                     True         0.025700        False             1.0   \n",
       "154                     True         0.030718        False             1.0   \n",
       "\n",
       "     ld_tag_chr  ld_tag_pos               source_gwas       source  locus_num  \\\n",
       "18            1   161186243    Alzheimers_Jansen_2018         GWAS        129   \n",
       "0             1   232528865  Chang_23andMe_Parkinsons  Nalls-Chang         22   \n",
       "163          11    60251677    Alzheimers_Jansen_2018         GWAS         11   \n",
       "151          11    60251677    Alzheimers_Jansen_2018         GWAS         11   \n",
       "154          11    60251677    Alzheimers_Jansen_2018         GWAS         11   \n",
       "\n",
       "     number_ld_buddies ld_tag_locus effect_allele noneffect_allele direction  \\\n",
       "18                   2  1_161186243             A                C         +   \n",
       "0                   14  1_232528865             T                C         +   \n",
       "163                385  11_60251677             C                T         -   \n",
       "151                385  11_60251677             C                T         -   \n",
       "154                385  11_60251677             C                T         -   \n",
       "\n",
       "           pvalue  has_coloc  direct_atac_overlap_narrow_tissue_regions  \\\n",
       "18   5.580000e-10       True                                       True   \n",
       "0    7.480000e-04      False                                       True   \n",
       "163  2.830000e-13      False                                      False   \n",
       "151  2.830000e-13      False                                      False   \n",
       "154  2.830000e-13      False                                      False   \n",
       "\n",
       "    containing_atac_tissues_narrow_tissue_regions  \\\n",
       "18                            HIPP,MDFG,PARL,SMTG   \n",
       "0                             CAUD,PARL,PTMN,SMTG   \n",
       "163                                          none   \n",
       "151                                          none   \n",
       "154                                          none   \n",
       "\n",
       "    nearest_atac_tissue_narrow_tissue_regions  start_narrow_tissue_regions  \\\n",
       "18                        HIPP,MDFG,PARL,SMTG                    161185040   \n",
       "0                         CAUD,PARL,PTMN,SMTG                    232528497   \n",
       "163                                      MDFG                     60264747   \n",
       "151                                      MDFG                     60264747   \n",
       "154                                      MDFG                     60264747   \n",
       "\n",
       "     end_narrow_tissue_regions  dist_narrow_tissue_regions  \\\n",
       "18                   161186430                       422.0   \n",
       "0                    232530050                       287.5   \n",
       "163                   60266009                     13070.0   \n",
       "151                   60266009                     13070.0   \n",
       "154                   60266009                     13070.0   \n",
       "\n",
       "     direct_atac_overlap_broad_tissue_regions  \\\n",
       "18                                       True   \n",
       "0                                       False   \n",
       "163                                     False   \n",
       "151                                     False   \n",
       "154                                     False   \n",
       "\n",
       "    containing_atac_tissues_broad_tissue_regions  \\\n",
       "18                                          ALLO   \n",
       "0                                           none   \n",
       "163                                         none   \n",
       "151                                         none   \n",
       "154                                         none   \n",
       "\n",
       "    nearest_atac_tissue_broad_tissue_regions  start_broad_tissue_regions  \\\n",
       "18                                      ALLO                   161185897   \n",
       "0                                       STRI                   232528568   \n",
       "163                                     ISOC                    60265062   \n",
       "151                                     ISOC                    60265062   \n",
       "154                                     ISOC                    60265062   \n",
       "\n",
       "     end_broad_tissue_regions  dist_broad_tissue_regions  \\\n",
       "18                  161186291                      149.0   \n",
       "0                   232528768                       97.0   \n",
       "163                  60265262                    13385.0   \n",
       "151                  60265262                    13385.0   \n",
       "154                  60265262                    13385.0   \n",
       "\n",
       "     direct_atac_overlap_single_cell  \\\n",
       "18                              True   \n",
       "0                               True   \n",
       "163                             True   \n",
       "151                             True   \n",
       "154                             True   \n",
       "\n",
       "                   containing_atac_tissues_single_cell  \\\n",
       "18                                           microglia   \n",
       "0    excitatory_neurons,inhibitory_neurons,neurons_...   \n",
       "163                       excitatory_neurons,microglia   \n",
       "151                       excitatory_neurons,microglia   \n",
       "154                       excitatory_neurons,microglia   \n",
       "\n",
       "                       nearest_atac_tissue_single_cell  start_single_cell  \\\n",
       "18                                           microglia          161185232   \n",
       "0    excitatory_neurons,inhibitory_neurons,neurons_...          232528552   \n",
       "163                       excitatory_neurons,microglia           60251481   \n",
       "151                       excitatory_neurons,microglia           60251481   \n",
       "154                       excitatory_neurons,microglia           60251481   \n",
       "\n",
       "     end_single_cell  dist_single_cell  \n",
       "18         161186486             384.0  \n",
       "0          232529236              29.0  \n",
       "163         60252738             432.5  \n",
       "151         60252738             432.5  \n",
       "154         60252738             432.5  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "(209, 53)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "snp_dict = {}\n",
    "for ind,val in trim_sig_merge.iterrows():\n",
    "    if val['rsid'] in snp_dict:\n",
    "        snp_dict[val['rsid']].append((val['source_gwas'],val['pvalue']))\n",
    "    else:\n",
    "        snp_dict[val['rsid']] = []\n",
    "        snp_dict[val['rsid']].append((val['source_gwas'],val['pvalue']))\n",
    "for snp in snp_dict:\n",
    "    snp_dict[snp] = sorted(snp_dict[snp], key=lambda x: x[-1])\n",
    "for ind,val in trim_sig_merge.iterrows():\n",
    "    val['source_gwas'] = snp_dict[val['rsid']][0][0]\n",
    "    val['pvalue'] = snp_dict[val['rsid']][0][1]\n",
    "gwas_trim_sig_merge = trim_sig_merge.drop_duplicates(subset=['chrom', 'snp_pos', 'rsid', 'feature', 'region'])\n",
    "display(gwas_trim_sig_merge.head())\n",
    "display(gwas_trim_sig_merge.shape)\n",
    "gwas_trim_sig_merge.to_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/gwas_rasqual_significant_snps.csv', sep='\\t', index=False)\n",
    "unique_gwas_trim_sig_merge = gwas_trim_sig_merge.sort_values(by=['rsid', 'cis_window_qvalue'])\n",
    "unique_gwas_trim_sig_merge.drop_duplicates(subset='rsid', inplace=True)\n",
    "unique_gwas_trim_sig_merge.to_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/unique_gwas_rasqual_significant_snps.csv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsid</th>\n",
       "      <th>ism_pval</th>\n",
       "      <th>delta_pval</th>\n",
       "      <th>explain_pval</th>\n",
       "      <th>confidence</th>\n",
       "      <th>motif</th>\n",
       "      <th>diff_motif</th>\n",
       "      <th>fc_track</th>\n",
       "      <th>pval_track</th>\n",
       "      <th>annotation</th>\n",
       "      <th>cluster</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs2296887</td>\n",
       "      <td>0.021475</td>\n",
       "      <td>0.022689</td>\n",
       "      <td>0.125544</td>\n",
       "      <td>1</td>\n",
       "      <td>AACAGGCTCC</td>\n",
       "      <td>AACAGGC(T/C)CC</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>---</td>\n",
       "      <td>1,2,4,11,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs7900536</td>\n",
       "      <td>0.029275</td>\n",
       "      <td>0.030456</td>\n",
       "      <td>0.191728</td>\n",
       "      <td>1</td>\n",
       "      <td>CCTCTAGAGGGATGATTCA</td>\n",
       "      <td>CCTCTAGAGGGATGATTC(A/C)</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>---</td>\n",
       "      <td>1,2,7,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs2856653</td>\n",
       "      <td>0.028402</td>\n",
       "      <td>0.028299</td>\n",
       "      <td>0.134918</td>\n",
       "      <td>1</td>\n",
       "      <td>TCTCCCACGG</td>\n",
       "      <td>TCTCCCA(C/T)GG</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>---</td>\n",
       "      <td>2,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs11039225</td>\n",
       "      <td>0.046656</td>\n",
       "      <td>0.047686</td>\n",
       "      <td>0.999999</td>\n",
       "      <td>0</td>\n",
       "      <td>TGGAAGTCCT</td>\n",
       "      <td>TGGA(A/G)GTCCT</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>---</td>\n",
       "      <td>19,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs636317</td>\n",
       "      <td>0.000131</td>\n",
       "      <td>0.000235</td>\n",
       "      <td>0.027467</td>\n",
       "      <td>2</td>\n",
       "      <td>TGCTGCCATCTGCTGGGAA</td>\n",
       "      <td>TGCTGC(C/T)ATCTGCTGGGAA</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>MS4A4A, microglia specific</td>\n",
       "      <td>1,4,19,20,24,</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         rsid  ism_pval  delta_pval  explain_pval  confidence  \\\n",
       "0   rs2296887  0.021475    0.022689      0.125544           1   \n",
       "1   rs7900536  0.029275    0.030456      0.191728           1   \n",
       "2   rs2856653  0.028402    0.028299      0.134918           1   \n",
       "3  rs11039225  0.046656    0.047686      0.999999           0   \n",
       "4    rs636317  0.000131    0.000235      0.027467           2   \n",
       "\n",
       "                 motif               diff_motif  \\\n",
       "0           AACAGGCTCC           AACAGGC(T/C)CC   \n",
       "1  CCTCTAGAGGGATGATTCA  CCTCTAGAGGGATGATTC(A/C)   \n",
       "2           TCTCCCACGG           TCTCCCA(C/T)GG   \n",
       "3           TGGAAGTCCT           TGGA(A/G)GTCCT   \n",
       "4  TGCTGCCATCTGCTGGGAA  TGCTGC(C/T)ATCTGCTGGGAA   \n",
       "\n",
       "                                            fc_track  \\\n",
       "0  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "1  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "2  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "3  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "4  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "\n",
       "                                          pval_track  \\\n",
       "0  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "1  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "2  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "3  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "4  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "\n",
       "                   annotation        cluster  \n",
       "0                         ---      1,2,4,11,  \n",
       "1                         ---         1,2,7,  \n",
       "2                         ---             2,  \n",
       "3                         ---            19,  \n",
       "4  MS4A4A, microglia specific  1,4,19,20,24,  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "(123, 11)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "orig_sig_snps = pd.read_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/snps/orig_sig_snps.tsv', sep='\\t')\n",
    "orig_sig_snps = orig_sig_snps[['rsid', 'ism_pval', 'delta_pval', 'explain_pval', 'confidence',\n",
    "                              'motif', 'diff_motif', 'fc_track', 'pval_track', 'annotation', 'cluster']]\n",
    "display(orig_sig_snps.head())\n",
    "display(orig_sig_snps.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>feature</th>\n",
       "      <th>rsid</th>\n",
       "      <th>chrom</th>\n",
       "      <th>snp_pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>allele_freq</th>\n",
       "      <th>effect_size</th>\n",
       "      <th>mapping_error_rate</th>\n",
       "      <th>ref_allele_mapping_bias</th>\n",
       "      <th>overdispersion</th>\n",
       "      <th>snp_id_within_region</th>\n",
       "      <th>num_feature_snps</th>\n",
       "      <th>num_testing_snps</th>\n",
       "      <th>in_feature</th>\n",
       "      <th>region</th>\n",
       "      <th>raw_pvalue</th>\n",
       "      <th>cis_window_qvalue</th>\n",
       "      <th>genomewide_bh_qvalue</th>\n",
       "      <th>permutation_significant</th>\n",
       "      <th>fdr10_threshold</th>\n",
       "      <th>in_original</th>\n",
       "      <th>r2_with_ld_tag</th>\n",
       "      <th>ld_tag_chr</th>\n",
       "      <th>ld_tag_pos</th>\n",
       "      <th>source_gwas</th>\n",
       "      <th>source</th>\n",
       "      <th>locus_num</th>\n",
       "      <th>number_ld_buddies</th>\n",
       "      <th>ld_tag_locus</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>noneffect_allele</th>\n",
       "      <th>direction</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>has_coloc</th>\n",
       "      <th>direct_atac_overlap_narrow_tissue_regions</th>\n",
       "      <th>containing_atac_tissues_narrow_tissue_regions</th>\n",
       "      <th>nearest_atac_tissue_narrow_tissue_regions</th>\n",
       "      <th>start_narrow_tissue_regions</th>\n",
       "      <th>end_narrow_tissue_regions</th>\n",
       "      <th>dist_narrow_tissue_regions</th>\n",
       "      <th>direct_atac_overlap_broad_tissue_regions</th>\n",
       "      <th>containing_atac_tissues_broad_tissue_regions</th>\n",
       "      <th>nearest_atac_tissue_broad_tissue_regions</th>\n",
       "      <th>start_broad_tissue_regions</th>\n",
       "      <th>end_broad_tissue_regions</th>\n",
       "      <th>dist_broad_tissue_regions</th>\n",
       "      <th>direct_atac_overlap_single_cell</th>\n",
       "      <th>containing_atac_tissues_single_cell</th>\n",
       "      <th>nearest_atac_tissue_single_cell</th>\n",
       "      <th>start_single_cell</th>\n",
       "      <th>end_single_cell</th>\n",
       "      <th>dist_single_cell</th>\n",
       "      <th>ism_pval</th>\n",
       "      <th>delta_pval</th>\n",
       "      <th>explain_pval</th>\n",
       "      <th>confidence</th>\n",
       "      <th>motif</th>\n",
       "      <th>diff_motif</th>\n",
       "      <th>fc_track</th>\n",
       "      <th>pval_track</th>\n",
       "      <th>annotation</th>\n",
       "      <th>cluster</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr11_60251518_60251763</td>\n",
       "      <td>rs636317</td>\n",
       "      <td>chr11</td>\n",
       "      <td>60251677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.833333</td>\n",
       "      <td>0.387836</td>\n",
       "      <td>0.007736</td>\n",
       "      <td>0.491990</td>\n",
       "      <td>18.930226</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>MDFG</td>\n",
       "      <td>0.008743</td>\n",
       "      <td>0.008745</td>\n",
       "      <td>0.190429</td>\n",
       "      <td>True</td>\n",
       "      <td>0.024660</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>11</td>\n",
       "      <td>60251677</td>\n",
       "      <td>Alzheimers_Jansen_2018</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>11</td>\n",
       "      <td>385</td>\n",
       "      <td>11_60251677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>2.830000e-13</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>MDFG</td>\n",
       "      <td>60264747</td>\n",
       "      <td>60266009</td>\n",
       "      <td>13070.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>ISOC</td>\n",
       "      <td>60265062</td>\n",
       "      <td>60265262</td>\n",
       "      <td>13385.0</td>\n",
       "      <td>True</td>\n",
       "      <td>excitatory_neurons,microglia</td>\n",
       "      <td>excitatory_neurons,microglia</td>\n",
       "      <td>60251481</td>\n",
       "      <td>60252738</td>\n",
       "      <td>432.5</td>\n",
       "      <td>0.000131</td>\n",
       "      <td>0.000235</td>\n",
       "      <td>0.027467</td>\n",
       "      <td>2</td>\n",
       "      <td>TGCTGCCATCTGCTGGGAA</td>\n",
       "      <td>TGCTGC(C/T)ATCTGCTGGGAA</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>MS4A4A, microglia specific</td>\n",
       "      <td>1,4,19,20,24,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr11_60251518_60251763</td>\n",
       "      <td>rs636317</td>\n",
       "      <td>chr11</td>\n",
       "      <td>60251677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.750000</td>\n",
       "      <td>0.287154</td>\n",
       "      <td>0.000079</td>\n",
       "      <td>0.377505</td>\n",
       "      <td>19.789926</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>MDTG</td>\n",
       "      <td>0.000024</td>\n",
       "      <td>0.000024</td>\n",
       "      <td>0.004054</td>\n",
       "      <td>True</td>\n",
       "      <td>0.025700</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>11</td>\n",
       "      <td>60251677</td>\n",
       "      <td>Alzheimers_Jansen_2018</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>11</td>\n",
       "      <td>385</td>\n",
       "      <td>11_60251677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>2.830000e-13</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>MDFG</td>\n",
       "      <td>60264747</td>\n",
       "      <td>60266009</td>\n",
       "      <td>13070.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>ISOC</td>\n",
       "      <td>60265062</td>\n",
       "      <td>60265262</td>\n",
       "      <td>13385.0</td>\n",
       "      <td>True</td>\n",
       "      <td>excitatory_neurons,microglia</td>\n",
       "      <td>excitatory_neurons,microglia</td>\n",
       "      <td>60251481</td>\n",
       "      <td>60252738</td>\n",
       "      <td>432.5</td>\n",
       "      <td>0.000131</td>\n",
       "      <td>0.000235</td>\n",
       "      <td>0.027467</td>\n",
       "      <td>2</td>\n",
       "      <td>TGCTGCCATCTGCTGGGAA</td>\n",
       "      <td>TGCTGC(C/T)ATCTGCTGGGAA</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>MS4A4A, microglia specific</td>\n",
       "      <td>1,4,19,20,24,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr11_60251518_60251763</td>\n",
       "      <td>rs636317</td>\n",
       "      <td>chr11</td>\n",
       "      <td>60251677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.818182</td>\n",
       "      <td>0.251451</td>\n",
       "      <td>0.000382</td>\n",
       "      <td>0.482460</td>\n",
       "      <td>21.296903</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>PTMN</td>\n",
       "      <td>0.000045</td>\n",
       "      <td>0.000045</td>\n",
       "      <td>0.012513</td>\n",
       "      <td>True</td>\n",
       "      <td>0.030718</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>11</td>\n",
       "      <td>60251677</td>\n",
       "      <td>Alzheimers_Jansen_2018</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>11</td>\n",
       "      <td>385</td>\n",
       "      <td>11_60251677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>2.830000e-13</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>MDFG</td>\n",
       "      <td>60264747</td>\n",
       "      <td>60266009</td>\n",
       "      <td>13070.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>ISOC</td>\n",
       "      <td>60265062</td>\n",
       "      <td>60265262</td>\n",
       "      <td>13385.0</td>\n",
       "      <td>True</td>\n",
       "      <td>excitatory_neurons,microglia</td>\n",
       "      <td>excitatory_neurons,microglia</td>\n",
       "      <td>60251481</td>\n",
       "      <td>60252738</td>\n",
       "      <td>432.5</td>\n",
       "      <td>0.000131</td>\n",
       "      <td>0.000235</td>\n",
       "      <td>0.027467</td>\n",
       "      <td>2</td>\n",
       "      <td>TGCTGCCATCTGCTGGGAA</td>\n",
       "      <td>TGCTGC(C/T)ATCTGCTGGGAA</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>MS4A4A, microglia specific</td>\n",
       "      <td>1,4,19,20,24,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr11_60251518_60251763</td>\n",
       "      <td>rs636317</td>\n",
       "      <td>chr11</td>\n",
       "      <td>60251677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.947368</td>\n",
       "      <td>0.349554</td>\n",
       "      <td>0.023827</td>\n",
       "      <td>0.454270</td>\n",
       "      <td>14.676850</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>SMTG</td>\n",
       "      <td>0.002259</td>\n",
       "      <td>0.002259</td>\n",
       "      <td>0.280309</td>\n",
       "      <td>True</td>\n",
       "      <td>0.032534</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>11</td>\n",
       "      <td>60251677</td>\n",
       "      <td>Alzheimers_Jansen_2018</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>11</td>\n",
       "      <td>385</td>\n",
       "      <td>11_60251677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>2.830000e-13</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>MDFG</td>\n",
       "      <td>60264747</td>\n",
       "      <td>60266009</td>\n",
       "      <td>13070.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>ISOC</td>\n",
       "      <td>60265062</td>\n",
       "      <td>60265262</td>\n",
       "      <td>13385.0</td>\n",
       "      <td>True</td>\n",
       "      <td>excitatory_neurons,microglia</td>\n",
       "      <td>excitatory_neurons,microglia</td>\n",
       "      <td>60251481</td>\n",
       "      <td>60252738</td>\n",
       "      <td>432.5</td>\n",
       "      <td>0.000131</td>\n",
       "      <td>0.000235</td>\n",
       "      <td>0.027467</td>\n",
       "      <td>2</td>\n",
       "      <td>TGCTGCCATCTGCTGGGAA</td>\n",
       "      <td>TGCTGC(C/T)ATCTGCTGGGAA</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>MS4A4A, microglia specific</td>\n",
       "      <td>1,4,19,20,24,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr11_60251518_60251763</td>\n",
       "      <td>rs636317</td>\n",
       "      <td>chr11</td>\n",
       "      <td>60251677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.678571</td>\n",
       "      <td>0.318225</td>\n",
       "      <td>0.016438</td>\n",
       "      <td>0.425320</td>\n",
       "      <td>10.754400</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>SUNI</td>\n",
       "      <td>0.007008</td>\n",
       "      <td>0.007008</td>\n",
       "      <td>0.165306</td>\n",
       "      <td>True</td>\n",
       "      <td>0.043506</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>11</td>\n",
       "      <td>60251677</td>\n",
       "      <td>Alzheimers_Jansen_2018</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>11</td>\n",
       "      <td>385</td>\n",
       "      <td>11_60251677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>2.830000e-13</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>MDFG</td>\n",
       "      <td>60264747</td>\n",
       "      <td>60266009</td>\n",
       "      <td>13070.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>ISOC</td>\n",
       "      <td>60265062</td>\n",
       "      <td>60265262</td>\n",
       "      <td>13385.0</td>\n",
       "      <td>True</td>\n",
       "      <td>excitatory_neurons,microglia</td>\n",
       "      <td>excitatory_neurons,microglia</td>\n",
       "      <td>60251481</td>\n",
       "      <td>60252738</td>\n",
       "      <td>432.5</td>\n",
       "      <td>0.000131</td>\n",
       "      <td>0.000235</td>\n",
       "      <td>0.027467</td>\n",
       "      <td>2</td>\n",
       "      <td>TGCTGCCATCTGCTGGGAA</td>\n",
       "      <td>TGCTGC(C/T)ATCTGCTGGGAA</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>MS4A4A, microglia specific</td>\n",
       "      <td>1,4,19,20,24,</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   feature      rsid  chrom   snp_pos ref alt  allele_freq  \\\n",
       "0  chr11_60251518_60251763  rs636317  chr11  60251677   C   T     0.833333   \n",
       "1  chr11_60251518_60251763  rs636317  chr11  60251677   C   T     0.750000   \n",
       "2  chr11_60251518_60251763  rs636317  chr11  60251677   C   T     0.818182   \n",
       "3  chr11_60251518_60251763  rs636317  chr11  60251677   C   T     0.947368   \n",
       "4  chr11_60251518_60251763  rs636317  chr11  60251677   C   T     0.678571   \n",
       "\n",
       "   effect_size  mapping_error_rate  ref_allele_mapping_bias  overdispersion  \\\n",
       "0     0.387836            0.007736                 0.491990       18.930226   \n",
       "1     0.287154            0.000079                 0.377505       19.789926   \n",
       "2     0.251451            0.000382                 0.482460       21.296903   \n",
       "3     0.349554            0.023827                 0.454270       14.676850   \n",
       "4     0.318225            0.016438                 0.425320       10.754400   \n",
       "\n",
       "   snp_id_within_region  num_feature_snps  num_testing_snps  in_feature  \\\n",
       "0                     0                 2                 2        True   \n",
       "1                     0                 2                 2        True   \n",
       "2                     0                 1                 1        True   \n",
       "3                     0                 1                 1        True   \n",
       "4                     0                 2                 1        True   \n",
       "\n",
       "  region  raw_pvalue  cis_window_qvalue  genomewide_bh_qvalue  \\\n",
       "0   MDFG    0.008743           0.008745              0.190429   \n",
       "1   MDTG    0.000024           0.000024              0.004054   \n",
       "2   PTMN    0.000045           0.000045              0.012513   \n",
       "3   SMTG    0.002259           0.002259              0.280309   \n",
       "4   SUNI    0.007008           0.007008              0.165306   \n",
       "\n",
       "   permutation_significant  fdr10_threshold  in_original  r2_with_ld_tag  \\\n",
       "0                     True         0.024660        False             1.0   \n",
       "1                     True         0.025700        False             1.0   \n",
       "2                     True         0.030718        False             1.0   \n",
       "3                     True         0.032534        False             1.0   \n",
       "4                     True         0.043506        False             1.0   \n",
       "\n",
       "   ld_tag_chr  ld_tag_pos             source_gwas source  locus_num  \\\n",
       "0          11    60251677  Alzheimers_Jansen_2018   GWAS         11   \n",
       "1          11    60251677  Alzheimers_Jansen_2018   GWAS         11   \n",
       "2          11    60251677  Alzheimers_Jansen_2018   GWAS         11   \n",
       "3          11    60251677  Alzheimers_Jansen_2018   GWAS         11   \n",
       "4          11    60251677  Alzheimers_Jansen_2018   GWAS         11   \n",
       "\n",
       "   number_ld_buddies ld_tag_locus effect_allele noneffect_allele direction  \\\n",
       "0                385  11_60251677             C                T         -   \n",
       "1                385  11_60251677             C                T         -   \n",
       "2                385  11_60251677             C                T         -   \n",
       "3                385  11_60251677             C                T         -   \n",
       "4                385  11_60251677             C                T         -   \n",
       "\n",
       "         pvalue  has_coloc  direct_atac_overlap_narrow_tissue_regions  \\\n",
       "0  2.830000e-13      False                                      False   \n",
       "1  2.830000e-13      False                                      False   \n",
       "2  2.830000e-13      False                                      False   \n",
       "3  2.830000e-13      False                                      False   \n",
       "4  2.830000e-13      False                                      False   \n",
       "\n",
       "  containing_atac_tissues_narrow_tissue_regions  \\\n",
       "0                                          none   \n",
       "1                                          none   \n",
       "2                                          none   \n",
       "3                                          none   \n",
       "4                                          none   \n",
       "\n",
       "  nearest_atac_tissue_narrow_tissue_regions  start_narrow_tissue_regions  \\\n",
       "0                                      MDFG                     60264747   \n",
       "1                                      MDFG                     60264747   \n",
       "2                                      MDFG                     60264747   \n",
       "3                                      MDFG                     60264747   \n",
       "4                                      MDFG                     60264747   \n",
       "\n",
       "   end_narrow_tissue_regions  dist_narrow_tissue_regions  \\\n",
       "0                   60266009                     13070.0   \n",
       "1                   60266009                     13070.0   \n",
       "2                   60266009                     13070.0   \n",
       "3                   60266009                     13070.0   \n",
       "4                   60266009                     13070.0   \n",
       "\n",
       "   direct_atac_overlap_broad_tissue_regions  \\\n",
       "0                                     False   \n",
       "1                                     False   \n",
       "2                                     False   \n",
       "3                                     False   \n",
       "4                                     False   \n",
       "\n",
       "  containing_atac_tissues_broad_tissue_regions  \\\n",
       "0                                         none   \n",
       "1                                         none   \n",
       "2                                         none   \n",
       "3                                         none   \n",
       "4                                         none   \n",
       "\n",
       "  nearest_atac_tissue_broad_tissue_regions  start_broad_tissue_regions  \\\n",
       "0                                     ISOC                    60265062   \n",
       "1                                     ISOC                    60265062   \n",
       "2                                     ISOC                    60265062   \n",
       "3                                     ISOC                    60265062   \n",
       "4                                     ISOC                    60265062   \n",
       "\n",
       "   end_broad_tissue_regions  dist_broad_tissue_regions  \\\n",
       "0                  60265262                    13385.0   \n",
       "1                  60265262                    13385.0   \n",
       "2                  60265262                    13385.0   \n",
       "3                  60265262                    13385.0   \n",
       "4                  60265262                    13385.0   \n",
       "\n",
       "   direct_atac_overlap_single_cell containing_atac_tissues_single_cell  \\\n",
       "0                             True        excitatory_neurons,microglia   \n",
       "1                             True        excitatory_neurons,microglia   \n",
       "2                             True        excitatory_neurons,microglia   \n",
       "3                             True        excitatory_neurons,microglia   \n",
       "4                             True        excitatory_neurons,microglia   \n",
       "\n",
       "  nearest_atac_tissue_single_cell  start_single_cell  end_single_cell  \\\n",
       "0    excitatory_neurons,microglia           60251481         60252738   \n",
       "1    excitatory_neurons,microglia           60251481         60252738   \n",
       "2    excitatory_neurons,microglia           60251481         60252738   \n",
       "3    excitatory_neurons,microglia           60251481         60252738   \n",
       "4    excitatory_neurons,microglia           60251481         60252738   \n",
       "\n",
       "   dist_single_cell  ism_pval  delta_pval  explain_pval  confidence  \\\n",
       "0             432.5  0.000131    0.000235      0.027467           2   \n",
       "1             432.5  0.000131    0.000235      0.027467           2   \n",
       "2             432.5  0.000131    0.000235      0.027467           2   \n",
       "3             432.5  0.000131    0.000235      0.027467           2   \n",
       "4             432.5  0.000131    0.000235      0.027467           2   \n",
       "\n",
       "                 motif               diff_motif  \\\n",
       "0  TGCTGCCATCTGCTGGGAA  TGCTGC(C/T)ATCTGCTGGGAA   \n",
       "1  TGCTGCCATCTGCTGGGAA  TGCTGC(C/T)ATCTGCTGGGAA   \n",
       "2  TGCTGCCATCTGCTGGGAA  TGCTGC(C/T)ATCTGCTGGGAA   \n",
       "3  TGCTGCCATCTGCTGGGAA  TGCTGC(C/T)ATCTGCTGGGAA   \n",
       "4  TGCTGCCATCTGCTGGGAA  TGCTGC(C/T)ATCTGCTGGGAA   \n",
       "\n",
       "                                            fc_track  \\\n",
       "0  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "1  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "2  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "3  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "4  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "\n",
       "                                          pval_track  \\\n",
       "0  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "1  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "2  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "3  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "4  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "\n",
       "                   annotation        cluster  \n",
       "0  MS4A4A, microglia specific  1,4,19,20,24,  \n",
       "1  MS4A4A, microglia specific  1,4,19,20,24,  \n",
       "2  MS4A4A, microglia specific  1,4,19,20,24,  \n",
       "3  MS4A4A, microglia specific  1,4,19,20,24,  \n",
       "4  MS4A4A, microglia specific  1,4,19,20,24,  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "(40, 63)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "orig_sig_merge = gwas_trim_sig_merge.merge(orig_sig_snps, on='rsid')\n",
    "display(orig_sig_merge.head())\n",
    "display(orig_sig_merge.shape)\n",
    "orig_sig_merge.to_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/ml_gwas_rasqual_significant_snps.csv', sep='\\t', index=False)\n",
    "unique_orig_sig_merge = orig_sig_merge.sort_values(by=['rsid'])\n",
    "unique_orig_sig_merge.drop_duplicates(subset='rsid', inplace=True)\n",
    "unique_orig_sig_merge.sort_values(by=['confidence', 'feature'], ascending=False, inplace=True)\n",
    "unique_orig_sig_merge.to_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/unique_ml_gwas_rasqual_significant_snps.csv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>feature</th>\n",
       "      <th>rsid</th>\n",
       "      <th>chrom</th>\n",
       "      <th>snp_pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>allele_freq</th>\n",
       "      <th>effect_size</th>\n",
       "      <th>mapping_error_rate</th>\n",
       "      <th>ref_allele_mapping_bias</th>\n",
       "      <th>overdispersion</th>\n",
       "      <th>snp_id_within_region</th>\n",
       "      <th>num_feature_snps</th>\n",
       "      <th>num_testing_snps</th>\n",
       "      <th>in_feature</th>\n",
       "      <th>region</th>\n",
       "      <th>raw_pvalue</th>\n",
       "      <th>cis_window_qvalue</th>\n",
       "      <th>genomewide_bh_qvalue</th>\n",
       "      <th>permutation_significant</th>\n",
       "      <th>fdr10_threshold</th>\n",
       "      <th>in_original</th>\n",
       "      <th>r2_with_ld_tag</th>\n",
       "      <th>ld_tag_chr</th>\n",
       "      <th>ld_tag_pos</th>\n",
       "      <th>source_gwas</th>\n",
       "      <th>source</th>\n",
       "      <th>locus_num</th>\n",
       "      <th>number_ld_buddies</th>\n",
       "      <th>ld_tag_locus</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>noneffect_allele</th>\n",
       "      <th>direction</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>has_coloc</th>\n",
       "      <th>direct_atac_overlap_narrow_tissue_regions</th>\n",
       "      <th>containing_atac_tissues_narrow_tissue_regions</th>\n",
       "      <th>nearest_atac_tissue_narrow_tissue_regions</th>\n",
       "      <th>start_narrow_tissue_regions</th>\n",
       "      <th>end_narrow_tissue_regions</th>\n",
       "      <th>dist_narrow_tissue_regions</th>\n",
       "      <th>direct_atac_overlap_broad_tissue_regions</th>\n",
       "      <th>containing_atac_tissues_broad_tissue_regions</th>\n",
       "      <th>nearest_atac_tissue_broad_tissue_regions</th>\n",
       "      <th>start_broad_tissue_regions</th>\n",
       "      <th>end_broad_tissue_regions</th>\n",
       "      <th>dist_broad_tissue_regions</th>\n",
       "      <th>direct_atac_overlap_single_cell</th>\n",
       "      <th>containing_atac_tissues_single_cell</th>\n",
       "      <th>nearest_atac_tissue_single_cell</th>\n",
       "      <th>start_single_cell</th>\n",
       "      <th>end_single_cell</th>\n",
       "      <th>dist_single_cell</th>\n",
       "      <th>ism_pval</th>\n",
       "      <th>delta_pval</th>\n",
       "      <th>explain_pval</th>\n",
       "      <th>confidence</th>\n",
       "      <th>motif</th>\n",
       "      <th>diff_motif</th>\n",
       "      <th>fc_track</th>\n",
       "      <th>pval_track</th>\n",
       "      <th>annotation</th>\n",
       "      <th>cluster</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr17_4901692_4902448</td>\n",
       "      <td>rs79436576</td>\n",
       "      <td>chr17</td>\n",
       "      <td>4901872</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.090909</td>\n",
       "      <td>0.673282</td>\n",
       "      <td>0.005003</td>\n",
       "      <td>0.514530</td>\n",
       "      <td>33.035472</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>True</td>\n",
       "      <td>PTMN</td>\n",
       "      <td>0.019599</td>\n",
       "      <td>0.019988</td>\n",
       "      <td>0.348312</td>\n",
       "      <td>True</td>\n",
       "      <td>0.030718</td>\n",
       "      <td>False</td>\n",
       "      <td>0.873516</td>\n",
       "      <td>17</td>\n",
       "      <td>4860256</td>\n",
       "      <td>Alzheimers_Kunkle_2019</td>\n",
       "      <td>LD</td>\n",
       "      <td>55</td>\n",
       "      <td>80</td>\n",
       "      <td>17_4860256</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>+</td>\n",
       "      <td>3.060000e-04</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>HIPP,MDFG,PARL,SMTG</td>\n",
       "      <td>HIPP,MDFG,PARL,SMTG</td>\n",
       "      <td>4901724</td>\n",
       "      <td>4903885</td>\n",
       "      <td>904.0</td>\n",
       "      <td>True</td>\n",
       "      <td>ISOC</td>\n",
       "      <td>ISOC</td>\n",
       "      <td>4901711</td>\n",
       "      <td>4901911</td>\n",
       "      <td>61.0</td>\n",
       "      <td>True</td>\n",
       "      <td>astrocytes,microglia,oligodendrocytes</td>\n",
       "      <td>astrocytes,microglia,oligodendrocytes</td>\n",
       "      <td>4901620</td>\n",
       "      <td>4902717</td>\n",
       "      <td>296.5</td>\n",
       "      <td>0.004616</td>\n",
       "      <td>0.004449</td>\n",
       "      <td>0.041519</td>\n",
       "      <td>2</td>\n",
       "      <td>TGGCCCCGCCCC</td>\n",
       "      <td>TGGCC(C/T)CGCCCC</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>Many possible linked genes. ARRB2 - arrestin B...</td>\n",
       "      <td>6,11,13,14,15,19,20,21,22,24,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr17_45825137_45827395</td>\n",
       "      <td>rs62057150</td>\n",
       "      <td>chr17</td>\n",
       "      <td>45826182</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.555556</td>\n",
       "      <td>0.662942</td>\n",
       "      <td>0.135228</td>\n",
       "      <td>0.132052</td>\n",
       "      <td>2.083686</td>\n",
       "      <td>8</td>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "      <td>True</td>\n",
       "      <td>PARL</td>\n",
       "      <td>0.009645</td>\n",
       "      <td>0.015776</td>\n",
       "      <td>0.448529</td>\n",
       "      <td>True</td>\n",
       "      <td>0.031533</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>17</td>\n",
       "      <td>45826182</td>\n",
       "      <td>23andme_PD_hg38</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>54</td>\n",
       "      <td>2496</td>\n",
       "      <td>17_45826182</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>1.330000e-22</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>CAUD,HIPP,MDFG,PARL,PTMN,SMTG,SUNI</td>\n",
       "      <td>CAUD,HIPP,MDFG,PARL,PTMN,SMTG,SUNI</td>\n",
       "      <td>45825194</td>\n",
       "      <td>45826634</td>\n",
       "      <td>268.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>ALLO</td>\n",
       "      <td>45826333</td>\n",
       "      <td>45826558</td>\n",
       "      <td>151.0</td>\n",
       "      <td>True</td>\n",
       "      <td>oligodendrocytes</td>\n",
       "      <td>oligodendrocytes</td>\n",
       "      <td>45825230</td>\n",
       "      <td>45826777</td>\n",
       "      <td>178.5</td>\n",
       "      <td>0.042619</td>\n",
       "      <td>0.045282</td>\n",
       "      <td>0.031732</td>\n",
       "      <td>2</td>\n",
       "      <td>CCATTGGTCG</td>\n",
       "      <td>CCAT(T/C)GGTCG</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>MAPT Locus</td>\n",
       "      <td>23,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr17_45702992_45704386</td>\n",
       "      <td>rs968027</td>\n",
       "      <td>chr17</td>\n",
       "      <td>45703884</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.906250</td>\n",
       "      <td>0.663832</td>\n",
       "      <td>0.083686</td>\n",
       "      <td>0.354419</td>\n",
       "      <td>2.166129</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>True</td>\n",
       "      <td>CAUD</td>\n",
       "      <td>0.000616</td>\n",
       "      <td>0.000715</td>\n",
       "      <td>0.018553</td>\n",
       "      <td>True</td>\n",
       "      <td>0.005189</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>17</td>\n",
       "      <td>45703884</td>\n",
       "      <td>23andme_PD_hg38</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>54</td>\n",
       "      <td>2496</td>\n",
       "      <td>17_45703884</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>2.070000e-22</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>CAUD,PARL,PTMN,SMTG</td>\n",
       "      <td>CAUD,PARL,PTMN,SMTG</td>\n",
       "      <td>45702996</td>\n",
       "      <td>45706091</td>\n",
       "      <td>659.5</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>ISOC</td>\n",
       "      <td>45703478</td>\n",
       "      <td>45703678</td>\n",
       "      <td>206.0</td>\n",
       "      <td>True</td>\n",
       "      <td>excitatory_neurons,inhibitory_neurons</td>\n",
       "      <td>excitatory_neurons,inhibitory_neurons</td>\n",
       "      <td>45703290</td>\n",
       "      <td>45704925</td>\n",
       "      <td>223.5</td>\n",
       "      <td>0.007962</td>\n",
       "      <td>0.008333</td>\n",
       "      <td>0.041340</td>\n",
       "      <td>2</td>\n",
       "      <td>TGACTAA</td>\n",
       "      <td>TGAC(T/C)AA</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>MAPT Locus</td>\n",
       "      <td>2,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr16_31117024_31117373</td>\n",
       "      <td>rs4889619</td>\n",
       "      <td>chr16</td>\n",
       "      <td>31117294</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.888889</td>\n",
       "      <td>0.611059</td>\n",
       "      <td>0.008871</td>\n",
       "      <td>0.476374</td>\n",
       "      <td>8.603434</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>PARL</td>\n",
       "      <td>0.022687</td>\n",
       "      <td>0.029441</td>\n",
       "      <td>0.572968</td>\n",
       "      <td>True</td>\n",
       "      <td>0.031533</td>\n",
       "      <td>False</td>\n",
       "      <td>0.919429</td>\n",
       "      <td>16</td>\n",
       "      <td>31110472</td>\n",
       "      <td>Chang_23andMe_Parkinsons</td>\n",
       "      <td>LD</td>\n",
       "      <td>48</td>\n",
       "      <td>47</td>\n",
       "      <td>16_31110472</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>+</td>\n",
       "      <td>3.401380e-02</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>CAUD,HIPP,MDFG,PARL,SMTG,SUNI</td>\n",
       "      <td>CAUD,HIPP,MDFG,PARL,SMTG,SUNI</td>\n",
       "      <td>31117280</td>\n",
       "      <td>31118525</td>\n",
       "      <td>608.5</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>SUNI</td>\n",
       "      <td>31117841</td>\n",
       "      <td>31118041</td>\n",
       "      <td>547.0</td>\n",
       "      <td>True</td>\n",
       "      <td>excitatory_neurons,inhibitory_neurons</td>\n",
       "      <td>microglia</td>\n",
       "      <td>31117415</td>\n",
       "      <td>31118239</td>\n",
       "      <td>121.0</td>\n",
       "      <td>0.022485</td>\n",
       "      <td>0.024622</td>\n",
       "      <td>0.044335</td>\n",
       "      <td>2</td>\n",
       "      <td>CGGAGGCGG</td>\n",
       "      <td>(C/T)GGAGGCGG</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>KAT8, multiple cells, also in AD?? Im surprisd...</td>\n",
       "      <td>23,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr11_86103106_86104633</td>\n",
       "      <td>rs1237999</td>\n",
       "      <td>chr11</td>\n",
       "      <td>86103988</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.342093</td>\n",
       "      <td>0.005153</td>\n",
       "      <td>0.373118</td>\n",
       "      <td>9.175715</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>PTMN</td>\n",
       "      <td>0.008599</td>\n",
       "      <td>0.010799</td>\n",
       "      <td>0.347583</td>\n",
       "      <td>True</td>\n",
       "      <td>0.030718</td>\n",
       "      <td>True</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11</td>\n",
       "      <td>86103988</td>\n",
       "      <td>Alzheimers_Jansen_2018</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>16</td>\n",
       "      <td>163</td>\n",
       "      <td>11_86103988</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>7.140000e-16</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>CAUD,HIPP,MDFG,PARL,PTMN,SMTG,SUNI</td>\n",
       "      <td>CAUD,HIPP,MDFG,PARL,PTMN,SMTG,SUNI</td>\n",
       "      <td>86103138</td>\n",
       "      <td>86104596</td>\n",
       "      <td>121.0</td>\n",
       "      <td>True</td>\n",
       "      <td>STRI</td>\n",
       "      <td>STRI</td>\n",
       "      <td>86103829</td>\n",
       "      <td>86104179</td>\n",
       "      <td>16.0</td>\n",
       "      <td>True</td>\n",
       "      <td>doublets,neurons_unknown,oligodendrocytes</td>\n",
       "      <td>doublets,neurons_unknown,oligodendrocytes</td>\n",
       "      <td>86103388</td>\n",
       "      <td>86104560</td>\n",
       "      <td>14.0</td>\n",
       "      <td>0.001043</td>\n",
       "      <td>0.001603</td>\n",
       "      <td>0.006443</td>\n",
       "      <td>2</td>\n",
       "      <td>TATGAGTCACC</td>\n",
       "      <td>TAT(G/A)AGTCACC</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>PICALM, oligo specific</td>\n",
       "      <td>1,3,7,13,18,19,20,21,22,23,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr11_60251518_60251763</td>\n",
       "      <td>rs636317</td>\n",
       "      <td>chr11</td>\n",
       "      <td>60251677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.833333</td>\n",
       "      <td>0.387836</td>\n",
       "      <td>0.007736</td>\n",
       "      <td>0.491990</td>\n",
       "      <td>18.930226</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>MDFG</td>\n",
       "      <td>0.008743</td>\n",
       "      <td>0.008745</td>\n",
       "      <td>0.190429</td>\n",
       "      <td>True</td>\n",
       "      <td>0.024660</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>11</td>\n",
       "      <td>60251677</td>\n",
       "      <td>Alzheimers_Jansen_2018</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>11</td>\n",
       "      <td>385</td>\n",
       "      <td>11_60251677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>2.830000e-13</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>MDFG</td>\n",
       "      <td>60264747</td>\n",
       "      <td>60266009</td>\n",
       "      <td>13070.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>ISOC</td>\n",
       "      <td>60265062</td>\n",
       "      <td>60265262</td>\n",
       "      <td>13385.0</td>\n",
       "      <td>True</td>\n",
       "      <td>excitatory_neurons,microglia</td>\n",
       "      <td>excitatory_neurons,microglia</td>\n",
       "      <td>60251481</td>\n",
       "      <td>60252738</td>\n",
       "      <td>432.5</td>\n",
       "      <td>0.000131</td>\n",
       "      <td>0.000235</td>\n",
       "      <td>0.027467</td>\n",
       "      <td>2</td>\n",
       "      <td>TGCTGCCATCTGCTGGGAA</td>\n",
       "      <td>TGCTGC(C/T)ATCTGCTGGGAA</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>MS4A4A, microglia specific</td>\n",
       "      <td>1,4,19,20,24,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chr8_22599221_22600284</td>\n",
       "      <td>rs755934</td>\n",
       "      <td>chr8</td>\n",
       "      <td>22599692</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.714286</td>\n",
       "      <td>0.385599</td>\n",
       "      <td>0.000215</td>\n",
       "      <td>0.428795</td>\n",
       "      <td>24.758803</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>SUNI</td>\n",
       "      <td>0.002970</td>\n",
       "      <td>0.002970</td>\n",
       "      <td>0.099234</td>\n",
       "      <td>True</td>\n",
       "      <td>0.043506</td>\n",
       "      <td>False</td>\n",
       "      <td>0.808324</td>\n",
       "      <td>8</td>\n",
       "      <td>22668467</td>\n",
       "      <td>Chang_23andMe_Parkinsons</td>\n",
       "      <td>LD</td>\n",
       "      <td>125</td>\n",
       "      <td>32</td>\n",
       "      <td>8_22668467</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>5.107430e-02</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>PARL</td>\n",
       "      <td>22597288</td>\n",
       "      <td>22597925</td>\n",
       "      <td>1751.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>ISOC</td>\n",
       "      <td>22597666</td>\n",
       "      <td>22597887</td>\n",
       "      <td>1805.0</td>\n",
       "      <td>True</td>\n",
       "      <td>astrocytes,inhibitory_neurons,microglia,opcs</td>\n",
       "      <td>astrocytes,inhibitory_neurons,microglia,opcs</td>\n",
       "      <td>22598476</td>\n",
       "      <td>22600229</td>\n",
       "      <td>339.5</td>\n",
       "      <td>0.043331</td>\n",
       "      <td>0.032394</td>\n",
       "      <td>0.092178</td>\n",
       "      <td>1</td>\n",
       "      <td>GCATTGGCCG</td>\n",
       "      <td>GCA(T/G)TGGCCG</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>---</td>\n",
       "      <td>7,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr7_100217213_100217819</td>\n",
       "      <td>rs1727139</td>\n",
       "      <td>chr7</td>\n",
       "      <td>100217744</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.583333</td>\n",
       "      <td>0.346145</td>\n",
       "      <td>0.000634</td>\n",
       "      <td>0.436404</td>\n",
       "      <td>16.179343</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>MDFG</td>\n",
       "      <td>0.008799</td>\n",
       "      <td>0.008799</td>\n",
       "      <td>0.190429</td>\n",
       "      <td>True</td>\n",
       "      <td>0.024660</td>\n",
       "      <td>False</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>7</td>\n",
       "      <td>100217744</td>\n",
       "      <td>Alzheimers_Jansen_2018</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>117</td>\n",
       "      <td>109</td>\n",
       "      <td>7_100217744</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>9.850000e-10</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>PTMN</td>\n",
       "      <td>PTMN</td>\n",
       "      <td>100217219</td>\n",
       "      <td>100217897</td>\n",
       "      <td>186.0</td>\n",
       "      <td>False</td>\n",
       "      <td>none</td>\n",
       "      <td>ALLO</td>\n",
       "      <td>100218052</td>\n",
       "      <td>100218259</td>\n",
       "      <td>308.0</td>\n",
       "      <td>True</td>\n",
       "      <td>microglia,oligodendrocytes</td>\n",
       "      <td>microglia,oligodendrocytes</td>\n",
       "      <td>100216695</td>\n",
       "      <td>100217925</td>\n",
       "      <td>403.5</td>\n",
       "      <td>0.039411</td>\n",
       "      <td>0.038600</td>\n",
       "      <td>0.120150</td>\n",
       "      <td>1</td>\n",
       "      <td>GCGGAGAGAGAGAG</td>\n",
       "      <td>G(C/T)GGAGAGAGAGAG</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>---</td>\n",
       "      <td>9,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr2_95323422_95324085</td>\n",
       "      <td>rs3755519</td>\n",
       "      <td>chr2</td>\n",
       "      <td>95323524</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>0.710526</td>\n",
       "      <td>0.377405</td>\n",
       "      <td>0.000066</td>\n",
       "      <td>0.419191</td>\n",
       "      <td>15.602100</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>SMTG</td>\n",
       "      <td>0.003689</td>\n",
       "      <td>0.003689</td>\n",
       "      <td>0.280309</td>\n",
       "      <td>True</td>\n",
       "      <td>0.032534</td>\n",
       "      <td>True</td>\n",
       "      <td>0.954653</td>\n",
       "      <td>2</td>\n",
       "      <td>95335195</td>\n",
       "      <td>Nalls_23andMe</td>\n",
       "      <td>LD</td>\n",
       "      <td>85</td>\n",
       "      <td>94</td>\n",
       "      <td>2_95335195</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>+</td>\n",
       "      <td>7.682450e-03</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>CAUD,MDFG,PARL,SMTG</td>\n",
       "      <td>CAUD,MDFG,PARL,SMTG</td>\n",
       "      <td>95323063</td>\n",
       "      <td>95324126</td>\n",
       "      <td>64.5</td>\n",
       "      <td>True</td>\n",
       "      <td>ISOC,STRI</td>\n",
       "      <td>ISOC,STRI</td>\n",
       "      <td>95323441</td>\n",
       "      <td>95323644</td>\n",
       "      <td>18.5</td>\n",
       "      <td>True</td>\n",
       "      <td>excitatory_neurons,inhibitory_neurons</td>\n",
       "      <td>excitatory_neurons,inhibitory_neurons</td>\n",
       "      <td>95323307</td>\n",
       "      <td>95323985</td>\n",
       "      <td>122.0</td>\n",
       "      <td>0.011536</td>\n",
       "      <td>0.014240</td>\n",
       "      <td>0.146132</td>\n",
       "      <td>1</td>\n",
       "      <td>AGTGCTCTCTG</td>\n",
       "      <td>(A/T)GTGCTCTCTG</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>---</td>\n",
       "      <td>1,3,4,5,7,10,14,15,22,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr19_18461919_18462167</td>\n",
       "      <td>rs8103622</td>\n",
       "      <td>chr19</td>\n",
       "      <td>18462024</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.375000</td>\n",
       "      <td>0.368914</td>\n",
       "      <td>0.000074</td>\n",
       "      <td>0.546707</td>\n",
       "      <td>16.761091</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>MDTG</td>\n",
       "      <td>0.020112</td>\n",
       "      <td>0.020112</td>\n",
       "      <td>0.501839</td>\n",
       "      <td>True</td>\n",
       "      <td>0.025700</td>\n",
       "      <td>False</td>\n",
       "      <td>0.988859</td>\n",
       "      <td>19</td>\n",
       "      <td>18448066</td>\n",
       "      <td>Alzheimers_Jansen_2018</td>\n",
       "      <td>LD</td>\n",
       "      <td>69</td>\n",
       "      <td>73</td>\n",
       "      <td>19_18448066</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>+</td>\n",
       "      <td>7.305735e-02</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>MDFG,PTMN,SMTG</td>\n",
       "      <td>MDFG,PTMN,SMTG</td>\n",
       "      <td>18461893</td>\n",
       "      <td>18462180</td>\n",
       "      <td>11.0</td>\n",
       "      <td>True</td>\n",
       "      <td>ISOC,STRI</td>\n",
       "      <td>ISOC,STRI</td>\n",
       "      <td>18461893</td>\n",
       "      <td>18462093</td>\n",
       "      <td>31.0</td>\n",
       "      <td>True</td>\n",
       "      <td>excitatory_neurons,inhibitory_neurons,oligoden...</td>\n",
       "      <td>excitatory_neurons,inhibitory_neurons,oligoden...</td>\n",
       "      <td>18461930</td>\n",
       "      <td>18462272</td>\n",
       "      <td>77.0</td>\n",
       "      <td>0.036737</td>\n",
       "      <td>0.039399</td>\n",
       "      <td>0.389165</td>\n",
       "      <td>1</td>\n",
       "      <td>AGAGCGCCACCTGCT</td>\n",
       "      <td>AGAG(C/T)GCCACCTGCT</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>http://mitra.stanford.edu/kundaje/projects/alz...</td>\n",
       "      <td>---</td>\n",
       "      <td>2,11,20,21,22,24,</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     feature        rsid  chrom    snp_pos ref alt  \\\n",
       "11     chr17_4901692_4902448  rs79436576  chr17    4901872   C   T   \n",
       "16   chr17_45825137_45827395  rs62057150  chr17   45826182   C   T   \n",
       "13   chr17_45702992_45704386    rs968027  chr17   45703884   C   T   \n",
       "9    chr16_31117024_31117373   rs4889619  chr16   31117294   C   T   \n",
       "7    chr11_86103106_86104633   rs1237999  chr11   86103988   G   A   \n",
       "0    chr11_60251518_60251763    rs636317  chr11   60251677   C   T   \n",
       "39    chr8_22599221_22600284    rs755934   chr8   22599692   T   G   \n",
       "36  chr7_100217213_100217819   rs1727139   chr7  100217744   C   T   \n",
       "30    chr2_95323422_95324085   rs3755519   chr2   95323524   A   T   \n",
       "28   chr19_18461919_18462167   rs8103622  chr19   18462024   C   T   \n",
       "\n",
       "    allele_freq  effect_size  mapping_error_rate  ref_allele_mapping_bias  \\\n",
       "11     0.090909     0.673282            0.005003                 0.514530   \n",
       "16     0.555556     0.662942            0.135228                 0.132052   \n",
       "13     0.906250     0.663832            0.083686                 0.354419   \n",
       "9      0.888889     0.611059            0.008871                 0.476374   \n",
       "7      0.500000     0.342093            0.005153                 0.373118   \n",
       "0      0.833333     0.387836            0.007736                 0.491990   \n",
       "39     0.714286     0.385599            0.000215                 0.428795   \n",
       "36     0.583333     0.346145            0.000634                 0.436404   \n",
       "30     0.710526     0.377405            0.000066                 0.419191   \n",
       "28     0.375000     0.368914            0.000074                 0.546707   \n",
       "\n",
       "    overdispersion  snp_id_within_region  num_feature_snps  num_testing_snps  \\\n",
       "11       33.035472                     0                 3                 3   \n",
       "16        2.083686                     8                14                14   \n",
       "13        2.166129                     2                 5                 5   \n",
       "9         8.603434                     0                 2                 2   \n",
       "7         9.175715                     1                 2                 2   \n",
       "0        18.930226                     0                 2                 2   \n",
       "39       24.758803                     0                 1                 1   \n",
       "36       16.179343                     0                 1                 1   \n",
       "30       15.602100                     0                 1                 1   \n",
       "28       16.761091                     0                 1                 1   \n",
       "\n",
       "    in_feature region  raw_pvalue  cis_window_qvalue  genomewide_bh_qvalue  \\\n",
       "11        True   PTMN    0.019599           0.019988              0.348312   \n",
       "16        True   PARL    0.009645           0.015776              0.448529   \n",
       "13        True   CAUD    0.000616           0.000715              0.018553   \n",
       "9         True   PARL    0.022687           0.029441              0.572968   \n",
       "7         True   PTMN    0.008599           0.010799              0.347583   \n",
       "0         True   MDFG    0.008743           0.008745              0.190429   \n",
       "39        True   SUNI    0.002970           0.002970              0.099234   \n",
       "36        True   MDFG    0.008799           0.008799              0.190429   \n",
       "30        True   SMTG    0.003689           0.003689              0.280309   \n",
       "28        True   MDTG    0.020112           0.020112              0.501839   \n",
       "\n",
       "    permutation_significant  fdr10_threshold  in_original  r2_with_ld_tag  \\\n",
       "11                     True         0.030718        False        0.873516   \n",
       "16                     True         0.031533        False        1.000000   \n",
       "13                     True         0.005189        False        1.000000   \n",
       "9                      True         0.031533        False        0.919429   \n",
       "7                      True         0.030718         True        1.000000   \n",
       "0                      True         0.024660        False        1.000000   \n",
       "39                     True         0.043506        False        0.808324   \n",
       "36                     True         0.024660        False        1.000000   \n",
       "30                     True         0.032534         True        0.954653   \n",
       "28                     True         0.025700        False        0.988859   \n",
       "\n",
       "    ld_tag_chr  ld_tag_pos               source_gwas source  locus_num  \\\n",
       "11          17     4860256    Alzheimers_Kunkle_2019     LD         55   \n",
       "16          17    45826182           23andme_PD_hg38   GWAS         54   \n",
       "13          17    45703884           23andme_PD_hg38   GWAS         54   \n",
       "9           16    31110472  Chang_23andMe_Parkinsons     LD         48   \n",
       "7           11    86103988    Alzheimers_Jansen_2018   GWAS         16   \n",
       "0           11    60251677    Alzheimers_Jansen_2018   GWAS         11   \n",
       "39           8    22668467  Chang_23andMe_Parkinsons     LD        125   \n",
       "36           7   100217744    Alzheimers_Jansen_2018   GWAS        117   \n",
       "30           2    95335195             Nalls_23andMe     LD         85   \n",
       "28          19    18448066    Alzheimers_Jansen_2018     LD         69   \n",
       "\n",
       "    number_ld_buddies ld_tag_locus effect_allele noneffect_allele direction  \\\n",
       "11                 80   17_4860256             T                C         +   \n",
       "16               2496  17_45826182             T                C         -   \n",
       "13               2496  17_45703884             T                C         -   \n",
       "9                  47  16_31110472             T                C         +   \n",
       "7                 163  11_86103988             G                A         -   \n",
       "0                 385  11_60251677             C                T         -   \n",
       "39                 32   8_22668467             T                G         -   \n",
       "36                109  7_100217744             C                T         -   \n",
       "30                 94   2_95335195             T                A         +   \n",
       "28                 73  19_18448066             T                C         +   \n",
       "\n",
       "          pvalue  has_coloc  direct_atac_overlap_narrow_tissue_regions  \\\n",
       "11  3.060000e-04       True                                       True   \n",
       "16  1.330000e-22       True                                       True   \n",
       "13  2.070000e-22       True                                       True   \n",
       "9   3.401380e-02      False                                       True   \n",
       "7   7.140000e-16      False                                       True   \n",
       "0   2.830000e-13      False                                      False   \n",
       "39  5.107430e-02      False                                      False   \n",
       "36  9.850000e-10       True                                       True   \n",
       "30  7.682450e-03      False                                       True   \n",
       "28  7.305735e-02       True                                       True   \n",
       "\n",
       "   containing_atac_tissues_narrow_tissue_regions  \\\n",
       "11                           HIPP,MDFG,PARL,SMTG   \n",
       "16            CAUD,HIPP,MDFG,PARL,PTMN,SMTG,SUNI   \n",
       "13                           CAUD,PARL,PTMN,SMTG   \n",
       "9                  CAUD,HIPP,MDFG,PARL,SMTG,SUNI   \n",
       "7             CAUD,HIPP,MDFG,PARL,PTMN,SMTG,SUNI   \n",
       "0                                           none   \n",
       "39                                          none   \n",
       "36                                          PTMN   \n",
       "30                           CAUD,MDFG,PARL,SMTG   \n",
       "28                                MDFG,PTMN,SMTG   \n",
       "\n",
       "   nearest_atac_tissue_narrow_tissue_regions  start_narrow_tissue_regions  \\\n",
       "11                       HIPP,MDFG,PARL,SMTG                      4901724   \n",
       "16        CAUD,HIPP,MDFG,PARL,PTMN,SMTG,SUNI                     45825194   \n",
       "13                       CAUD,PARL,PTMN,SMTG                     45702996   \n",
       "9              CAUD,HIPP,MDFG,PARL,SMTG,SUNI                     31117280   \n",
       "7         CAUD,HIPP,MDFG,PARL,PTMN,SMTG,SUNI                     86103138   \n",
       "0                                       MDFG                     60264747   \n",
       "39                                      PARL                     22597288   \n",
       "36                                      PTMN                    100217219   \n",
       "30                       CAUD,MDFG,PARL,SMTG                     95323063   \n",
       "28                            MDFG,PTMN,SMTG                     18461893   \n",
       "\n",
       "    end_narrow_tissue_regions  dist_narrow_tissue_regions  \\\n",
       "11                    4903885                       904.0   \n",
       "16                   45826634                       268.0   \n",
       "13                   45706091                       659.5   \n",
       "9                    31118525                       608.5   \n",
       "7                    86104596                       121.0   \n",
       "0                    60266009                     13070.0   \n",
       "39                   22597925                      1751.0   \n",
       "36                  100217897                       186.0   \n",
       "30                   95324126                        64.5   \n",
       "28                   18462180                        11.0   \n",
       "\n",
       "    direct_atac_overlap_broad_tissue_regions  \\\n",
       "11                                      True   \n",
       "16                                     False   \n",
       "13                                     False   \n",
       "9                                      False   \n",
       "7                                       True   \n",
       "0                                      False   \n",
       "39                                     False   \n",
       "36                                     False   \n",
       "30                                      True   \n",
       "28                                      True   \n",
       "\n",
       "   containing_atac_tissues_broad_tissue_regions  \\\n",
       "11                                         ISOC   \n",
       "16                                         none   \n",
       "13                                         none   \n",
       "9                                          none   \n",
       "7                                          STRI   \n",
       "0                                          none   \n",
       "39                                         none   \n",
       "36                                         none   \n",
       "30                                    ISOC,STRI   \n",
       "28                                    ISOC,STRI   \n",
       "\n",
       "   nearest_atac_tissue_broad_tissue_regions  start_broad_tissue_regions  \\\n",
       "11                                     ISOC                     4901711   \n",
       "16                                     ALLO                    45826333   \n",
       "13                                     ISOC                    45703478   \n",
       "9                                      SUNI                    31117841   \n",
       "7                                      STRI                    86103829   \n",
       "0                                      ISOC                    60265062   \n",
       "39                                     ISOC                    22597666   \n",
       "36                                     ALLO                   100218052   \n",
       "30                                ISOC,STRI                    95323441   \n",
       "28                                ISOC,STRI                    18461893   \n",
       "\n",
       "    end_broad_tissue_regions  dist_broad_tissue_regions  \\\n",
       "11                   4901911                       61.0   \n",
       "16                  45826558                      151.0   \n",
       "13                  45703678                      206.0   \n",
       "9                   31118041                      547.0   \n",
       "7                   86104179                       16.0   \n",
       "0                   60265262                    13385.0   \n",
       "39                  22597887                     1805.0   \n",
       "36                 100218259                      308.0   \n",
       "30                  95323644                       18.5   \n",
       "28                  18462093                       31.0   \n",
       "\n",
       "    direct_atac_overlap_single_cell  \\\n",
       "11                             True   \n",
       "16                             True   \n",
       "13                             True   \n",
       "9                              True   \n",
       "7                              True   \n",
       "0                              True   \n",
       "39                             True   \n",
       "36                             True   \n",
       "30                             True   \n",
       "28                             True   \n",
       "\n",
       "                  containing_atac_tissues_single_cell  \\\n",
       "11              astrocytes,microglia,oligodendrocytes   \n",
       "16                                   oligodendrocytes   \n",
       "13              excitatory_neurons,inhibitory_neurons   \n",
       "9               excitatory_neurons,inhibitory_neurons   \n",
       "7           doublets,neurons_unknown,oligodendrocytes   \n",
       "0                        excitatory_neurons,microglia   \n",
       "39       astrocytes,inhibitory_neurons,microglia,opcs   \n",
       "36                         microglia,oligodendrocytes   \n",
       "30              excitatory_neurons,inhibitory_neurons   \n",
       "28  excitatory_neurons,inhibitory_neurons,oligoden...   \n",
       "\n",
       "                      nearest_atac_tissue_single_cell  start_single_cell  \\\n",
       "11              astrocytes,microglia,oligodendrocytes            4901620   \n",
       "16                                   oligodendrocytes           45825230   \n",
       "13              excitatory_neurons,inhibitory_neurons           45703290   \n",
       "9                                           microglia           31117415   \n",
       "7           doublets,neurons_unknown,oligodendrocytes           86103388   \n",
       "0                        excitatory_neurons,microglia           60251481   \n",
       "39       astrocytes,inhibitory_neurons,microglia,opcs           22598476   \n",
       "36                         microglia,oligodendrocytes          100216695   \n",
       "30              excitatory_neurons,inhibitory_neurons           95323307   \n",
       "28  excitatory_neurons,inhibitory_neurons,oligoden...           18461930   \n",
       "\n",
       "    end_single_cell  dist_single_cell  ism_pval  delta_pval  explain_pval  \\\n",
       "11          4902717             296.5  0.004616    0.004449      0.041519   \n",
       "16         45826777             178.5  0.042619    0.045282      0.031732   \n",
       "13         45704925             223.5  0.007962    0.008333      0.041340   \n",
       "9          31118239             121.0  0.022485    0.024622      0.044335   \n",
       "7          86104560              14.0  0.001043    0.001603      0.006443   \n",
       "0          60252738             432.5  0.000131    0.000235      0.027467   \n",
       "39         22600229             339.5  0.043331    0.032394      0.092178   \n",
       "36        100217925             403.5  0.039411    0.038600      0.120150   \n",
       "30         95323985             122.0  0.011536    0.014240      0.146132   \n",
       "28         18462272              77.0  0.036737    0.039399      0.389165   \n",
       "\n",
       "    confidence                motif               diff_motif  \\\n",
       "11           2         TGGCCCCGCCCC         TGGCC(C/T)CGCCCC   \n",
       "16           2           CCATTGGTCG           CCAT(T/C)GGTCG   \n",
       "13           2              TGACTAA              TGAC(T/C)AA   \n",
       "9            2            CGGAGGCGG            (C/T)GGAGGCGG   \n",
       "7            2          TATGAGTCACC          TAT(G/A)AGTCACC   \n",
       "0            2  TGCTGCCATCTGCTGGGAA  TGCTGC(C/T)ATCTGCTGGGAA   \n",
       "39           1           GCATTGGCCG           GCA(T/G)TGGCCG   \n",
       "36           1       GCGGAGAGAGAGAG       G(C/T)GGAGAGAGAGAG   \n",
       "30           1          AGTGCTCTCTG          (A/T)GTGCTCTCTG   \n",
       "28           1      AGAGCGCCACCTGCT      AGAG(C/T)GCCACCTGCT   \n",
       "\n",
       "                                             fc_track  \\\n",
       "11  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "16  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "13  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "9   http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "7   http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "0   http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "39  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "36  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "30  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "28  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "\n",
       "                                           pval_track  \\\n",
       "11  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "16  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "13  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "9   http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "7   http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "0   http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "39  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "36  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "30  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "28  http://mitra.stanford.edu/kundaje/projects/alz...   \n",
       "\n",
       "                                           annotation  \\\n",
       "11  Many possible linked genes. ARRB2 - arrestin B...   \n",
       "16                                         MAPT Locus   \n",
       "13                                         MAPT Locus   \n",
       "9   KAT8, multiple cells, also in AD?? Im surprisd...   \n",
       "7                              PICALM, oligo specific   \n",
       "0                          MS4A4A, microglia specific   \n",
       "39                                                ---   \n",
       "36                                                ---   \n",
       "30                                                ---   \n",
       "28                                                ---   \n",
       "\n",
       "                          cluster  \n",
       "11  6,11,13,14,15,19,20,21,22,24,  \n",
       "16                            23,  \n",
       "13                             2,  \n",
       "9                             23,  \n",
       "7     1,3,7,13,18,19,20,21,22,23,  \n",
       "0                   1,4,19,20,24,  \n",
       "39                             7,  \n",
       "36                             9,  \n",
       "30         1,3,4,5,7,10,14,15,22,  \n",
       "28              2,11,20,21,22,24,  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "(19, 63)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "display(unique_orig_sig_merge.head(10))\n",
    "display(unique_orig_sig_merge.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "chr17_78431572_78433595     53\n",
       "chr17_45894052_45896463     24\n",
       "chr17_45825137_45827395     18\n",
       "chr17_45702992_45704386     13\n",
       "chr5_102971168_102972471     8\n",
       "chr17_45818390_45819780      7\n",
       "chr11_60251518_60251763      7\n",
       "chr17_45583892_45585718      4\n",
       "chr17_45740360_45742185      4\n",
       "chr16_31117024_31117373      4\n",
       "chr11_86103106_86104633      4\n",
       "chr11_86067972_86070115      4\n",
       "chr8_22622374_22623929       3\n",
       "chr17_4901692_4902448        3\n",
       "chr8_22599221_22600284       3\n",
       "chr17_45976701_45977374      2\n",
       "chr4_89834926_89835137       2\n",
       "chr2_127105481_127107473     2\n",
       "chr17_46130574_46131153      2\n",
       "chr17_45585766_45586646      2\n",
       "chr4_89762368_89763049       2\n",
       "chr3_48692599_48693032       2\n",
       "chr2_95323422_95324085       2\n",
       "chr17_45914420_45914901      2\n",
       "chr4_89722620_89722820       2\n",
       "chr17_45841532_45841760      1\n",
       "chr19_18513115_18513639      1\n",
       "chr16_70676530_70676768      1\n",
       "chr8_22672754_22674048       1\n",
       "chr2_101823575_101823819     1\n",
       "chr17_45907812_45908508      1\n",
       "chr3_48966889_48967612       1\n",
       "chr3_52190197_52191026       1\n",
       "chr17_45639355_45639798      1\n",
       "chr16_31092884_31093415      1\n",
       "chr17_45848512_45849812      1\n",
       "chr17_45783544_45785590      1\n",
       "chr6_27761922_27762971       1\n",
       "chr2_134717757_134719357     1\n",
       "chr1_161185016_161186504     1\n",
       "chr3_52530664_52531364       1\n",
       "chr7_65960407_65960696       1\n",
       "chr17_46118988_46119206      1\n",
       "chr1_232528515_232530038     1\n",
       "chr7_23184659_23185279       1\n",
       "chr7_100217213_100217819     1\n",
       "chr21_37451892_37452691      1\n",
       "chr11_60264784_60265960      1\n",
       "chr16_70686336_70686821      1\n",
       "chr5_102970963_102971163     1\n",
       "chr12_40499566_40500949      1\n",
       "chr17_45870069_45870629      1\n",
       "chr19_18461919_18462167      1\n",
       "chr17_45832143_45832976      1\n",
       "chr4_875302_876377           1\n",
       "Name: feature, dtype: int64"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gwas_trim_sig_merge['feature'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True    209\n",
       "Name: in_feature, dtype: int64"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gwas_trim_sig_merge['in_feature'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total Number of Unique Features:\n",
      "385725\n",
      "Total Number of Unique SNPs:\n",
      "9707\n",
      "Total Number of Unique SNPs in Features:\n",
      "1072\n",
      "Total Number of Unique Features with SNPs:\n",
      "616\n",
      "---\n",
      "Number of Tests:\n",
      "4026\n",
      "Number of Unique Feature-SNP Combos Tested:\n",
      "685\n",
      "Number of Unique SNPs Tested:\n",
      "685\n",
      "Number of Unique Features Tested:\n",
      "398\n",
      "---\n",
      "Number of Significant Feature-SNP Combos:\n",
      "209\n",
      "Number of Unique Significant Feature-SNP Combos:\n",
      "116\n",
      "Number of Unique Significant SNPs:\n",
      "116\n",
      "Number of Unique Significant Features:\n",
      "55\n",
      "---\n",
      "Number of Significant Feature-SNP Combos in ML intersect:\n",
      "40\n",
      "Number of Unique Significant Feature-SNP Combos in ML intersect:\n",
      "19\n",
      "Number of Unique Significant SNPs in ML intersect:\n",
      "19\n",
      "Number of Unique Significant Features in ML intersect:\n",
      "18\n",
      "---\n",
      "Number of SNPs from Ryan's List that were in features\n",
      "26\n",
      "Number of SNPs from Ryan's List that were tested\n",
      "23\n",
      "Intersect with Ryan's List\n",
      "14\n"
     ]
    }
   ],
   "source": [
    "print(\"Total Number of Unique Features:\")\n",
    "print(len(idr_counts))\n",
    "print(\"Total Number of Unique SNPs:\")\n",
    "all_unique_snps = adpd_stage3.sort_values(by='rsid')\n",
    "all_unique_snps.drop_duplicates(subset='rsid', inplace=True)\n",
    "print(len(all_unique_snps))\n",
    "print(\"Total Number of Unique SNPs in Features:\")\n",
    "new_intersect_bed = snps_bed.intersect(counts_bed, u=True, wa=True)\n",
    "new_intersect_df = pybedtools.BedTool.to_dataframe(new_intersect_bed, header=None)\n",
    "new_intersect_df.columns = ['chr', 'new_start', 'new_end', 'rsid']\n",
    "new_intersect_df.sort_values(by='rsid', inplace=True)\n",
    "new_intersect_df.drop_duplicates(subset='rsid', inplace=True)\n",
    "print(len(new_intersect_df))\n",
    "print(\"Total Number of Unique Features with SNPs:\")\n",
    "uniq_features_with_snps = intersect_counts.sort_values(by=['chrom', 'start'])\n",
    "uniq_features_with_snps.drop_duplicates(subset=['chrom', 'start'], inplace=True)\n",
    "print(len(uniq_features_with_snps))\n",
    "print('---')\n",
    "print(\"Number of Tests:\")\n",
    "print(len(combined_output))\n",
    "print(\"Number of Unique Feature-SNP Combos Tested:\")\n",
    "feature_snp_combos = combined_output.sort_values(by=['rsid', 'feature'])\n",
    "feature_snp_combos.drop_duplicates(subset=['rsid', 'feature'], inplace=True)\n",
    "print(len(feature_snp_combos))\n",
    "print(\"Number of Unique SNPs Tested:\")\n",
    "unique_snps = combined_output.sort_values(by=['rsid'])\n",
    "unique_snps.drop_duplicates(subset=['rsid'], inplace=True)\n",
    "print(len(unique_snps))\n",
    "print(\"Number of Unique Features Tested:\")\n",
    "unique_features = combined_output.sort_values(by=['feature'])\n",
    "unique_features.drop_duplicates(subset=['feature'], inplace=True)\n",
    "print(len(unique_features))\n",
    "print('---')\n",
    "print(\"Number of Significant Feature-SNP Combos:\")\n",
    "print(len(gwas_trim_sig_merge))\n",
    "print(\"Number of Unique Significant Feature-SNP Combos:\")\n",
    "sig_feature_snp_combos = gwas_trim_sig_merge.sort_values(by=['rsid', 'feature'])\n",
    "sig_feature_snp_combos.drop_duplicates(subset=['rsid', 'feature'], inplace=True)\n",
    "print(len(sig_feature_snp_combos))\n",
    "print(\"Number of Unique Significant SNPs:\")\n",
    "unique_sig_snps = gwas_trim_sig_merge.sort_values(by=['rsid'])\n",
    "unique_sig_snps.drop_duplicates(subset=['rsid'], inplace=True)\n",
    "print(len(unique_sig_snps))\n",
    "print(\"Number of Unique Significant Features:\")\n",
    "unique_sig_features = gwas_trim_sig_merge.sort_values(by=['feature'])\n",
    "unique_sig_features.drop_duplicates(subset=['feature'], inplace=True)\n",
    "print(len(unique_sig_features))\n",
    "print('---')\n",
    "print(\"Number of Significant Feature-SNP Combos in ML intersect:\")\n",
    "print(len(orig_sig_merge))\n",
    "print(\"Number of Unique Significant Feature-SNP Combos in ML intersect:\")\n",
    "sig_feature_snp_combos_ml = orig_sig_merge.sort_values(by=['rsid', 'feature'])\n",
    "sig_feature_snp_combos_ml.drop_duplicates(subset=['rsid', 'feature'], inplace=True)\n",
    "print(len(sig_feature_snp_combos_ml))\n",
    "print(\"Number of Unique Significant SNPs in ML intersect:\")\n",
    "unique_sig_snps_ml = orig_sig_merge.sort_values(by=['rsid'])\n",
    "unique_sig_snps_ml.drop_duplicates(subset=['rsid'], inplace=True)\n",
    "print(len(unique_sig_snps_ml))\n",
    "print(\"Number of Unique Significant Features in ML intersect:\")\n",
    "unique_sig_features_ml = orig_sig_merge.sort_values(by=['feature'])\n",
    "unique_sig_features_ml.drop_duplicates(subset=['feature'], inplace=True)\n",
    "print(len(unique_sig_features_ml))\n",
    "print('---')\n",
    "print('Number of SNPs from Ryan\\'s List that were in features')\n",
    "print(len([i for i in list(new_intersect_df['rsid']) if i in ryan_bias_snps]))\n",
    "print('Number of SNPs from Ryan\\'s List that were tested')\n",
    "print(len(unique_snps.loc[unique_snps['in_original'] == True]))\n",
    "print('Intersect with Ryan\\'s List')\n",
    "print(len(unique_gwas_trim_sig_merge.loc[unique_gwas_trim_sig_merge['in_original'] == True]))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
